---
title: "MGnifyR: interface to MGnify database"
date: "`r Sys.Date()`"
package: MGnifyR
output:
    BiocStyle::html_document:
        fig_height: 7
        fig_width: 10
        toc: yes
        toc_depth: 2
        number_sections: true
vignette: >
    %\VignetteIndexEntry{mia}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8}
bibliography: references.bib
---

# Introduction

`MGnifyR` is a package designed to ease access to the EBI's
[MGnify](https://www.ebi.ac.uk/metagenomics) resource, allowing searching and
retrieval of multiple datasets for downstream analysis.

The latest version of MGnifyR seamlessly integrates with the
[miaverse framework](https://microbiome.github.io/) providing access to
cutting-edge tools in microbiome down-stream analytics. 

# Installation

`MGnifyR` is currently hosted on GitHub, and can be installed using via
`devtools`. `MGnifyR` should be built using the following snippet.

```{r install, eval=FALSE}
BiocManager::install(MGnifyR)
```

# Load `MGnifyR` package

Once installed, `MGnifyR` is made available in the usual way.

```{r load_package}
library(MGnifyR)
```

# Create a client

All functions in `MGnifyR` make use of a `MgnifyClient` object to keep track
of the JSONAPI url, disk cache location and user access tokens. Thus the first
thing to do when starting any analysis is to instantiate this object. The
following snippet creates this.

```{r create_client, echo=TRUE, fig.keep='all', message = FALSE}
mg <- MgnifyClient()
```

# Functions for fetching the data

## Search data

Below, we fetch information on samples of drinking water.

```{r search_studies}
# Fetch studies
samples <- doQuery(
    mg,
    type = "samples",
    biome_name = "root:Environmental:Aquatic:Freshwater:Drinking water",
    max.hits = 10)

# For demonstrative purpose, take only few samples
set.seed(595)
samples <- samples[ sample(rownames(samples), 5), ]
```

## Find relevent **analyses** accessions

Now we want to find analysis accessions. Each sample might have multiple
analyses. Each analysis ID corresponds to a single run of a particular pipeline
on a single sample in a single study.

```{r convert_to_analyses}
analyses_accessions <- searchAnalysis(mg, "samples", samples$accession)

head(analyses_accessions)
```

## Fetch metadata

We can now check the metadata to get hint of what kind of data we have.

```{r get_metadata}
analyses_metadata <- getMetadata(mg, analyses_accessions)

head(analyses_metadata)
```

## Fetch microbiome data

After we have selected the data to fetch, we can use `getResult()`

The output is `r BiocStyle::Biocpkg("TreeSummarizedExperiment")` (`TreeSE`) or
`r BiocStyle::Biocpkg("MultiAssayExperiment")` (`MAE`) depending on the dataset.
If the dataset includes only taxonomic profiling data, the output is a single
`TreeSE`. If dataset includes also functional data, the output is multiple
`TreeSE` objects that are linked together by utilizing `MAE`.

```{r get_mae}
mae <- getResult(mg, accession = analyses_accessions)
mae
```

You can get access to individual `TreeSE` object in `MAE` by specifying
index or name.

```{r mae_access}
mae[[1]]
```

`TreeSE` object is uniquely positioned to support `SummarizedExperiment`-based
microbiome data manipulation and visualization. Moreover, it enables access
to `miaverse` tools. For example, we can estimate diversity of samples.

```{r calculate_diversity}
mae[[1]] <- estimateDiversity(mae[[1]], index = "shannon")

library(scater)

plotColData(mae[[1]], "shannon", x = "sample_environment.feature")
```


```{r plot_abundance}
# Agglomerate data
altExps(mae[[1]]) <- splitByRanks(mae[[1]])

library(miaViz)

# Plot top taxa
top_taxa <- getTopFeatures(altExp(mae[[1]], "Phylum"), 10)
plotAbundance(altExp(mae[[1]], "Phylum")[top_taxa, ], rank = "Phylum")
```

We can perform principal component analysis to microbial profiling data by
utilizing miaverse tools.

```{r pcoa}
# Apply relative transformation
mae[[1]] <- transformCounts(mae[[1]], method = "relabundance")
# Perform PCoA
mae[[1]] <- runMDS(
    mae[[1]], assay.type = "relabundance",
    FUN = vegan::vegdist, method = "bray")
# Plot
plotReducedDim(mae[[1]], "MDS", colour_by = "sample_environment.feature")
```

## Fetch sequence files

Finally, we can use `searchFile()` and `getFile()` to retrieve other MGnify
pipeline outputs such as merged sequence reads, assembled contigs, and details
of the functional analyses.

With `searchFile()`, we can search files from the database.

```{r get_download_urls}
# Find list of available downloads, and filter for 
dl_urls <- searchFile(mg, analyses_accessions, type = "analyses")
```

```{r show_tgt_urls}
target_urls <- dl_urls[
    dl_urls$attributes.description.label == "Predicted CDS with annotation", ]
head(target_urls)
```

Finally, we can download the files with `getFile()`.

```{r filter_dl_urls, message=FALSE}
# Just select a single file from the target_urls list for demonstration.
cached_location <- getFile(mg, target_urls$download_url[[1]])

# Where are the files?
c(cached_location)
```

```{r session_info}
sessionInfo()
```
