---
title: "MGnifyR"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{MGnifyR}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
library(knitr)
knitr::opts_chunk$set(
  collapse = TRUE,
  eval =FALSE,
  comment = "#>",
  cache = TRUE
)
```
<!-- #```{r, cache=TRUE} -->
<!-- #knitr::opts_current$get(c( -->
<!-- #  "cache", -->
<!-- #  "cache.path", -->
<!-- #  "cache.rebuild", -->
<!-- #  "dependson", -->
<!-- #  "autodep" -->
<!-- #)) -->
<!-- #``` -->

Ben Allen

<ben.allen@ncl.ac.uk>

[MGnifyR homepage](http://github.com/microbiome/MGnifyR)


# Introduction

`MGnifyR` is a package designed to ease access to the EBI's [MGnify](https://www.ebi.ac.uk/metagenomics) resource, allowing searching and retrieval of multiple datasets for downstream analysis. While MGnify pipelines are undoubtedly useful, as currently implemented they produce results on a stricly per-sample basis. While some whole study results are available, comparisons across studies are difficult. The `MGnifyR` package is designed to facilitate cross-study analyses by handling all the per-sample data retrieval and merging details internally, leaving the user free to perform the analysis as they see fit.

## Installation

`MGnifyR` is hosted on github, and is installed using via `devtools`. As things stand `MGnifyR` has dependencies on `ape`, `dplyr`, `plyr`, `httr`, `urltools` and `reshape2`, which should be installed automatically if missing. The `phyloseq` package is also required and should be installed according to the instructions at (https://joey711.github.io/phyloseq/install.html). If following through this vignette, `ggplot2` and `Biostrings` will also be required, but aren't essential for using MGnifyR (and will be brought in by installing `phyloseq` anyway.)

Once package dependencies are installed, `MGnifyR` should be built using the following snippet.

```{r devtools_install, eval=FALSE}
devtools::install_github("microbiome/MGnifyR")
```


## Getting started - creating the client
Once installed, `MGnifyR` is made available in the usual way.
```{r load_package}
library(MGnifyR)
```

All functions in `MGnifyR` make use of a `MgnifyClient` object to keep track of the JSONAPI url, disk cache location and user access tokens. Thus the first thing to do when starting any analysis is to instantiate this object. The following snippet creates this.

```{r create_client, echo=TRUE, fig.keep='all', message = FALSE}
mg <- MgnifyClient( usecache=T, cache_dir='~/.MGnify_cache')
```

It's recommended that local caching is enabled with `usecache = T`. Queries to the MGnify API can be quite slow, particularly when retrieving multipage results for many analyses (such as many `Interpro` results). Using a local disk cache can significantly speed up subsequent work, bypassing the need to re-query the API. Use of the cache should be entirely transparent, as the caching occurs at the raw data level. The cache can persist across `MGnifyR` sessions, and can even be used for multiple sessions simultaneously - provided that different sets of accessions are queried at once.

Optionaly a username and password may be specified during client creation, causing `MGnifyR` to attempt retrieval of an authentication token from the API. Doing so gives access to non-public results, such as those currently under an author imposed embargo period.

```{r create_client_passwd, eval=FALSE}
mg <- MgnifyClient(username="Webin-username", password="your-password", usecache = T)
```

## Searching MGnify

`MGnifyR` gives users access to the complete range of search functionality implemented in the MGnify JSON API. A single function `doQuery` is used to do perform this searching, allowing Studies, Samples, Runs and Accession to be interogated from a common interface. As with all MGnifyR functions the first argument `client` must be a valid `MgnifyClient` instance. The only remaining **required** parameter is `qtype`, specifying the type of data to be queried, and may be one of `studies`, `samples`, `runs`, `analyses` or `assemblies`. Other general parameters include `usecache` and `max.hits`. 

Unlike most other `MGnifyR` high level functions, caching is turned off by default for `doQuery`. New data and analyses are being added to MGnify all the time, so enabling caching by default may lead to out-of-date search results for long-lived sessions. However, it's easy to switch back on, and may be useful in many cases. Also, given the huge and ever increasing number of datasets available in MGnify, a limit to the number of results returned may be set using `max.hits`. By default this is set to 200, which for most exploratory queries should be sufficient. It may be increased or decreased by directly specifying `max.hits`, and disabled completely (no limit) by setting `max.hits=-1`.

In most cases we will want to be more specific about the search, and will also use either an `accession` parameter, or the many filter options available through the API, and discussed below. Specifying an `accession` id, which in the case of `samples`, `runs` and `assemblies` may be a vector of ids, returns a data.frame of metadata with one row per matching accession.

If `accession` is `NULL` (the default) then remaining parameters define the filters applied by the API to the search result. Details of these parameters are given in `help(doQuery)`. By way of example though, supposing we are interested in amplicon Illumina samples from the arctic, we might try the following query:

```{r search_studies, fig.keep='all', message = FALSE, eval=FALSE}
northpolar <- doQuery(mg, "samples", latitude_gte=60.0, experiment_type="amplicon", biome_name="Soil", instrument_platform = "Illumina", usecache = F )
northpolar[1:5,]
```

specifing an `accession` parameter will restrict results to just those matching that particular entry, be it a study, sample or run. For example, to retrieve information for study "MGYS00003725":


```{r search_studies_accession, message = FALSE, eval = FALSE}
study_samples <- doQuery(mg, "samples", study_accession="MGYS00003725", usecache=TRUE)
#kable(study_samples[1:10,])
study_samples[1:5,]
```


## Finding relevent `analyses` accessions
Having obtained a particular set of search hits, it's now time to retrieve the associated results. General automated analysis is complicated by the MGnify database design, wherein for example samples may be shared between multiple studies, or studies analysed multiple times using different versions of the pipeline.  Navigating these "many-to-one" relationships can be tricky, so `MGnifyR` resorts to using `analyses` accessions as it's canonical identifier. Each analysis corresponds to a single run of a particular pipeline on a single sample in a single study. The downside of this approach is that queries returning `studies`, `samples` (or anything other than `analyses`) accessions need converting to the corresponding `analyses`. 

`MGnifyR` therefore provides a few helper functions to handle this conversion - `mgnify_analyses_from_xxx`. Following on from our previous search, we have a list of `sample` accessions, so to convert to corresponding `analyses` we use:

```{r convert_to_analyses, fig.keep='all', results='hide', message = FALSE, eval = FALSE}
analyses_accessions <- mgnify_analyses_from_samples(mg, accession = study_samples$accession)
```

```{r show_accessions, eval = FALSE}
analyses_accessions[1:20]
```

A useful side effect of the above call is that some attribute metadata for each sample has now been retrieved and stored in the local cache. Thus subsequent API calls for these samples (which will occur multiple times in later steps) will be significantly faster.

It's important to be aware that the results of a `mgnify_analyses_from_xxx` command will not neccesarily be a one-to-one match with the input accessions. `MGnify` analysis runs are sometimes performed multiple times, perhaps using different versions of the pipeline. Thus further filtering of the result list may be required, but is easily performed and is illustrated in the next section.

<!-- `mgnify_get_accession_xxx` set of functions take as input a list of `analyses` accession ids. Thus it's necessary to convert the `studies`, `samples` or `runs` accessions identified with `doQuery` into corresponding `analyses` for further processing. This is performed with the -->

## Examining metadata

At this point we have a long list of analysis instances (with potential duplicates) corresponding to the samples previously found. We use the `mgnify_get_analyses_metadata` function to download and combine all associated `sample`, `run` and `study` metadata, which we then filter as required to include only the rows we want.

```{r get_metadata, fig.keep='all', results='hide', message = FALSE}
analyses_metadata <- mgnify_get_analyses_metadata(mg, analyses_accessions)
```
```{r show_metadata}
head(analyses_metadata)
```

The resulting data.frame has columns with names prefixed with their source type. For example, "sample_xxx" columns correspond to metadata gleaned from querying an accession's `sample` entry. MGnify allows quite flexible specification of arbitray metadata at submission time, in many cases leading to quite sparse `data.frame` results if accession queries are sourced from more than one study. For instance, if only one sample contains an entry for "sample_soil_PH", entries for other rows will be filled with `NA`. `MGnifyR` does not automatically clean these missing values - instead opting to allow the the user to choose the a correct action. The particular study we're looking at is from the marine biome, suppose we were interested in only those samples or analyses for which the sampling depth was known. The following snippet filters the full `data.frame` selecting only entries which contain a valid `sample_depth`. It's worth noting the `as.numeric` call to ensure the column is converted to `numeric` type before it is checked. *All* sample data from MGnifyR is initially retrieved as type `character`, and it's up to the user to make sure ostensibly numeric entries are converted properly.

```{r filter_show_metadata}
known_depths <- analyses_metadata[!is.na(as.numeric(analyses_metadata$sample_depth)),]
#How many are left?
dim(known_depths)
```


## Worked example.

Although the previous queries have been based on the results from `doQuery`, from now on we will concentrate on combining and comparing results from specific studies.  Since newly performed analyses are retrieved first in the `doQuery` call, it's likely that by the time this vignette is read, the query results will be different.  This is principally due to the rapid increase in MGnify submissions, leading to a potential lack of consistency between even closely spaced queries. As mentioned previously, it may be best to use `usecache=FALSE` for `doQuery` calls, to ensure queries are actually returning the latest data. *** REWRITE THIS BIT***.

For the remainder of this vignette however, we'll be comparing 3 ostensibly different studies. A study of saltmarsh soils from York University, human faecal samples from a survey of healthy Sardinians, and a set of samples from hydrothermal vents in the Mid-Cayman rise in the Carribbean Sea. To simplify things, only the first 20 samples from each study will be used. Furthermore, the intention is only to demonstrate the functionality of the MGnifyR package, rather than produce scientifically rigorous results.

```{r get_analyses, results='hide', fig.keep='all', message = FALSE}
soil <- searchAnalysis(mg, "study", "MGYS00001447")[1:20]
human <- mgnify_analyses_from_studies(mg, "MGYS00001442")[1:20]
marine <- mgnify_analyses_from_studies(mg, "MGYS00001282")[1:20]

all_accessions <- c(soil,human,marine)
all_accessions <- c(soil)
```

The first step with this new accession list is, as previously, to retrieve the associated metadata using `mgnify_get_analyses_metadata`, and as seen with the `doQuery` results, the returned `data.frame` contains a large number of columns. Being autogenerated and flexible, the column names can be a little difficult to predict, but examining `colnames(full_metadata)` should make things clearer.

```{r get__new_metadata, echo=TRUE, results='hide', fig.keep='all', message = FALSE}
full_metadata <- getMetadata(mg, all_accessions)
```
```{r show_new_metadata}
colnames(full_metadata)
head(full_metadata)
```

From `full_metadata` we get an idea of the type of data we're dealing with, and can extract useul information such as sequencing platform, source biome, etc etc. The next code snippet tallies a few of these columns to give an idea about what's available. The boxplot also indicates that while within study read counts are similar, we probably need to use some sort of normalization procedure when comparing across samples. We might also want to drop particularly low read coverage samples from further analysis.

```{r full_metatdata_explore}

# Load ggplot2 
library(ggplot2)

#Distribution of sample source material:
table(full_metadata$`sample_environment-material`)

#What sequencing machine(s) were used?
table(full_metadata$`sample_instrument model`)

# Boxplot of raw read counts:
ggplot(full_metadata, aes(x=study_accession, y=log(as.numeric(`analysis_Submitted nucleotide sequences`)))) + geom_boxplot(aes(group=study_accession)) + theme_bw() + ylab("log(submitted reads)")

```

## Converting analyses to `phyloseq` objects

Having selected the analyses we wish to examine further, `mgnify_analyses_phyloseq` is used to both download associated OTU tables and taxonomy, and join all results into a single `phyloseq` object. [phyloseq](https://joey711.github.io/phyloseq/) is becoming a defacto standard for taxonomic abundance *munging* in R. `phyloseq` objects integrate abundance, taxonomic, phylogenetic, sample and sequence data into a single object, with powerful facilities for filtering, processing and plotting the results. Although

To convert the existing list of analysis accessions to a phyloseq object, we simply use the following:

```{r get_phyloseq, echo=TRUE, results='hide', fig.keep='all', message = FALSE}
full_phyloseq <- mgnify_get_analyses_phyloseq(mg, full_metadata$analysis_accession)
```
```{r show_phyloseq}
full_phyloseq
```

  Once the results are available in phyloseq format, a plethora of analysis options become available. Please refer to the excellent phyloseq documnentation available at [https://joey711.github.io/phyloseq/] for more information. For the purposes this vignette, a couple of diversity plots are sufficient to demonstrate the functionality. The first plot nicely illustrates the differences in the taxonomic composition of the various samples, with a clear delineation between the three study groups.

```{r plot_taxa, echo=TRUE, fig.align="center", fig.height=4, fig.width=6, fig.keep='all', message = FALSE}
library(phyloseq)
library(ggplot2)

#rarefy the data (~sorry~) for alpha diversity
normed_ps <- rarefy_even_depth(full_phyloseq, rngseed=1)

class_ps <- tax_glom(normed_ps, "Class")
plot_bar(class_ps,  fill="Phylum")  + theme_bw() + theme(legend.position = "none")
```

We can also use `phyloseq`'s built-in diversity calculations. The boxplot below shows, in keeping with expectations, that the soil/sediment samples exhibit higher taxonomic diversity than both the seawater and gut samples. 

```{r plot_diversity, fig.keep='all', message = FALSE, fig.align="center", fig.height=4, fig.width=6}
alphadiversity = estimate_richness(normed_ps)

adf <- cbind.data.frame(phyloseq::sample_data(normed_ps)$`sample_environment.biome`, alphadiversity$InvSimpson)
colnames(adf) <- c("study","diversity")
ggplot(adf, aes(x=study, group=study, y=diversity)) + geom_boxplot() + theme_bw()
```


## Retrieving functional analysis results

For shotgun metagenomics samples (as opposed to amplicon based studies), recent pipelines also determine estimates of per-sample gene functionality. Functional counts for GO terms (either the full GO ontology or trimmed down GO-slim annotations), antiSMASH gene clusters, and Pfam protein families are available with `mgnify_get_analyses_results`. In keeping with previous MGNifyR commands, the first argument to `mgnify_get_analyses_results` is a `MgnifyClient` object, followed by a list of analysis accessions. `retrievelist` requires us to specify which functional estimations we wish to examine, and may be one or more of `go-slim`, `go-terms`, `interpro-identifiers` or `antismash-gene-clusters`. `retrievelist` may also be `all`, in which case all available results are retrieved, along with all taxonomic assignments.

Returning to the three study example used above, `go-slim` annotations are retrieved using the code below, with results being returned as a named list with entries for all types in `retrievelist`. If a particular result set is not available, the corresponding entry in the returned list will be `NULL`

```{r, get_functions, cache=TRUE, echo=TRUE, results='hide', fig.keep='all', message = FALSE}

func_res <- mgnify_get_analyses_results(mg, full_metadata$analysis_accession, retrievelist = "go-slim", bulk_dl = T, usecache = T)

goslim <- func_res$`go-slim`
```
```{r,show_goslim}
head(goslim)

```

Each named entry in the results is a single data.frame, with the first 2 or 3 (depending on the result type) columns being descriptors, and remaining columns containing corresponding counts for each sample. Perhaps confusingly the `accession` column refers to the functional accession rather than a particular MGnify accession. In this case it contains the GO term accessions, but might equally contain Interpro identifiers or AntiSMASH ids. 

The above call to `mgnify_get_analyses_results` included `bulk_dl=TRUE` which has the potential to significantly speed up data retrieval. MGnify makes its functional results available in two seperate ways, either on a per-analysis basis through the web api, or at the whole study level as large files, tab seperated (TSV), and with columns representing the results for each analysis. When `bulk_dl` is `FALSE`, `MGnifyR` queries the web api to get results which (given some functional analyses results may consist of thousands of entries) may take significant time. Setting `bulk_dl` to `TRUE` causes `MGnifyR` to determine the source study associated with a particular `analysis` and to instead download and parse its corresponding results file. Since this result file contains entries for all analyses associated with the study, by taking advantage of `MGnifyR`'s local caching this single download provides results for many future analyses. In some cases this affords several orders of magnitude speedup over the api query case. 

Unfortunately, column entries in the per-study results files do not always directly correspond to those from a particular analysis run, causing the retrieval to fail. The principal cause of this is believed to be the running of multiple analyses jobs on the same sample. Thus for reliability, `bulk_dl` is `FALSE` by default. As a general recommendation though, you should try setting it `TRUE` the first time `mgnify_get_analyses_results` is used on a set of accessions. If this fails, setting `bulk_dl` to `FALSE` will enable the more robust approach allowing the analysis to continue. It might take a while though. Hopefully in the future the sample/analysis correspondence mismatches will be fixed and the default `bulk_dl` will be switch to `TRUE`.


## Functional comparisons between samples
Best practice methods for the analysis of such functional data are still under development, but simple exploratory analysis may reveal interesting insights. For instance, the code below performs a simple "read depth" normalization on the returned `go-slim` dataset, followed by NMDS dimensional reduction and plotting. It's worth pointing out that we need to ensure row ordering remains consistent during these analyses. In this case, the ordering of accessions in `full_metadata` (retrieved earlier with `mgnify_get_analayses_metadata` is NOT consistent with the order of columns in the `goslim` results. We can fix this easily by indexing the metadata with `goslim` colulm names as shown below. Failing to take this into account could lead to invalid results. 

```{r, nmds_function, results='hide', fig.keep='all', message = FALSE, fig.align="center", fig.height=4, fig.width=6}
library(vegan)

#Find the per-sample raw read count for each sample
seqvect <- as.numeric(full_metadata[colnames(goslim)[-c(1,2,3)],"analysis_Nucleotide sequences after format-specific filtering"])

#scale factors for normalizing the go term results
scale_factors = 1/(seqvect/median(seqvect))

#Extract the numeric matrix
normalized= goslim[,-c(1,2,3)] * scale_factors

#Calculate NMDS 
nmds_res = vegan::metaMDS(t(normalized))

#Build the plot
results_df <-  merge(nmds_res$points, 
                     full_metadata[,c("analysis_accession","sample_environment-feature")], 
                     by.x="row.names", by.y="analysis_accession")

ggplot(results_df, aes(x=MDS1, y=MDS2)) + geom_point(aes_string(color="`sample_environment-feature`")) + theme_bw() 

```

The ordination plot shows a clear distinction between the gut and marine samples which is both unsurprising (given the two marine habitats) and reassuring (human guts are not similar to the sea)(REWRITE!!!). Taking the analysis further we might be interested in finding out if any particular GO terms are over or under represented between biome types or studies. Since we are essentially working with count data, it's tempting to use a negative binomial model to do this (c.f. differential gene expression). In the code below, we use the `glm.nb` model function 


```{r, differential_taxa, warning = FALSE}
 # IN PROGRESS - not working properly - or maybe there's just nothing to see.
library(reshape2)
library(dplyr)
library(MASS)
remerged <- cbind(goslim[,c(1,2,3)], as.data.frame(normalized))

longform <- melt(remerged, id.vars=c("accession", "description", "category"), variable.name="sample", value.name="abund")

longform <- merge(longform, full_metadata[,c("analysis_accession","sample_environment-feature")], by.x="sample", by.y="analysis_accession")

r <- longform  %>%
  filter(`sample_environment-feature` %in% c("hydrothermal vent", "sea shore")) %>%
  mutate(transformed_abund = log(abund+1)) %>%
  group_by(accession) %>%
  do(
    broom::tidy(glm.nb(transformed_abund ~ `sample_environment-feature`, .))
    ) 

#Which terms are "significantly" different after adjustment?
significant_goterms <- r[!grepl("Intercept", r$term) & (p.adjust(r$p.value) < 0.05),]$accession

goslim[goslim$accession %in% significant_goterms,c(1,2,3)]


```

while undoubtedly crude, the GO-terms for photosynthesis and bioluminescence appearing as being significantly differentially abundant between esturine (light) and deep sea vents (dark) lends some credence to the approach simple approach above. Clearly further research is needed...


## Downloading sequence files
Finally, we can use `mgnify_download` to retrieve other MGnify pipeline outputs such as merged sequence reads, assembled contigs, and details of the functional analyses. While it's possible to use the generic \code{.mgnify_retrieve_json} function to find the urls, parsing the nested lists it returns can be a little cumbersome. \code{mgnify_get_downloads} is a simple wrapper function whcih, when supplied a list of accessions, finds the urls of the files we're after. In most cases we'll want to filter the returned list down to only the files of interest, which is easily done on the resulting data.frame object. In addition to the actual download location (the `download_url` column), extra columns include file type, contents and compression. It's recommended that the `colnames` of the `data.frame` be examined to get a grasp on the available metadata. To demonstrate the process, the code below retrieves a data.frame containing all available downloads for each accession we've been examining previously. It then filters this to retain only those files corresponding retain the annotated amino acid sequence files.

```{r, get_download_urls, results='hide',message=FALSE}
#Find list of available downloads, and filter for 
dl_urls <- mgnify_get_download_urls(mg, full_metadata$analysis_accession, accession_type = "analyses")
```
```{r,show_tgt_urls}
target_urls <- dl_urls[dl_urls$attributes.description.label == "Predicted CDS with annotation",]
head(target_urls)
```
To list the types of available files, and guide the filtering, something like the following might be useful. 
```{r, list_descriptions}
table(dl_urls$attributes.description.label)
```

Unlike other `MGnifyR` functions, `mgnify_get_downloads` is not limited to `analyses`, and by specifying `accession_type` other results types may be found. For instance, while general `genome` functionality is not yet integrated into `MGnifyR`, we can retrieve associated files for a particular `genome` accession with the following:

```{r get_genome_urls, results='hide', eval=FALSE}
# FIXME
genome_urls <- mgnify_get_download_urls(mg, "MGYG-HGUT-04644", accession_type = "genomes")
```

```{r show_genome_urls, eval=FALSE}
genome_urls[c("id","attributes.file.format.name","download_url")]
```
Having found the a set of target urls, the final step is to use `mgnify_download` to actually retrieve the file. Unlike other functions, this only works with a single url location at once, so each entry in `target_urls` from above must be downloaded individually - easily done by either looping or `apply`ing over the list.

If the files are intended to be used with external programs, it might be easiest to provide a `file` parameter to the function call, which specifies a local filename for writing the file. By default `MGnifyR` will use the local cache, which can make getting to the file afterwards more awkward. Regardless, the default befahiour of `mgnify_download` is to retrive the file specified in the parameter `url`, save it to disk, and return the filepath it was saved to. 


```{r, filter_dl_urls, echo=T, message=FALSE}
#Just select a single file from the target_urls list for demonstration.

#Default behaviour - use local cache.
cached_location = mgnify_download(mg, target_urls$download_url[[1]])

#Specifying a file
specified_location = mgnify_download(mg, target_urls$download_url[[1]], file = "Coding_Sequences_1.fasta.gz")

#Where are the files?
c(cached_location,specified_location)
```

A second download option is available, which allows built-in parsing of the file. If we know ahead of time what processing will be performed, it may be possible to integrate it into a function, pass this function to `mgnify_download` as the `read_func` argument. The function in question should take a single argument (the complete pathname of the locally downloaded file) and the result of the call will be returned in place of the usual output filename. 

Alternatively the files could first be downloaded in the standard way, and then processed using this same function in a loop. Therefore in many cases the `read_func` parameter is redundant. However, many of the outputs from MGnify can be quite large, meaning local storage of many files may become an issue. By providing a `read_func` parameter (and necessarily setting `usecache=F`) analysis of a large number of datasets may be possible with minimal storage requirements.

To illustrate, suppose we were intereted in retrieving all detected sequences matching a particular PFAM motif in a set of analyses. The simple function below uses the `Biostrings` package to read an amino acid fasta file, searches for a matching PFAM tag in the sequence name, and then tallies up the unique sequences into a single data.frame row. In this case the PFAM motif identifies sequences coding for the amoC gene, found in both ammonia and methane oxidizing organisms, but any other filtering method could be used.

```{r, simple_parse_function}
library(Biostrings)

#Simple function to a count of unique sequences matching PFAM amoC/mmoC motif
getAmoCseqs <- function(fname){
  sequences <- readAAStringSet(fname)
  tgtvec <- grepl("PF04896", names(sequences))
  as.data.frame(as.list(table(as.character(sequences[tgtvec]))))
}
```

Having defined the function, it just remains to include it in the call to `mgnify_download`. 

```{r,do_download_with_read}

#Just download a single accession for demonstration, specifying a read_function
amoC_seq_counts <- mgnify_download(mg, target_urls$download_url[[1]] , read_func = getAmoCseqs, usecache = F)

amoC_seq_counts %>% t  

```


# Conclusion
So that's it - go use the `MGnifyR` package.





