<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Search MGnify database for studies, samples, runs, analyses, biomes, assemblies, and genomes. — doQuery • MGnifyR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Search MGnify database for studies, samples, runs, analyses, biomes, assemblies, and genomes. — doQuery"><meta name="description" content="Search MGnify database for studies, samples, runs, analyses, biomes,
assemblies, and genomes."><meta property="og:description" content="Search MGnify database for studies, samples, runs, analyses, biomes,
assemblies, and genomes."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MGnifyR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.30</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/MGnifyR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/MGnifyR_long.html">MGnifyR, extended vignette</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/EBI-Metagenomics/MGnifyR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Search MGnify database for studies, samples, runs, analyses, biomes, assemblies, and genomes.</h1>
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/devel/R/AllGenerics.R" class="external-link"><code>R/AllGenerics.R</code></a>, <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/devel/R/doQuery.R" class="external-link"><code>R/doQuery.R</code></a></small>
      <div class="d-none name"><code>doQuery.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Search MGnify database for studies, samples, runs, analyses, biomes,
assemblies, and genomes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">doQuery</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MgnifyClient'</span></span>
<span><span class="fu">doQuery</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"studies"</span>,</span>
<span>  accession <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  as.df <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  max.hits <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>A <code>MgnifyClient</code> object.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Remaining parameter key/value pairs may be supplied to filter
the returned values. Available options differ between <code>types</code>.
See discussion Details section for details.</p></dd>


<dt id="arg-type">type<a class="anchor" aria-label="anchor" href="#arg-type"></a></dt>
<dd><p>A single character value specifying the type of objects to
query. Must be one of the following options: <code>studies</code>, <code>samples</code>,
<code>runs</code>, <code>analyses</code>, <code>biomes</code>, <code>assemblies</code>,
<code>super-studies</code>, <code>experiment-types</code>, <code>pipelines</code>,
<code>pipeline-tools</code>, <code>publications</code>, <code>genomes</code>,
<code>genome-search</code>, <code>genome-search/gather</code>, <code>genome-catalogues</code>,
<code>genomeset</code>, <code>cogs</code>, <code>kegg-modules</code>, <code>kegg-classes</code>,
<code>antismash-geneclusters</code>, <code>annotations/go-terms</code>,
<code>annotations/interpro-identifiers</code>, <code>annotations/kegg-modules</code>,
<code>annotations/pfam-entries</code>, <code>annotations/kegg-orthologs</code>,
<code>annotations/genome-properties</code>,
<code>annotations/antismash-gene-clusters</code>, <code>annotations/organisms</code>, or
<code>mydata</code>.
(By default: <code>type = "studies"</code>)</p></dd>


<dt id="arg-accession">accession<a class="anchor" aria-label="anchor" href="#arg-accession"></a></dt>
<dd><p>A single character value or a vector of character values
specifying MGnify accession identifiers (of type <code>type</code>) or NULL. When
NULL, all results defined by other parameters are retrieved.
(By default: <code>accession = NULL</code>)</p></dd>


<dt id="arg-as-df">as.df<a class="anchor" aria-label="anchor" href="#arg-as-df"></a></dt>
<dd><p>A single boolean value specifying whether to return the
results as a data.frame or leave as a nested list. In most cases,
<code>as.df = TRUE</code> will make the most sense.
(By default: <code>as.df = TRUE</code>)</p></dd>


<dt id="arg-max-hits">max.hits<a class="anchor" aria-label="anchor" href="#arg-max-hits"></a></dt>
<dd><p>A single integer value specifying the maximum number of
results to return or FALSE. The actual number of results will actually be
higher than <code>max.hits</code>, as clipping only occurs on pagination page
boundaries. To disable the limit, set <code>max.hits = NULL</code>.
(By default: <code>max.hits = 200</code>)</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A nested list or data.frame containing the results of the query.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><code>doQuery</code> is a flexible query function, harnessing the "full"
power of the JSONAPI MGnify search filters. Search results may be filtered
by metadata value, associated study/sample/analyse etc.</p>
<p>See <a href="https://www.ebi.ac.uk/metagenomics/api/v1/" class="external-link">Api browser</a> for
information on MGnify database filters.
You can find help on customizing queries from
<a href="https://emg-docs.readthedocs.io/en/latest/api.html#customising-queries" class="external-link">here</a>.</p>
<p>For example the following filters are available:</p><ul><li><p><strong>studies</strong>: accession, biome_name, lineage, centre_name,
    include</p></li>
<li><p><strong>samples</strong>: accession, experiment_type, biome_name,
    lineage, geo_loc_name, latitude_gte, latitude_lte,
    longitude_gte, longitude_lte, species, instrument_model,
    instrument_platform, metadata_key, metadata_value_gte,
    metadata_value_lte, metadata_value, environment_material,
    environment_feature, study_accession, include</p></li>
<li><p><strong>runs</strong>: accession, experiment_type, biome_name, lineage,
    species, instrument_platform, instrument_model, metdata_key,
    metadata_value_gte, metadata_value_lte, metadata_value, sample_accession,
    study_accession, include</p></li>
<li><p><strong>analyses</strong>: biome_name, lineage, experiment_type, species,
    sample_accession, pipeline_version</p></li>
<li><p><strong>biomes</strong>: depth_gte, depth_lte</p></li>
<li><p><strong>assemblies</strong>: depth_gte, depth_lte</p></li>
</ul><p>Unfortunately it appears that in some cases, some of these filters don't work
as expected, so it is important to check the results returned match up with
what's expected. Even more unfortunately if there's an error in the parameter
specification, the query will run as if no filter parameters were present
at all. Thus the result will appear superficially correct but will infact
correspond to something completely different. This behaviour will hopefully
be fixed in future incarnations of the MGnifyR or JSONAPI, but for now users
should double check returned values.</p>
<p>It is currently not possible to combine queries of the same type in a single
call (for example to search for samples <em>between</em> latitude). However,
it is possible to run multiple queries and combine the results using set
operations in R to get the desired behaviour.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="MgnifyClient.html">MgnifyClient</a></span><span class="op">(</span>useCache <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Get a list of studies from the Agricultural Wastewater :</span></span></span>
<span class="r-in"><span><span class="va">agwaste_studies</span> <span class="op">&lt;-</span> <span class="fu">doQuery</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">mg</span>, <span class="st">"studies"</span>, biome_name<span class="op">=</span><span class="st">"Agricultural wastewater"</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Get all samples from a particular study</span></span></span>
<span class="r-in"><span><span class="va">samps</span> <span class="op">&lt;-</span> <span class="fu">doQuery</span><span class="op">(</span><span class="va">mg</span>, <span class="st">"samples"</span>, accession<span class="op">=</span><span class="st">"MGYS00004521"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Search polar samples</span></span></span>
<span class="r-in"><span><span class="va">samps_np</span> <span class="op">&lt;-</span> <span class="fu">doQuery</span><span class="op">(</span><span class="va">mg</span>, <span class="st">"samples"</span>, latitude_gte<span class="op">=</span><span class="fl">66</span>, max.hits<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">samps_sp</span> <span class="op">&lt;-</span> <span class="fu">doQuery</span><span class="op">(</span><span class="va">mg</span>, <span class="st">"samples"</span>, latitude_lte<span class="op">=</span><span class="op">-</span><span class="fl">66</span>, max.hits<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Search studies that have studied drinking water</span></span></span>
<span class="r-in"><span><span class="va">tbl</span> <span class="op">&lt;-</span> <span class="fu">doQuery</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">mg</span>,</span></span>
<span class="r-in"><span>    type <span class="op">=</span> <span class="st">"studies"</span>,</span></span>
<span class="r-in"><span>    biome_name <span class="op">=</span> <span class="st">"root:Environmental:Aquatic:Freshwater:Drinking water"</span>,</span></span>
<span class="r-in"><span>    max.hits <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tuomas Borman, Ben Allen, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

