<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Download any MGnify files, also including processed reads and identified protein sequences — getFile • MGnifyR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Download any MGnify files, also including processed reads and identified protein sequences — getFile"><meta name="description" content="Download any MGnify files, also including processed reads and
identified protein sequences
Listing files available for download"><meta property="og:description" content="Download any MGnify files, also including processed reads and
identified protein sequences
Listing files available for download"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MGnifyR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.30</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/MGnifyR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/MGnifyR_long.html">MGnifyR, extended vignette</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/EBI-Metagenomics/MGnifyR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Download any MGnify files, also including processed reads and identified protein sequences</h1>
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/devel/R/AllGenerics.R" class="external-link"><code>R/AllGenerics.R</code></a>, <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/devel/R/getFile.R" class="external-link"><code>R/getFile.R</code></a></small>
      <div class="d-none name"><code>getFile.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Download any MGnify files, also including processed reads and
identified protein sequences</p>
<p>Listing files available for download</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getFile</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">searchFile</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MgnifyClient'</span></span>
<span><span class="fu">getFile</span><span class="op">(</span><span class="va">x</span>, <span class="va">url</span>, file <span class="op">=</span> <span class="cn">NULL</span>, read.func <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MgnifyClient'</span></span>
<span><span class="fu">searchFile</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">accession</span>,</span>
<span>  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"studies"</span>, <span class="st">"samples"</span>, <span class="st">"analyses"</span>, <span class="st">"assemblies"</span>, <span class="st">"genomes"</span>, <span class="st">"run"</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>A <code>MgnifyClient</code> object.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments; not used currently.</p></dd>


<dt id="arg-url">url<a class="anchor" aria-label="anchor" href="#arg-url"></a></dt>
<dd><p>A single character value specifying the url address of the file
we wish to download.</p></dd>


<dt id="arg-file">file<a class="anchor" aria-label="anchor" href="#arg-file"></a></dt>
<dd><p>A single character value or NULL specifying an
optional local filename to use for saving the file. If <code>NULL</code>,
MGNify local cache settings will be used. If the file is intended to be
processed in a separate program, it may be sensible to provide a
meaningful <code>file</code>, rather than having to hunt through the
cache folders. If <code>file</code> is <code>NULL</code> and <code>useCache(client)</code>
is <code>FALSE</code>, the <code>read.func</code> parameter must be supplied or the
file will be downloaded and then deleted.
(By default: <code>file = NULL</code>)</p></dd>


<dt id="arg-read-func">read.func<a class="anchor" aria-label="anchor" href="#arg-read-func"></a></dt>
<dd><p>A function specifying an optional function to process the
downloaded file and return the results, rather than relying on post
processing. The primary use-case for this parameter is when local disk
space is limited and downloaded files can be quickly processed and
discarded. The function should take a single parameter, the downloaded
filename, and may return any valid R object.
(By default: <code>read.func = NULL</code>)</p></dd>


<dt id="arg-accession">accession<a class="anchor" aria-label="anchor" href="#arg-accession"></a></dt>
<dd><p>A single character value or a vector of character values
specifying accession IDs to return results for.</p></dd>


<dt id="arg-type">type<a class="anchor" aria-label="anchor" href="#arg-type"></a></dt>
<dd><p>A single character value specifying the type of objects to
query. Must be one of the following options: <code>analysis</code>, <code>samples</code>,
<code>studies</code>, <code>assembly</code>, <code>genome</code> or <code>run</code>.
(By default: <code>type = "samples"</code>)</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>For <code>getFile()</code>, either the local filename of the downloaded
file, be it either the location in the MGNifyR cache or file. If
<code>read.func</code> is used, its result will be returned.</p>
<p>For <code>searchFile()</code> <code>data.frame</code> containing all discovered
downloads. If multiple <code>accessions</code> are queried, the <code>accessions</code>
column may to filter the results - since rownames are not set (and wouldn't
make sense as each query will return multiple items)</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><code>getFile</code> is a convenient wrapper round generic the URL
downloading functionality in R, taking care of things like local
caching and authentication.</p>
<p><code>searchFile()</code> function is a wrapper function allowing easy
enumeration of downloads available for a given accession IDs.
Returns a single data.frame containing all available downloads and associated
metadata, including the url location and description. This can then be
filtered to extract the urls of interest, before actually
retrieving the files using <code>getFile()</code></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Make a client object</span></span></span>
<span class="r-in"><span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="MgnifyClient.html">MgnifyClient</a></span><span class="op">(</span>useCache <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create a vector of accession ids - these happen to be \code{analysis}</span></span></span>
<span class="r-in"><span><span class="co"># accessions</span></span></span>
<span class="r-in"><span><span class="va">accession_vect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MGYA00563876"</span>, <span class="st">"MGYA00563877"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">downloads</span> <span class="op">&lt;-</span> <span class="fu">searchFile</span><span class="op">(</span><span class="va">mg</span>, <span class="va">accession_vect</span>, <span class="st">"analyses"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Searching files...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |===================================                                   |  50%  |                                                                              |======================================================================| 100%</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Filter to find the urls of 16S encoding sequences</span></span></span>
<span class="r-in"><span><span class="va">url_list</span> <span class="op">&lt;-</span> <span class="va">downloads</span><span class="op">[</span></span></span>
<span class="r-in"><span>    <span class="va">downloads</span><span class="op">$</span><span class="va">attributes.description.label</span> <span class="op">==</span> <span class="st">"Contigs encoding SSU rRNA"</span>,</span></span>
<span class="r-in"><span>    <span class="st">"download_url"</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example 1:</span></span></span>
<span class="r-in"><span><span class="co"># Download the first file</span></span></span>
<span class="r-in"><span><span class="va">supplied_filename</span> <span class="op">&lt;-</span> <span class="fu">getFile</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">mg</span>, <span class="va">url_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, file<span class="op">=</span><span class="st">"SSU_file.fasta.gz"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Example 2:</span></span></span>
<span class="r-in"><span><span class="co"># Just use local caching</span></span></span>
<span class="r-in"><span><span class="va">cached_filename</span> <span class="op">&lt;-</span> <span class="fu">getFile</span><span class="op">(</span><span class="va">mg</span>, <span class="va">url_list</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example 3:</span></span></span>
<span class="r-in"><span><span class="co"># Using read.func to open the reads with readDNAStringSet from</span></span></span>
<span class="r-in"><span><span class="co"># \code{biostrings}. Without retaining on disk</span></span></span>
<span class="r-in"><span><span class="va">dna_seqs</span> <span class="op">&lt;-</span> <span class="fu">getFile</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">mg</span>, <span class="va">url_list</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span>, read.func <span class="op">=</span> <span class="va">readDNAStringSet</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Make a client object</span></span></span>
<span class="r-in"><span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="MgnifyClient.html">MgnifyClient</a></span><span class="op">(</span>useCache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Create a vector of accession ids - these happen to be \code{analysis}</span></span></span>
<span class="r-in"><span><span class="co"># accessions</span></span></span>
<span class="r-in"><span><span class="va">accession_vect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="st">"MGYA00563876"</span>, <span class="st">"MGYA00563877"</span>, <span class="st">"MGYA00563878"</span>,</span></span>
<span class="r-in"><span>    <span class="st">"MGYA00563879"</span>, <span class="st">"MGYA00563880"</span> <span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">downloads</span> <span class="op">&lt;-</span> <span class="fu">searchFile</span><span class="op">(</span><span class="va">mg</span>, <span class="va">accession_vect</span>, <span class="st">"analyses"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Searching files...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |==============                                                        |  20%  |                                                                              |============================                                          |  40%  |                                                                              |==========================================                            |  60%  |                                                                              |========================================================              |  80%  |                                                                              |======================================================================| 100%</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tuomas Borman, Ben Allen, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

