<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Get microbial and/or functional profiling data for a list of accessions"><title>Get microbial and/or functional profiling data for a list of accessions — getResult • MGnifyR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get microbial and/or functional profiling data for a list of accessions — getResult"><meta property="og:description" content="Get microbial and/or functional profiling data for a list of accessions"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MGnifyR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.27</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/MGnifyR.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/MGnifyR_long.html">MGnifyR, extended vignette</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/EBI-Metagenomics/MGnifyR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Get microbial and/or functional profiling data for a list of accessions</h1>
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/HEAD/R/AllGenerics.R" class="external-link"><code>R/AllGenerics.R</code></a>, <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/HEAD/R/getResult.R" class="external-link"><code>R/getResult.R</code></a></small>
      <div class="d-none name"><code>getResult.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Get microbial and/or functional profiling data for a list of accessions</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getResult</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for MgnifyClient</span></span>
<span><span class="fu">getResult</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">accession</span>,</span>
<span>  get.taxa <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  get.func <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  output <span class="op">=</span> <span class="st">"TreeSE"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>x</dt>
<dd><p>A <code>MgnifyClient</code> object.</p></dd>


<dt>...</dt>
<dd><p>optional arguments:</p><ul><li><p><strong>taxa.su</strong> A single character value specifying which taxa
  subunit results should be selected? Currently, taxonomy assignments in the
  MGnify pipelines rely on rRNA matches to existing databases
  (GreenGenes and SILVA), with later pipelines checking both the SSU and
  LSU portions of the rRNA sequence. <code>taxa.su</code> allows then selection
  of either the Small subunit (SSU) or Large subunit results
  in the final <code>TreeSummarizedExperiment</code> object. Older pipeline
  versions do not report results for both subunits, and thus for some
  accessions this value will have no effect.</p></li>
<li><p><strong>get.tree</strong> A single boolean value specifying whether to
  include available phylogenetic trees in the <code>TreeSummarizedExperiment</code>
  object. (By default: <code>get.tree = TRUE</code>)</p></li>
<li><p><strong>as.df</strong> A single boolean value enabled when
  <code>output = "list"</code>. The argument specifies whether return functional
  data as a named list (one entry per element in the output list) of
  data.frames, with each data.frame containing results for all requested
  accessions. If <code>FALSE</code>, The function returns a list of lists, each
  element consisting of results for a single accession. (By default:
  <code>as.df = TRUE</code>)</p></li>
<li><p><strong>bulk.dl</strong> A single boolean value specifying should
  MGnifyR attempt to speed things up by downloading
  relevant studies TSV results and only extracting the required columns,
  rather than using the JSONAPI interface. When getting results where
  multiple accessions share the same study, this option may result in
  significantly faster processing. However, there appear to be (quite a few)
  cases in the database where the TSV result columns do NOT match the
  expected accession names. This will hopefully be fixed in the future,
  but for now <code>bulk.dl</code> defaults to TRUE. When it does work, it can
  be orders of magnitude more efficient.
  (By default: <code>buld_dl = TRUE</code>)</p></li>
</ul></dd>


<dt>accession</dt>
<dd><p>A single character value or a vector of character values
specifying accession IDs to return results for.</p></dd>


<dt>get.taxa</dt>
<dd><p>A boolean value specifying whether to retrieve metagenomic
data. (By default: <code>get.taxa = TRUE</code>)</p></dd>


<dt>get.func</dt>
<dd><p>A boolean value or a single character value or a vector
character values specifying functional analysis types to retrieve. If
<code>get.func = TRUE</code>, all available functional datatypes are retrieved,
and if <code>FALSE</code>, functional data is not retrieved. The current list of
available types is "antismash-gene-clusters", "go-slim", "go-terms",
"interpro-identifiers", "taxonomy", "taxonomy-itsonedb",
"taxonomy-itsunite", "taxonomy-lsu", and "taxonomy-ssu". Note that
depending on the particular analysis type, pipeline version etc., not all
functional results will be available. (By default: <code>get.func = TRUE</code>)</p></dd>


<dt>output</dt>
<dd><p>A single character value specifying the format of an output.
Must be one of the following options: "TreeSE", "list", or "phyloseq".
(By default: <code>output = "TreeSE"</code>)</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>If only metagenomic data is retrieved, the result is returned in
<code>TreeSummarizedExperiment</code> object by default. The result can also be
returned as a <code>phyloseq</code> object or as a list of <code>data.frames</code>.
Note that <code>phyloseq</code> object can include only one phylogenetic tree
meaning that some taxa might be lost when data is subsetted based on tree.</p>


<p>When functional data is retrieved in addition to metagenomic data, the result
is returned as a <code>MultiAssayExperiment</code> object. Other options are a list
containing <code>phyloseq</code> object and <code>data.frames</code> or just
<code>data.frames</code>.</p>


<p>Functional data can be returned as a <code>MultiAssayExperiment</code> object or
as a list of <code>data.frames</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Given a set of analysis accessions and collection of annotation types,
the function queries the MGNify API and returns the results. This function
is convenient for retrieving highly structured (analysis vs counts) data on
certain instances. For example, BIOM files are downloaded automatically.
If you want just to retrieve raw data from the database, see <code>getData</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="getData.html">getData</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Create a client object</span></span></span>
<span class="r-in"><span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="MgnifyClient.html">MgnifyClient</a></span><span class="op">(</span>useCache <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Get OTU tables as TreeSE</span></span></span>
<span class="r-in"><span><span class="va">accession_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MGYA00377505"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">getResult</span><span class="op">(</span><span class="va">mg</span>, <span class="va">accession_list</span>, get.func<span class="op">=</span><span class="cn">FALSE</span>, get.taxa<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Fetching taxonomy data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Merging with full join...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1/1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="co"># Get functional data along with OTU tables as MAE</span></span></span>
<span class="r-in"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu">getResult</span><span class="op">(</span><span class="va">mg</span>, <span class="va">accession_list</span>, get.func<span class="op">=</span><span class="cn">TRUE</span>, get.taxa<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Get same data as list</span></span></span>
<span class="r-in"><span><span class="va">list</span> <span class="op">&lt;-</span> <span class="fu">getResult</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">mg</span>, <span class="va">accession_list</span>, get.func<span class="op">=</span><span class="cn">TRUE</span>, get.taxa<span class="op">=</span><span class="cn">TRUE</span>, output <span class="op">=</span> <span class="st">"list"</span>,</span></span>
<span class="r-in"><span>    as.df <span class="op">=</span> <span class="cn">TRUE</span>, use.cache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Tuomas Borman, Ben Allen, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

