<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MGnifyR: interface to MGnify database • MGnifyR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MGnifyR: interface to MGnify database">
<meta property="og:description" content="MGnifyR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MGnifyR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.15</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/MGnifyR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MGnifyR_long.html">MGnifyR: interface to MGnify database</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/EBI-Metagenomics/MGnifyR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MGnifyR: interface to MGnify database</h1>
            
            <h4 data-toc-skip class="date">2024-02-02</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/HEAD/vignettes/MGnifyR.Rmd" class="external-link"><code>vignettes/MGnifyR.Rmd</code></a></small>
      <div class="hidden name"><code>MGnifyR.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>MGnifyR</code> is a package designed to ease access to the
EBI’s <a href="https://www.ebi.ac.uk/metagenomics" class="external-link">MGnify</a> resource,
allowing searching and retrieval of multiple datasets for downstream
analysis.</p>
<p>The latest version of MGnifyR seamlessly integrates with the <a href="https://microbiome.github.io/" class="external-link">miaverse framework</a> providing
access to cutting-edge tools in microbiome down-stream analytics.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><code>MGnifyR</code> is currently hosted on GitHub, and can be
installed using via <code>devtools</code>. <code>MGnifyR</code> should
be built using the following snippet.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">MGnifyR</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-mgnifyr-package">Load <code>MGnifyR</code> package<a class="anchor" aria-label="anchor" href="#load-mgnifyr-package"></a>
</h2>
<p>Once installed, <code>MGnifyR</code> is made available in the usual
way.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/EBI-Metagenomics/MGnifyR" class="external-link">MGnifyR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: mia</span></span>
<span><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span><span class="co">#&gt;     tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span><span class="co">#&gt; Loading required package: SingleCellExperiment</span></span>
<span><span class="co">#&gt; Loading required package: TreeSummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: Biostrings</span></span>
<span><span class="co">#&gt; Loading required package: XVector</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biostrings'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     strsplit</span></span>
<span><span class="co">#&gt; Loading required package: MultiAssayExperiment</span></span>
<span><span class="co">#&gt; Loading required package: biomformat</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-a-client">Create a client<a class="anchor" aria-label="anchor" href="#create-a-client"></a>
</h2>
<p>All functions in <code>MGnifyR</code> make use of a
<code>MgnifyClient</code> object to keep track of the JSONAPI url, disk
cache location and user access tokens. Thus the first thing to do when
starting any analysis is to instantiate this object. The following
snippet creates this.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MgnifyClient.html">MgnifyClient</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">mg</span></span>
<span><span class="co">#&gt; An object of class "MgnifyClient"</span></span>
<span><span class="co">#&gt; Slot "databaseUrl":</span></span>
<span><span class="co">#&gt; [1] "https://www.ebi.ac.uk/metagenomics/api/v1"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "authTok":</span></span>
<span><span class="co">#&gt; [1] NA</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "useCache":</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "cacheDir":</span></span>
<span><span class="co">#&gt; [1] "/__w/MGnifyR/MGnifyR/vignettes/.MGnifyR_cache"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "showWarnings":</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "clearCache":</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "verbose":</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="functions-for-fetching-the-data">Functions for fetching the data<a class="anchor" aria-label="anchor" href="#functions-for-fetching-the-data"></a>
</h2>
<div class="section level3">
<h3 id="search-data">Search data<a class="anchor" aria-label="anchor" href="#search-data"></a>
</h3>
<p>Below, we fetch information on samples of drinking water.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fetch studies</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doQuery.html">doQuery</a></span><span class="op">(</span></span>
<span>    <span class="va">mg</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"samples"</span>,</span>
<span>    biome_name <span class="op">=</span> <span class="st">"root:Environmental:Aquatic:Freshwater:Drinking water"</span>,</span>
<span>    max.hits <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># For demonstrative purpose, take only few samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">595</span><span class="op">)</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="va">samples</span><span class="op">[</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span>, <span class="fl">5</span><span class="op">)</span>, <span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">samples</span></span>
<span><span class="co">#&gt;                  biosample   accession</span></span>
<span><span class="co">#&gt; ERS14399432 SAMEA112288606 ERS14399432</span></span>
<span><span class="co">#&gt; ERS14399426 SAMEA112288600 ERS14399426</span></span>
<span><span class="co">#&gt; ERS5222929    SAMEA7465225  ERS5222929</span></span>
<span><span class="co">#&gt; ERS5222919    SAMEA7465215  ERS5222919</span></span>
<span><span class="co">#&gt; ERS5222921    SAMEA7465217  ERS5222921</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                               sample-desc</span></span>
<span><span class="co">#&gt; ERS14399432                                                                                                                                                                                       biomass</span></span>
<span><span class="co">#&gt; ERS14399426                                                                                                                                                                                       biomass</span></span>
<span><span class="co">#&gt; ERS5222929             Raw water source [Surface water], Treatment [Ferric sulfate coagulation, clarification, sand filtration, ozonisation, activated carbon filtration], Disinfection [UV-light, NH2Cl]</span></span>
<span><span class="co">#&gt; ERS5222919  Raw water source [Artificial groundwater], Treatment [Aeration, lime stabilization, flocculation, clarification, addition of sulphuric acid, sand filtration], Disinfection [No disinfection]</span></span>
<span><span class="co">#&gt; ERS5222921                              Raw water source [Artificial groundwater], Treatment [Aeration, lime stabilization, flocculation, clarification, sand filtration], Disinfection [No disinfection]</span></span>
<span><span class="co">#&gt;             environment-biome                environment-feature</span></span>
<span><span class="co">#&gt; ERS14399432              &lt;NA&gt;                               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399426              &lt;NA&gt;                               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS5222929         Freshwater Drinking water distribution system</span></span>
<span><span class="co">#&gt; ERS5222919         Freshwater Drinking water distribution system</span></span>
<span><span class="co">#&gt; ERS5222921         Freshwater Drinking water distribution system</span></span>
<span><span class="co">#&gt;             environment-material     sample-name    sample-alias</span></span>
<span><span class="co">#&gt; ERS14399432                 &lt;NA&gt;         GREEN64         GREEN64</span></span>
<span><span class="co">#&gt; ERS14399426                 &lt;NA&gt;         GREEN58         GREEN58</span></span>
<span><span class="co">#&gt; ERS5222929                 water THS_D2_SW_CHM_R THS_D2_SW_CHM_R</span></span>
<span><span class="co">#&gt; ERS5222919                 water  THS_A1_GW_ND_R  THS_A1_GW_ND_R</span></span>
<span><span class="co">#&gt; ERS5222921                 water  THS_B1_GW_ND_D  THS_B1_GW_ND_D</span></span>
<span><span class="co">#&gt;                     last-update</span></span>
<span><span class="co">#&gt; ERS14399432 2023-04-23T00:43:15</span></span>
<span><span class="co">#&gt; ERS14399426 2023-04-23T02:07:52</span></span>
<span><span class="co">#&gt; ERS5222929  2020-11-10T18:16:23</span></span>
<span><span class="co">#&gt; ERS5222919  2020-11-10T18:02:30</span></span>
<span><span class="co">#&gt; ERS5222921  2021-06-02T14:21:12</span></span>
<span><span class="co">#&gt;                                                                    project name</span></span>
<span><span class="co">#&gt; ERS14399432                        ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; ERS14399426                        ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; ERS5222929  DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">#&gt; ERS5222919  DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">#&gt; ERS5222921  DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">#&gt;             geographic location (country and/or sea,region) collection date</span></span>
<span><span class="co">#&gt; ERS14399432                                           Spain      2022-09-14</span></span>
<span><span class="co">#&gt; ERS14399426                                           Spain      2022-09-14</span></span>
<span><span class="co">#&gt; ERS5222929                                          Finland      2015-09-01</span></span>
<span><span class="co">#&gt; ERS5222919                                          Finland      2015-08-03</span></span>
<span><span class="co">#&gt; ERS5222921                                          Finland      2015-08-03</span></span>
<span><span class="co">#&gt;             environment (biome)              environment (feature)</span></span>
<span><span class="co">#&gt; ERS14399432                &lt;NA&gt;                               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399426                &lt;NA&gt;                               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS5222929           Freshwater Drinking water distribution system</span></span>
<span><span class="co">#&gt; ERS5222919           Freshwater Drinking water distribution system</span></span>
<span><span class="co">#&gt; ERS5222921           Freshwater Drinking water distribution system</span></span>
<span><span class="co">#&gt;             environment (material) ENA checklist acc_type</span></span>
<span><span class="co">#&gt; ERS14399432                   &lt;NA&gt;     ERC000023  samples</span></span>
<span><span class="co">#&gt; ERS14399426                   &lt;NA&gt;     ERC000023  samples</span></span>
<span><span class="co">#&gt; ERS5222929                   water     ERC000025  samples</span></span>
<span><span class="co">#&gt; ERS5222919                   water     ERC000025  samples</span></span>
<span><span class="co">#&gt; ERS5222921                   water     ERC000025  samples</span></span>
<span><span class="co">#&gt;                                                                              biome</span></span>
<span><span class="co">#&gt; ERS14399432                   root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; ERS14399426                   root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; ERS5222929  root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">#&gt; ERS5222919  root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">#&gt; ERS5222921  root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">#&gt;                  studies    type collection-date latitude longitude</span></span>
<span><span class="co">#&gt; ERS14399432 MGYS00006211 samples      2022-09-14     &lt;NA&gt;      &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399426 MGYS00006211 samples      2022-09-14     &lt;NA&gt;      &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS5222929  MGYS00005650 samples      2015-09-01    61.92     25.75</span></span>
<span><span class="co">#&gt; ERS5222919  MGYS00005650 samples      2015-08-03    61.92     25.75</span></span>
<span><span class="co">#&gt; ERS5222921  MGYS00005650 samples      2015-08-03    61.92     25.75</span></span>
<span><span class="co">#&gt;             investigation type geographic location (longitude)</span></span>
<span><span class="co">#&gt; ERS14399432               &lt;NA&gt;                            &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399426               &lt;NA&gt;                            &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS5222929   metatranscriptome                           25.75</span></span>
<span><span class="co">#&gt; ERS5222919   metatranscriptome                           25.75</span></span>
<span><span class="co">#&gt; ERS5222921          metagenome                           25.75</span></span>
<span><span class="co">#&gt;                                       environmental package sequencing method</span></span>
<span><span class="co">#&gt; ERS14399432                                            &lt;NA&gt;              &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399426                                            &lt;NA&gt;              &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS5222929  miscellaneous natural or artificial environment          Illumina</span></span>
<span><span class="co">#&gt; ERS5222919  miscellaneous natural or artificial environment          Illumina</span></span>
<span><span class="co">#&gt; ERS5222921  miscellaneous natural or artificial environment          Illumina</span></span>
<span><span class="co">#&gt;             geographic location (latitude) analysis-completed geo-loc-name</span></span>
<span><span class="co">#&gt; ERS14399432                           &lt;NA&gt;               &lt;NA&gt;         &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399426                           &lt;NA&gt;               &lt;NA&gt;         &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS5222929                           61.92               &lt;NA&gt;         &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS5222919                           61.92               &lt;NA&gt;         &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS5222921                           61.92               &lt;NA&gt;         &lt;NA&gt;</span></span>
<span><span class="co">#&gt;             instrument model</span></span>
<span><span class="co">#&gt; ERS14399432             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399426             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS5222929              &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS5222919              &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS5222921              &lt;NA&gt;</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="find-relevent-analyses-accessions">Find relevent <strong>analyses</strong> accessions<a class="anchor" aria-label="anchor" href="#find-relevent-analyses-accessions"></a>
</h3>
<p>Now we want to find analysis accessions. Each sample might have
multiple analyses. Each analysis ID corresponds to a single run of a
particular pipeline on a single sample in a single study.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analyses_accessions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchAnalysis.html">searchAnalysis</a></span><span class="op">(</span><span class="va">mg</span>, <span class="st">"samples"</span>, <span class="va">samples</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">analyses_accessions</span><span class="op">)</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fetch-metadata">Fetch metadata<a class="anchor" aria-label="anchor" href="#fetch-metadata"></a>
</h3>
<p>We can now check the metadata to get hint of what kind of data we
have.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analyses_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMetadata.html">getMetadata</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">analyses_accessions</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">analyses_metadata</span><span class="op">)</span></span>
<span><span class="co">#&gt;              analysis_experiment-type analysis_pipeline-version</span></span>
<span><span class="co">#&gt; MGYA00643475                 assembly                       5.0</span></span>
<span><span class="co">#&gt; MGYA00643477                 assembly                       5.0</span></span>
<span><span class="co">#&gt; MGYA00575721       metatranscriptomic                       5.0</span></span>
<span><span class="co">#&gt; MGYA00575717       metatranscriptomic                       5.0</span></span>
<span><span class="co">#&gt; MGYA00575713              metagenomic                       5.0</span></span>
<span><span class="co">#&gt;              analysis_analysis-status analysis_accession analysis_is-private</span></span>
<span><span class="co">#&gt; MGYA00643475                completed       MGYA00643475                TRUE</span></span>
<span><span class="co">#&gt; MGYA00643477                completed       MGYA00643477                TRUE</span></span>
<span><span class="co">#&gt; MGYA00575721                completed       MGYA00575721               FALSE</span></span>
<span><span class="co">#&gt; MGYA00575717                completed       MGYA00575717               FALSE</span></span>
<span><span class="co">#&gt; MGYA00575713                completed       MGYA00575713               FALSE</span></span>
<span><span class="co">#&gt;              analysis_complete-time analysis_instrument-platform</span></span>
<span><span class="co">#&gt; MGYA00643475    2023-04-23T00:43:18                     ILLUMINA</span></span>
<span><span class="co">#&gt; MGYA00643477    2023-04-23T02:07:56                     ILLUMINA</span></span>
<span><span class="co">#&gt; MGYA00575721    2020-11-10T18:16:27                     ILLUMINA</span></span>
<span><span class="co">#&gt; MGYA00575717    2020-11-10T18:02:34                     ILLUMINA</span></span>
<span><span class="co">#&gt; MGYA00575713    2020-11-10T17:47:00                     ILLUMINA</span></span>
<span><span class="co">#&gt;              analysis_instrument-model analysis_Submitted nucleotide sequences</span></span>
<span><span class="co">#&gt; MGYA00643475     Illumina NovaSeq 6000                                  187536</span></span>
<span><span class="co">#&gt; MGYA00643477     Illumina NovaSeq 6000                                  167208</span></span>
<span><span class="co">#&gt; MGYA00575721       Illumina HiSeq 4000                                 1766611</span></span>
<span><span class="co">#&gt; MGYA00575717       Illumina HiSeq 4000                                 5454850</span></span>
<span><span class="co">#&gt; MGYA00575713       Illumina HiSeq 4000                                17338992</span></span>
<span><span class="co">#&gt;              analysis_Nucleotide sequences after format-specific filtering</span></span>
<span><span class="co">#&gt; MGYA00643475                                                        187536</span></span>
<span><span class="co">#&gt; MGYA00643477                                                        167208</span></span>
<span><span class="co">#&gt; MGYA00575721                                                        293112</span></span>
<span><span class="co">#&gt; MGYA00575717                                                       1159251</span></span>
<span><span class="co">#&gt; MGYA00575713                                                       2179562</span></span>
<span><span class="co">#&gt;              analysis_Nucleotide sequences after length filtering</span></span>
<span><span class="co">#&gt; MGYA00643475                                               187536</span></span>
<span><span class="co">#&gt; MGYA00643477                                               167208</span></span>
<span><span class="co">#&gt; MGYA00575721                                               293112</span></span>
<span><span class="co">#&gt; MGYA00575717                                              1159251</span></span>
<span><span class="co">#&gt; MGYA00575713                                              2179562</span></span>
<span><span class="co">#&gt;              analysis_Nucleotide sequences after undetermined bases filtering</span></span>
<span><span class="co">#&gt; MGYA00643475                                                           187536</span></span>
<span><span class="co">#&gt; MGYA00643477                                                           167208</span></span>
<span><span class="co">#&gt; MGYA00575721                                                           293112</span></span>
<span><span class="co">#&gt; MGYA00575717                                                          1159251</span></span>
<span><span class="co">#&gt; MGYA00575713                                                          2179562</span></span>
<span><span class="co">#&gt;              analysis_Reads with predicted CDS</span></span>
<span><span class="co">#&gt; MGYA00643475                            185060</span></span>
<span><span class="co">#&gt; MGYA00643477                            163559</span></span>
<span><span class="co">#&gt; MGYA00575721                             34145</span></span>
<span><span class="co">#&gt; MGYA00575717                             79238</span></span>
<span><span class="co">#&gt; MGYA00575713                           2132038</span></span>
<span><span class="co">#&gt;              analysis_Reads with predicted RNA</span></span>
<span><span class="co">#&gt; MGYA00643475                              3748</span></span>
<span><span class="co">#&gt; MGYA00643477                              2841</span></span>
<span><span class="co">#&gt; MGYA00575721                            254446</span></span>
<span><span class="co">#&gt; MGYA00575717                           1065067</span></span>
<span><span class="co">#&gt; MGYA00575713                              6062</span></span>
<span><span class="co">#&gt;              analysis_Reads with InterProScan match analysis_Predicted CDS</span></span>
<span><span class="co">#&gt; MGYA00643475                                 140771                 586640</span></span>
<span><span class="co">#&gt; MGYA00643477                                 127017                 464277</span></span>
<span><span class="co">#&gt; MGYA00575721                                   7922                  34824</span></span>
<span><span class="co">#&gt; MGYA00575717                                  14140                  80536</span></span>
<span><span class="co">#&gt; MGYA00575713                                 850215                2213838</span></span>
<span><span class="co">#&gt;              analysis_Predicted CDS with InterProScan match</span></span>
<span><span class="co">#&gt; MGYA00643475                                         368387</span></span>
<span><span class="co">#&gt; MGYA00643477                                         313613</span></span>
<span><span class="co">#&gt; MGYA00575721                                           7947</span></span>
<span><span class="co">#&gt; MGYA00575717                                          14203</span></span>
<span><span class="co">#&gt; MGYA00575713                                         853169</span></span>
<span><span class="co">#&gt;              analysis_Total InterProScan matches</span></span>
<span><span class="co">#&gt; MGYA00643475                             1261625</span></span>
<span><span class="co">#&gt; MGYA00643477                             1069719</span></span>
<span><span class="co">#&gt; MGYA00575721                               12971</span></span>
<span><span class="co">#&gt; MGYA00575717                               24971</span></span>
<span><span class="co">#&gt; MGYA00575713                             1513676</span></span>
<span><span class="co">#&gt;              analysis_Predicted SSU sequences analysis_Predicted LSU sequences</span></span>
<span><span class="co">#&gt; MGYA00643475                              192                              318</span></span>
<span><span class="co">#&gt; MGYA00643477                              171                              278</span></span>
<span><span class="co">#&gt; MGYA00575721                           118755                           143666</span></span>
<span><span class="co">#&gt; MGYA00575717                           573080                           504459</span></span>
<span><span class="co">#&gt; MGYA00575713                             1593                             2969</span></span>
<span><span class="co">#&gt;              analysis_acc_type study_attributes.accession</span></span>
<span><span class="co">#&gt; MGYA00643475     analysis-jobs               MGYS00006211</span></span>
<span><span class="co">#&gt; MGYA00643477     analysis-jobs               MGYS00006211</span></span>
<span><span class="co">#&gt; MGYA00575721     analysis-jobs               MGYS00005650</span></span>
<span><span class="co">#&gt; MGYA00575717     analysis-jobs               MGYS00005650</span></span>
<span><span class="co">#&gt; MGYA00575713     analysis-jobs               MGYS00005650</span></span>
<span><span class="co">#&gt;              study_attributes.bioproject study_attributes.samples-count</span></span>
<span><span class="co">#&gt; MGYA00643475                  PRJEB58755                             23</span></span>
<span><span class="co">#&gt; MGYA00643477                  PRJEB58755                             23</span></span>
<span><span class="co">#&gt; MGYA00575721                  PRJEB40814                             15</span></span>
<span><span class="co">#&gt; MGYA00575717                  PRJEB40814                             15</span></span>
<span><span class="co">#&gt; MGYA00575713                  PRJEB40814                             15</span></span>
<span><span class="co">#&gt;              study_attributes.is-private study_attributes.secondary-accession</span></span>
<span><span class="co">#&gt; MGYA00643475                       FALSE                            ERP143823</span></span>
<span><span class="co">#&gt; MGYA00643477                       FALSE                            ERP143823</span></span>
<span><span class="co">#&gt; MGYA00575721                       FALSE                            ERP124491</span></span>
<span><span class="co">#&gt; MGYA00575717                       FALSE                            ERP124491</span></span>
<span><span class="co">#&gt; MGYA00575713                       FALSE                            ERP124491</span></span>
<span><span class="co">#&gt;              study_attributes.centre-name</span></span>
<span><span class="co">#&gt; MGYA00643475  Leitat Technological Center</span></span>
<span><span class="co">#&gt; MGYA00643477  Leitat Technological Center</span></span>
<span><span class="co">#&gt; MGYA00575721                       US EPA</span></span>
<span><span class="co">#&gt; MGYA00575717                       US EPA</span></span>
<span><span class="co">#&gt; MGYA00575713                       US EPA</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          study_attributes.study-abstract</span></span>
<span><span class="co">#&gt; MGYA00643475                                                                                                                                                                                                                                                                                                Elimination of several antibiotics in water by bioelectrochemical cells. The main objective is study how the concentration of antibiotic resistant genes (ARG) changed depending on the voltage application.</span></span>
<span><span class="co">#&gt; MGYA00643477                                                                                                                                                                                                                                                                                                Elimination of several antibiotics in water by bioelectrochemical cells. The main objective is study how the concentration of antibiotic resistant genes (ARG) changed depending on the voltage application.</span></span>
<span><span class="co">#&gt; MGYA00575721 The water microbiome in the drinking water distribution systems (DWDSs) of five waterworks in Finland with different raw water sources and treatment processes was explored. The sampled DWDSs were from two waterworks AB with non-disinfected, recharged groundwater as source water and from three waterworks utilizing chlorinated water (two DWDSs of surface waterworks CD and one of ground waterworks E). The water microbiome was characterized by Illumina high-throughput sequencing technology.</span></span>
<span><span class="co">#&gt; MGYA00575717 The water microbiome in the drinking water distribution systems (DWDSs) of five waterworks in Finland with different raw water sources and treatment processes was explored. The sampled DWDSs were from two waterworks AB with non-disinfected, recharged groundwater as source water and from three waterworks utilizing chlorinated water (two DWDSs of surface waterworks CD and one of ground waterworks E). The water microbiome was characterized by Illumina high-throughput sequencing technology.</span></span>
<span><span class="co">#&gt; MGYA00575713 The water microbiome in the drinking water distribution systems (DWDSs) of five waterworks in Finland with different raw water sources and treatment processes was explored. The sampled DWDSs were from two waterworks AB with non-disinfected, recharged groundwater as source water and from three waterworks utilizing chlorinated water (two DWDSs of surface waterworks CD and one of ground waterworks E). The water microbiome was characterized by Illumina high-throughput sequencing technology.</span></span>
<span><span class="co">#&gt;                                                                                                                                            study_attributes.study-name</span></span>
<span><span class="co">#&gt; MGYA00643475                                                                                                              ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00643477                                                                                                              ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00575721 Metagenomic and metatranscriptomic analysis of the microbial community in drinking water distribution systems of ground and surface waterworks in Finland</span></span>
<span><span class="co">#&gt; MGYA00575717 Metagenomic and metatranscriptomic analysis of the microbial community in drinking water distribution systems of ground and surface waterworks in Finland</span></span>
<span><span class="co">#&gt; MGYA00575713 Metagenomic and metatranscriptomic analysis of the microbial community in drinking water distribution systems of ground and surface waterworks in Finland</span></span>
<span><span class="co">#&gt;              study_attributes.data-origination study_attributes.last-update</span></span>
<span><span class="co">#&gt; MGYA00643475                         SUBMITTED          2023-08-07T11:15:24</span></span>
<span><span class="co">#&gt; MGYA00643477                         SUBMITTED          2023-08-07T11:15:24</span></span>
<span><span class="co">#&gt; MGYA00575721                         SUBMITTED          2021-06-02T16:00:57</span></span>
<span><span class="co">#&gt; MGYA00575717                         SUBMITTED          2021-06-02T16:00:57</span></span>
<span><span class="co">#&gt; MGYA00575713                         SUBMITTED          2021-06-02T16:00:57</span></span>
<span><span class="co">#&gt;              study_accession study_acc_type sample_biosample sample_accession</span></span>
<span><span class="co">#&gt; MGYA00643475    MGYS00006211        studies   SAMEA112288606      ERS14399432</span></span>
<span><span class="co">#&gt; MGYA00643477    MGYS00006211        studies   SAMEA112288600      ERS14399426</span></span>
<span><span class="co">#&gt; MGYA00575721    MGYS00005650        studies     SAMEA7465225       ERS5222929</span></span>
<span><span class="co">#&gt; MGYA00575717    MGYS00005650        studies     SAMEA7465215       ERS5222919</span></span>
<span><span class="co">#&gt; MGYA00575713    MGYS00005650        studies     SAMEA7465217       ERS5222921</span></span>
<span><span class="co">#&gt;              sample_collection-date</span></span>
<span><span class="co">#&gt; MGYA00643475             2022-09-14</span></span>
<span><span class="co">#&gt; MGYA00643477             2022-09-14</span></span>
<span><span class="co">#&gt; MGYA00575721             2015-09-01</span></span>
<span><span class="co">#&gt; MGYA00575717             2015-08-03</span></span>
<span><span class="co">#&gt; MGYA00575713             2015-08-03</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                         sample_sample-desc</span></span>
<span><span class="co">#&gt; MGYA00643475                                                                                                                                                                                       biomass</span></span>
<span><span class="co">#&gt; MGYA00643477                                                                                                                                                                                       biomass</span></span>
<span><span class="co">#&gt; MGYA00575721            Raw water source [Surface water], Treatment [Ferric sulfate coagulation, clarification, sand filtration, ozonisation, activated carbon filtration], Disinfection [UV-light, NH2Cl]</span></span>
<span><span class="co">#&gt; MGYA00575717 Raw water source [Artificial groundwater], Treatment [Aeration, lime stabilization, flocculation, clarification, addition of sulphuric acid, sand filtration], Disinfection [No disinfection]</span></span>
<span><span class="co">#&gt; MGYA00575713                             Raw water source [Artificial groundwater], Treatment [Aeration, lime stabilization, flocculation, clarification, sand filtration], Disinfection [No disinfection]</span></span>
<span><span class="co">#&gt;              sample_sample-name sample_sample-alias  sample_last-update</span></span>
<span><span class="co">#&gt; MGYA00643475            GREEN64             GREEN64 2023-04-23T00:43:15</span></span>
<span><span class="co">#&gt; MGYA00643477            GREEN58             GREEN58 2023-04-23T02:07:52</span></span>
<span><span class="co">#&gt; MGYA00575721    THS_D2_SW_CHM_R     THS_D2_SW_CHM_R 2020-11-10T18:16:23</span></span>
<span><span class="co">#&gt; MGYA00575717     THS_A1_GW_ND_R      THS_A1_GW_ND_R 2020-11-10T18:02:30</span></span>
<span><span class="co">#&gt; MGYA00575713     THS_B1_GW_ND_D      THS_B1_GW_ND_D 2021-06-02T14:21:12</span></span>
<span><span class="co">#&gt;                                                              sample_project name</span></span>
<span><span class="co">#&gt; MGYA00643475                        ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00643477                        ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00575721 DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">#&gt; MGYA00575717 DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">#&gt; MGYA00575713 DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">#&gt;              sample_geographic location (country and/or sea,region)</span></span>
<span><span class="co">#&gt; MGYA00643475                                                  Spain</span></span>
<span><span class="co">#&gt; MGYA00643477                                                  Spain</span></span>
<span><span class="co">#&gt; MGYA00575721                                                Finland</span></span>
<span><span class="co">#&gt; MGYA00575717                                                Finland</span></span>
<span><span class="co">#&gt; MGYA00575713                                                Finland</span></span>
<span><span class="co">#&gt;              sample_collection date sample_ENA checklist sample_acc_type</span></span>
<span><span class="co">#&gt; MGYA00643475             2022-09-14            ERC000023         samples</span></span>
<span><span class="co">#&gt; MGYA00643477             2022-09-14            ERC000023         samples</span></span>
<span><span class="co">#&gt; MGYA00575721             2015-09-01            ERC000025         samples</span></span>
<span><span class="co">#&gt; MGYA00575717             2015-08-03            ERC000025         samples</span></span>
<span><span class="co">#&gt; MGYA00575713             2015-08-03            ERC000025         samples</span></span>
<span><span class="co">#&gt;              assembly_accession</span></span>
<span><span class="co">#&gt; MGYA00643475        ERZ16299693</span></span>
<span><span class="co">#&gt; MGYA00643477        ERZ16299677</span></span>
<span><span class="co">#&gt; MGYA00575721               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00575717               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00575713               &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                                                                        biome_string</span></span>
<span><span class="co">#&gt; MGYA00643475                   root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; MGYA00643477                   root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; MGYA00575721 root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">#&gt; MGYA00575717 root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">#&gt; MGYA00575713 root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">#&gt;              sample_latitude sample_longitude sample_environment-biome</span></span>
<span><span class="co">#&gt; MGYA00643475            &lt;NA&gt;             &lt;NA&gt;                     &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643477            &lt;NA&gt;             &lt;NA&gt;                     &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00575721           61.92            25.75               Freshwater</span></span>
<span><span class="co">#&gt; MGYA00575717           61.92            25.75               Freshwater</span></span>
<span><span class="co">#&gt; MGYA00575713           61.92            25.75               Freshwater</span></span>
<span><span class="co">#&gt;                      sample_environment-feature sample_environment-material</span></span>
<span><span class="co">#&gt; MGYA00643475                               &lt;NA&gt;                        &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643477                               &lt;NA&gt;                        &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00575721 Drinking water distribution system                       water</span></span>
<span><span class="co">#&gt; MGYA00575717 Drinking water distribution system                       water</span></span>
<span><span class="co">#&gt; MGYA00575713 Drinking water distribution system                       water</span></span>
<span><span class="co">#&gt;              sample_investigation type sample_geographic location (longitude)</span></span>
<span><span class="co">#&gt; MGYA00643475                      &lt;NA&gt;                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643477                      &lt;NA&gt;                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00575721         metatranscriptome                                  25.75</span></span>
<span><span class="co">#&gt; MGYA00575717         metatranscriptome                                  25.75</span></span>
<span><span class="co">#&gt; MGYA00575713                metagenome                                  25.75</span></span>
<span><span class="co">#&gt;              sample_environment (biome)       sample_environment (feature)</span></span>
<span><span class="co">#&gt; MGYA00643475                       &lt;NA&gt;                               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643477                       &lt;NA&gt;                               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00575721                 Freshwater Drinking water distribution system</span></span>
<span><span class="co">#&gt; MGYA00575717                 Freshwater Drinking water distribution system</span></span>
<span><span class="co">#&gt; MGYA00575713                 Freshwater Drinking water distribution system</span></span>
<span><span class="co">#&gt;              sample_environment (material)</span></span>
<span><span class="co">#&gt; MGYA00643475                          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643477                          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00575721                         water</span></span>
<span><span class="co">#&gt; MGYA00575717                         water</span></span>
<span><span class="co">#&gt; MGYA00575713                         water</span></span>
<span><span class="co">#&gt;                                 sample_environmental package</span></span>
<span><span class="co">#&gt; MGYA00643475                                            &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643477                                            &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00575721 miscellaneous natural or artificial environment</span></span>
<span><span class="co">#&gt; MGYA00575717 miscellaneous natural or artificial environment</span></span>
<span><span class="co">#&gt; MGYA00575713 miscellaneous natural or artificial environment</span></span>
<span><span class="co">#&gt;              sample_sequencing method sample_geographic location (latitude)</span></span>
<span><span class="co">#&gt; MGYA00643475                     &lt;NA&gt;                                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643477                     &lt;NA&gt;                                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00575721                 Illumina                                 61.92</span></span>
<span><span class="co">#&gt; MGYA00575717                 Illumina                                 61.92</span></span>
<span><span class="co">#&gt; MGYA00575713                 Illumina                                 61.92</span></span>
<span><span class="co">#&gt;              run_accession</span></span>
<span><span class="co">#&gt; MGYA00643475          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643477          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00575721    ERR4702562</span></span>
<span><span class="co">#&gt; MGYA00575717    ERR4702552</span></span>
<span><span class="co">#&gt; MGYA00575713    ERR4702554</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fetch-microbiome-data">Fetch microbiome data<a class="anchor" aria-label="anchor" href="#fetch-microbiome-data"></a>
</h3>
<p>After we have selected the data to fetch, we can use
<code><a href="../reference/getResult.html">getResult()</a></code></p>
<p>The output is <em><a href="https://bioconductor.org/packages/3.19/TreeSummarizedExperiment" class="external-link">TreeSummarizedExperiment</a></em>
(<code>TreeSE</code>) or <em><a href="https://bioconductor.org/packages/3.19/MultiAssayExperiment" class="external-link">MultiAssayExperiment</a></em>
(<code>MAE</code>) depending on the dataset. If the dataset includes
only taxonomic profiling data, the output is a single
<code>TreeSE</code>. If dataset includes also functional data, the
output is multiple <code>TreeSE</code> objects that are linked together
by utilizing <code>MAE</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getResult.html">getResult</a></span><span class="op">(</span><span class="va">mg</span>, accession <span class="op">=</span> <span class="va">analyses_accessions</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span></span>
<span><span class="co">#&gt; A MultiAssayExperiment object of 6 listed</span></span>
<span><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">#&gt;  Containing an ExperimentList class object of length 6:</span></span>
<span><span class="co">#&gt;  [1] microbiota: TreeSummarizedExperiment with 2029 rows and 5 columns</span></span>
<span><span class="co">#&gt;  [2] go-slim: TreeSummarizedExperiment with 116 rows and 5 columns</span></span>
<span><span class="co">#&gt;  [3] go-terms: TreeSummarizedExperiment with 3133 rows and 5 columns</span></span>
<span><span class="co">#&gt;  [4] interpro-identifiers: TreeSummarizedExperiment with 18223 rows and 5 columns</span></span>
<span><span class="co">#&gt;  [5] taxonomy: TreeSummarizedExperiment with 3112 rows and 5 columns</span></span>
<span><span class="co">#&gt;  [6] taxonomy-lsu: TreeSummarizedExperiment with 3378 rows and 5 columns</span></span>
<span><span class="co">#&gt; Functionality:</span></span>
<span><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code></pre></div>
<p>You can get access to individual <code>TreeSE</code> object in
<code>MAE</code> by specifying index or name.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; class: TreeSummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 2029 5 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(2029): 200154 3353 ... 5820 100053</span></span>
<span><span class="co">#&gt; rowData names(8): Kingdom Phylum ... Species taxonomy</span></span>
<span><span class="co">#&gt; colnames(5): MGYA00575713 MGYA00575717 MGYA00575721 MGYA00643475</span></span>
<span><span class="co">#&gt;   MGYA00643477</span></span>
<span><span class="co">#&gt; colData names(61): analysis_experiment.type analysis_pipeline.version</span></span>
<span><span class="co">#&gt;   ... sample_geographic.location..latitude. run_accession</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; rowLinks: NULL</span></span>
<span><span class="co">#&gt; rowTree: NULL</span></span>
<span><span class="co">#&gt; colLinks: NULL</span></span>
<span><span class="co">#&gt; colTree: NULL</span></span></code></pre></div>
<p><code>TreeSE</code> object is uniquely positioned to support
<code>SummarizedExperiment</code>-based microbiome data manipulation and
visualization. Moreover, it enables access to <code>miaverse</code>
tools. For example, we can estimate diversity of samples.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/estimateDiversity.html" class="external-link">estimateDiversity</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, index <span class="op">=</span> <span class="st">"shannon"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: scuttle</span></span>
<span><span class="co">#&gt; Loading required package: ggplot2</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"shannon"</span>, x <span class="op">=</span> <span class="st">"sample_sample.desc"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Groups with fewer than two data points have been dropped.</span></span>
<span><span class="co">#&gt; Groups with fewer than two data points have been dropped.</span></span>
<span><span class="co">#&gt; Groups with fewer than two data points have been dropped.</span></span></code></pre></div>
<p><img src="MGnifyR_files/figure-html/calculate_diversity-1.png" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Agglomerate data</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExps</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/splitByRanks.html" class="external-link">splitByRanks</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">miaViz</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: ggraph</span></span>
<span></span>
<span><span class="co"># Plot top taxa</span></span>
<span><span class="va">top_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/summaries.html" class="external-link">getTopFeatures</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"Phylum"</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/miaViz/man/plotAbundance.html" class="external-link">plotAbundance</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"Phylum"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>, <span class="op">]</span>, rank <span class="op">=</span> <span class="st">"Phylum"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MGnifyR_files/figure-html/plot_abundance-1.png" width="700"></p>
<p>We can perform principal component analysis to microbial profiling
data by utilizing miaverse tools.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Apply relative transformation</span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span><span class="co"># Perform PCoA</span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html" class="external-link">runMDS</a></span><span class="op">(</span></span>
<span>    <span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>    FUN <span class="op">=</span> <span class="fu">vegan</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></span>, method <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span></span>
<span><span class="co"># Plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span></span>
<span>    <span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"MDS"</span>, colour_by <span class="op">=</span> <span class="st">"sample_sample.desc"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MGnifyR_files/figure-html/pcoa-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="fetch-sequence-files">Fetch sequence files<a class="anchor" aria-label="anchor" href="#fetch-sequence-files"></a>
</h3>
<p>Finally, we can use <code><a href="../reference/getFile.html">searchFile()</a></code> and
<code><a href="../reference/getFile.html">getFile()</a></code> to retrieve other MGnify pipeline outputs such as
merged sequence reads, assembled contigs, and details of the functional
analyses.</p>
<p>With <code><a href="../reference/getFile.html">searchFile()</a></code>, we can search files from the
database.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find list of available downloads, and filter for </span></span>
<span><span class="va">dl_urls</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">searchFile</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">analyses_accessions</span>, type <span class="op">=</span> <span class="st">"analyses"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">target_urls</span> <span class="op">&lt;-</span> <span class="va">dl_urls</span><span class="op">[</span></span>
<span>    <span class="va">dl_urls</span><span class="op">$</span><span class="va">attributes.description.label</span> <span class="op">==</span> <span class="st">"Predicted alpha tmRNA"</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">target_urls</span><span class="op">)</span></span>
<span><span class="co">#&gt;         type                                       id</span></span>
<span><span class="co">#&gt; 24  analyses ERZ16299693_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 74  analyses ERZ16299677_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 121 analyses  ERR4702562_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 174 analyses  ERR4702554_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt;                             attributes.alias attributes.file.format.name</span></span>
<span><span class="co">#&gt; 24  ERZ16299693_alpha_tmRNA.RF01849.fasta.gz                       FASTA</span></span>
<span><span class="co">#&gt; 74  ERZ16299677_alpha_tmRNA.RF01849.fasta.gz                       FASTA</span></span>
<span><span class="co">#&gt; 121  ERR4702562_alpha_tmRNA.RF01849.fasta.gz                       FASTA</span></span>
<span><span class="co">#&gt; 174  ERR4702554_alpha_tmRNA.RF01849.fasta.gz                       FASTA</span></span>
<span><span class="co">#&gt;     attributes.file.format.extension attributes.file.format.compression</span></span>
<span><span class="co">#&gt; 24                             fasta                               TRUE</span></span>
<span><span class="co">#&gt; 74                             fasta                               TRUE</span></span>
<span><span class="co">#&gt; 121                            fasta                               TRUE</span></span>
<span><span class="co">#&gt; 174                            fasta                               TRUE</span></span>
<span><span class="co">#&gt;     attributes.description.label</span></span>
<span><span class="co">#&gt; 24         Predicted alpha tmRNA</span></span>
<span><span class="co">#&gt; 74         Predicted alpha tmRNA</span></span>
<span><span class="co">#&gt; 121        Predicted alpha tmRNA</span></span>
<span><span class="co">#&gt; 174        Predicted alpha tmRNA</span></span>
<span><span class="co">#&gt;                                 attributes.description.description</span></span>
<span><span class="co">#&gt; 24  Predicted Alphaproteobacteria transfer-messenger RNA (RF01849)</span></span>
<span><span class="co">#&gt; 74  Predicted Alphaproteobacteria transfer-messenger RNA (RF01849)</span></span>
<span><span class="co">#&gt; 121 Predicted Alphaproteobacteria transfer-messenger RNA (RF01849)</span></span>
<span><span class="co">#&gt; 174 Predicted Alphaproteobacteria transfer-messenger RNA (RF01849)</span></span>
<span><span class="co">#&gt;     attributes.group.type attributes.file.checksum.checksum</span></span>
<span><span class="co">#&gt; 24        non-coding RNAs                                  </span></span>
<span><span class="co">#&gt; 74        non-coding RNAs                                  </span></span>
<span><span class="co">#&gt; 121       non-coding RNAs                                  </span></span>
<span><span class="co">#&gt; 174       non-coding RNAs                                  </span></span>
<span><span class="co">#&gt;     attributes.file.checksum.checksum.algorithm</span></span>
<span><span class="co">#&gt; 24                                             </span></span>
<span><span class="co">#&gt; 74                                             </span></span>
<span><span class="co">#&gt; 121                                            </span></span>
<span><span class="co">#&gt; 174                                            </span></span>
<span><span class="co">#&gt;     relationships.pipeline.data.type relationships.pipeline.data.id</span></span>
<span><span class="co">#&gt; 24                         pipelines                            5.0</span></span>
<span><span class="co">#&gt; 74                         pipelines                            5.0</span></span>
<span><span class="co">#&gt; 121                        pipelines                            5.0</span></span>
<span><span class="co">#&gt; 174                        pipelines                            5.0</span></span>
<span><span class="co">#&gt;                                          relationships.pipeline.related</span></span>
<span><span class="co">#&gt; 24  https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/5.0?format=json</span></span>
<span><span class="co">#&gt; 74  https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/5.0?format=json</span></span>
<span><span class="co">#&gt; 121 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/5.0?format=json</span></span>
<span><span class="co">#&gt; 174 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/5.0?format=json</span></span>
<span><span class="co">#&gt;                                                                                                      download_url</span></span>
<span><span class="co">#&gt; 24  https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00643475/file/ERZ16299693_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 74  https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00643477/file/ERZ16299677_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 121  https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575721/file/ERR4702562_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 174  https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00575713/file/ERR4702554_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt;        accession</span></span>
<span><span class="co">#&gt; 24  MGYA00643475</span></span>
<span><span class="co">#&gt; 74  MGYA00643477</span></span>
<span><span class="co">#&gt; 121 MGYA00575721</span></span>
<span><span class="co">#&gt; 174 MGYA00575713</span></span></code></pre></div>
<p>Finally, we can download the files with <code><a href="../reference/getFile.html">getFile()</a></code>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Just select a single file from the target_urls list for demonstration.</span></span>
<span><span class="va">file_url</span> <span class="op">&lt;-</span> <span class="va">target_urls</span><span class="op">$</span><span class="va">download_url</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">cached_location</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">getFile</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">file_url</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Where are the files?</span></span>
<span><span class="va">cached_location</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmp5cE7Yz/filebb468bb04c3"</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R Under development (unstable) (2024-01-23 r85822)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] miaViz_1.11.0                   ggraph_2.1.0                   </span></span>
<span><span class="co">#&gt;  [3] scater_1.31.2                   ggplot2_3.4.4                  </span></span>
<span><span class="co">#&gt;  [5] scuttle_1.13.0                  MGnifyR_0.99.15                </span></span>
<span><span class="co">#&gt;  [7] biomformat_1.31.0               mia_1.11.1                     </span></span>
<span><span class="co">#&gt;  [9] MultiAssayExperiment_1.29.0     TreeSummarizedExperiment_2.11.0</span></span>
<span><span class="co">#&gt; [11] Biostrings_2.71.2               XVector_0.43.1                 </span></span>
<span><span class="co">#&gt; [13] SingleCellExperiment_1.25.0     SummarizedExperiment_1.33.3    </span></span>
<span><span class="co">#&gt; [15] Biobase_2.63.0                  GenomicRanges_1.55.2           </span></span>
<span><span class="co">#&gt; [17] GenomeInfoDb_1.39.5             IRanges_2.37.1                 </span></span>
<span><span class="co">#&gt; [19] S4Vectors_0.41.3                BiocGenerics_0.49.1            </span></span>
<span><span class="co">#&gt; [21] MatrixGenerics_1.15.0           matrixStats_1.2.0              </span></span>
<span><span class="co">#&gt; [23] knitr_1.45                      BiocStyle_2.31.0               </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] jsonlite_1.8.8              magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [3] ggbeeswarm_0.7.2            farver_2.1.1               </span></span>
<span><span class="co">#&gt;   [5] rmarkdown_2.25              fs_1.6.3                   </span></span>
<span><span class="co">#&gt;   [7] zlibbioc_1.49.0             ragg_1.2.7                 </span></span>
<span><span class="co">#&gt;   [9] vctrs_0.6.5                 memoise_2.0.1              </span></span>
<span><span class="co">#&gt;  [11] DelayedMatrixStats_1.25.1   RCurl_1.98-1.14            </span></span>
<span><span class="co">#&gt;  [13] ggtree_3.11.0               htmltools_0.5.7            </span></span>
<span><span class="co">#&gt;  [15] S4Arrays_1.3.3              BiocBaseUtils_1.5.0        </span></span>
<span><span class="co">#&gt;  [17] BiocNeighbors_1.21.2        Rhdf5lib_1.25.1            </span></span>
<span><span class="co">#&gt;  [19] gridGraphics_0.5-1          SparseArray_1.3.3          </span></span>
<span><span class="co">#&gt;  [21] rhdf5_2.47.2                sass_0.4.8                 </span></span>
<span><span class="co">#&gt;  [23] bslib_0.6.1                 desc_1.4.3                 </span></span>
<span><span class="co">#&gt;  [25] plyr_1.8.9                  DECIPHER_2.31.1            </span></span>
<span><span class="co">#&gt;  [27] cachem_1.0.8                igraph_2.0.1.1             </span></span>
<span><span class="co">#&gt;  [29] lifecycle_1.0.4             pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [31] rsvd_1.0.5                  Matrix_1.6-5               </span></span>
<span><span class="co">#&gt;  [33] R6_2.5.1                    fastmap_1.1.1              </span></span>
<span><span class="co">#&gt;  [35] GenomeInfoDbData_1.2.11     aplot_0.2.2                </span></span>
<span><span class="co">#&gt;  [37] digest_0.6.34               ggnewscale_0.4.9           </span></span>
<span><span class="co">#&gt;  [39] colorspace_2.1-0            patchwork_1.2.0            </span></span>
<span><span class="co">#&gt;  [41] irlba_2.3.5.1               textshaping_0.3.7          </span></span>
<span><span class="co">#&gt;  [43] RSQLite_2.3.5               vegan_2.6-4                </span></span>
<span><span class="co">#&gt;  [45] beachmat_2.19.1             labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [47] fansi_1.0.6                 urltools_1.7.3             </span></span>
<span><span class="co">#&gt;  [49] polyclip_1.10-6             httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [51] abind_1.4-5                 mgcv_1.9-1                 </span></span>
<span><span class="co">#&gt;  [53] compiler_4.4.0              withr_3.0.0                </span></span>
<span><span class="co">#&gt;  [55] bit64_4.0.5                 BiocParallel_1.37.0        </span></span>
<span><span class="co">#&gt;  [57] viridis_0.6.5               DBI_1.2.1                  </span></span>
<span><span class="co">#&gt;  [59] highr_0.10                  ggforce_0.4.1              </span></span>
<span><span class="co">#&gt;  [61] MASS_7.3-60.2               DelayedArray_0.29.1        </span></span>
<span><span class="co">#&gt;  [63] bluster_1.13.0              permute_0.9-7              </span></span>
<span><span class="co">#&gt;  [65] tools_4.4.0                 vipor_0.4.7                </span></span>
<span><span class="co">#&gt;  [67] beeswarm_0.4.0              ape_5.7-1                  </span></span>
<span><span class="co">#&gt;  [69] glue_1.7.0                  nlme_3.1-164               </span></span>
<span><span class="co">#&gt;  [71] rhdf5filters_1.15.1         grid_4.4.0                 </span></span>
<span><span class="co">#&gt;  [73] cluster_2.1.6               reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [75] generics_0.1.3              gtable_0.3.4               </span></span>
<span><span class="co">#&gt;  [77] tidyr_1.3.1                 tidygraph_1.3.1            </span></span>
<span><span class="co">#&gt;  [79] BiocSingular_1.19.0         ScaledMatrix_1.11.0        </span></span>
<span><span class="co">#&gt;  [81] utf8_1.2.4                  ggrepel_0.9.5              </span></span>
<span><span class="co">#&gt;  [83] pillar_1.9.0                stringr_1.5.1              </span></span>
<span><span class="co">#&gt;  [85] yulab.utils_0.1.4           splines_4.4.0              </span></span>
<span><span class="co">#&gt;  [87] tweenr_2.0.2                dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [89] treeio_1.27.0               lattice_0.22-5             </span></span>
<span><span class="co">#&gt;  [91] bit_4.0.5                   tidyselect_1.2.0           </span></span>
<span><span class="co">#&gt;  [93] DirichletMultinomial_1.45.0 gridExtra_2.3              </span></span>
<span><span class="co">#&gt;  [95] bookdown_0.37               xfun_0.41                  </span></span>
<span><span class="co">#&gt;  [97] graphlayouts_1.1.0          stringi_1.8.3              </span></span>
<span><span class="co">#&gt;  [99] ggfun_0.1.4                 lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt; [101] yaml_2.3.8                  evaluate_0.23              </span></span>
<span><span class="co">#&gt; [103] codetools_0.2-19            tibble_3.2.1               </span></span>
<span><span class="co">#&gt; [105] BiocManager_1.30.22         ggplotify_0.1.2            </span></span>
<span><span class="co">#&gt; [107] cli_3.6.2                   systemfonts_1.0.5          </span></span>
<span><span class="co">#&gt; [109] munsell_0.5.0               jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [111] Rcpp_1.0.12                 triebeard_0.4.1            </span></span>
<span><span class="co">#&gt; [113] parallel_4.4.0              pkgdown_2.0.7              </span></span>
<span><span class="co">#&gt; [115] blob_1.2.4                  sparseMatrixStats_1.15.0   </span></span>
<span><span class="co">#&gt; [117] bitops_1.0-7                decontam_1.23.0            </span></span>
<span><span class="co">#&gt; [119] viridisLite_0.4.2           tidytree_0.4.6             </span></span>
<span><span class="co">#&gt; [121] scales_1.3.0                purrr_1.0.2                </span></span>
<span><span class="co">#&gt; [123] crayon_1.5.2                rlang_1.1.3</span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Tuomas Borman, Ben Allen, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
