<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MGnifyR">
<title>MGnifyR: interface to MGnify database • MGnifyR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="MGnifyR: interface to MGnify database">
<meta property="og:description" content="MGnifyR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MGnifyR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.22</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/MGnifyR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/MGnifyR_long.html">MGnifyR: interface to MGnify database</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/EBI-Metagenomics/MGnifyR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>MGnifyR: interface to MGnify database</h1>
            
            <h4 data-toc-skip class="date">2024-02-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/HEAD/vignettes/MGnifyR.Rmd" class="external-link"><code>vignettes/MGnifyR.Rmd</code></a></small>
      <div class="d-none name"><code>MGnifyR.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>MGnifyR</code> is a package designed to ease access to the
EBI’s <a href="https://www.ebi.ac.uk/metagenomics" class="external-link">MGnify</a> resource,
allowing searching and retrieval of multiple datasets for downstream
analysis.</p>
<p>The latest version of MGnifyR seamlessly integrates with the <a href="https://microbiome.github.io/" class="external-link">miaverse framework</a> providing
access to cutting-edge tools in microbiome down-stream analytics.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><code>MGnifyR</code> is hosted on Bioconductor, and can be installed
using via <code>BiocManager</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">MGnifyR</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-mgnifyr-package">Load <code>MGnifyR</code> package<a class="anchor" aria-label="anchor" href="#load-mgnifyr-package"></a>
</h2>
<p>Once installed, <code>MGnifyR</code> is made available in the usual
way.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/EBI-Metagenomics/MGnifyR" class="external-link">MGnifyR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: mia</span></span>
<span><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span><span class="co">#&gt;     tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span><span class="co">#&gt; Loading required package: SingleCellExperiment</span></span>
<span><span class="co">#&gt; Loading required package: TreeSummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: Biostrings</span></span>
<span><span class="co">#&gt; Loading required package: XVector</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biostrings'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     strsplit</span></span>
<span><span class="co">#&gt; Loading required package: MultiAssayExperiment</span></span>
<span><span class="co">#&gt; Loading required package: biomformat</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-a-client">Create a client<a class="anchor" aria-label="anchor" href="#create-a-client"></a>
</h2>
<p>All functions in <code>MGnifyR</code> make use of a
<code>MgnifyClient</code> object to keep track of the JSONAPI url, disk
cache location and user access tokens. Thus the first thing to do when
starting any analysis is to instantiate this object. The following
snippet creates this.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MgnifyClient.html">MgnifyClient</a></span><span class="op">(</span>useCache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">mg</span></span>
<span><span class="co">#&gt; An object of class "MgnifyClient"</span></span>
<span><span class="co">#&gt; Slot "databaseUrl":</span></span>
<span><span class="co">#&gt; [1] "https://www.ebi.ac.uk/metagenomics/api/v1"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "authTok":</span></span>
<span><span class="co">#&gt; [1] NA</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "useCache":</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "cacheDir":</span></span>
<span><span class="co">#&gt; [1] "/__w/MGnifyR/MGnifyR/vignettes/.MGnifyR_cache"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "showWarnings":</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "clearCache":</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "verbose":</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="functions-for-fetching-the-data">Functions for fetching the data<a class="anchor" aria-label="anchor" href="#functions-for-fetching-the-data"></a>
</h2>
<div class="section level3">
<h3 id="search-data">Search data<a class="anchor" aria-label="anchor" href="#search-data"></a>
</h3>
<p><code><a href="../reference/doQuery.html">doQuery()</a></code> function can be utilized to search results
such as samples and studies from MGnify database. Below, we fetch
information drinking water samples.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fetch studies</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doQuery.html">doQuery</a></span><span class="op">(</span></span>
<span>    <span class="va">mg</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"samples"</span>,</span>
<span>    biome_name <span class="op">=</span> <span class="st">"root:Environmental:Aquatic:Freshwater:Drinking water"</span>,</span>
<span>    max.hits <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p>The result is a table containing accession IDs and description – in
this case – on samples.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span></span>
<span><span class="co">#&gt;                  biosample   accession sample-desc</span></span>
<span><span class="co">#&gt; ERS14399436 SAMEA112288610 ERS14399436     biofilm</span></span>
<span><span class="co">#&gt; ERS14399429 SAMEA112288603 ERS14399429     biofilm</span></span>
<span><span class="co">#&gt; ERS14399431 SAMEA112288605 ERS14399431     biomass</span></span>
<span><span class="co">#&gt; ERS14399428 SAMEA112288602 ERS14399428     biofilm</span></span>
<span><span class="co">#&gt; ERS14399416 SAMEA112288590 ERS14399416     biofilm</span></span>
<span><span class="co">#&gt; ERS14399423 SAMEA112288597 ERS14399423     biofilm</span></span>
<span><span class="co">#&gt;                                  environment-biome</span></span>
<span><span class="co">#&gt; ERS14399436 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; ERS14399429 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; ERS14399431                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399428                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399416                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399423                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                                environment-feature</span></span>
<span><span class="co">#&gt; ERS14399436 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; ERS14399429 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; ERS14399431                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399428                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399416                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399423                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                         environment-material sample-name sample-alias</span></span>
<span><span class="co">#&gt; ERS14399436 Biofilm material (ENVO:01000156)     GREEN68      GREEN68</span></span>
<span><span class="co">#&gt; ERS14399429 Biofilm material (ENVO:01000156)     GREEN61      GREEN61</span></span>
<span><span class="co">#&gt; ERS14399431                             &lt;NA&gt;     GREEN63      GREEN63</span></span>
<span><span class="co">#&gt; ERS14399428                             &lt;NA&gt;     GREEN60      GREEN60</span></span>
<span><span class="co">#&gt; ERS14399416                             &lt;NA&gt;     GREEN48      GREEN48</span></span>
<span><span class="co">#&gt; ERS14399423                             &lt;NA&gt;     GREEN55      GREEN55</span></span>
<span><span class="co">#&gt;                     last-update                                 project name</span></span>
<span><span class="co">#&gt; ERS14399436 2023-08-07T11:15:24 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; ERS14399429 2023-08-07T10:52:31 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; ERS14399431 2023-04-23T08:53:19 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; ERS14399428 2023-04-23T07:57:35 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; ERS14399416 2023-04-23T06:14:08 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; ERS14399423 2023-04-23T05:00:17 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt;             geographic location (country and/or sea,region) collection date</span></span>
<span><span class="co">#&gt; ERS14399436                                           Spain      2022-09-14</span></span>
<span><span class="co">#&gt; ERS14399429                                           Spain      2022-09-14</span></span>
<span><span class="co">#&gt; ERS14399431                                           Spain      2022-09-14</span></span>
<span><span class="co">#&gt; ERS14399428                                           Spain      2022-09-14</span></span>
<span><span class="co">#&gt; ERS14399416                                           Spain      2022-09-14</span></span>
<span><span class="co">#&gt; ERS14399423                                           Spain      2022-09-14</span></span>
<span><span class="co">#&gt;                                environment (biome)</span></span>
<span><span class="co">#&gt; ERS14399436 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; ERS14399429 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; ERS14399431                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399428                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399416                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399423                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                              environment (feature)</span></span>
<span><span class="co">#&gt; ERS14399436 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; ERS14399429 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; ERS14399431                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399428                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399416                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399423                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                       environment (material) ENA checklist acc_type</span></span>
<span><span class="co">#&gt; ERS14399436 Biofilm material (ENVO:01000156)     ERC000023  samples</span></span>
<span><span class="co">#&gt; ERS14399429 Biofilm material (ENVO:01000156)     ERC000023  samples</span></span>
<span><span class="co">#&gt; ERS14399431                             &lt;NA&gt;     ERC000023  samples</span></span>
<span><span class="co">#&gt; ERS14399428                             &lt;NA&gt;     ERC000023  samples</span></span>
<span><span class="co">#&gt; ERS14399416                             &lt;NA&gt;     ERC000023  samples</span></span>
<span><span class="co">#&gt; ERS14399423                             &lt;NA&gt;     ERC000023  samples</span></span>
<span><span class="co">#&gt;                  studies                                                biome</span></span>
<span><span class="co">#&gt; ERS14399436 MGYS00006211 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; ERS14399429 MGYS00006211 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; ERS14399431 MGYS00006211 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; ERS14399428 MGYS00006211 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; ERS14399416 MGYS00006211 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; ERS14399423 MGYS00006211 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt;                type collection-date latitude longitude investigation type</span></span>
<span><span class="co">#&gt; ERS14399436 samples            &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399429 samples            &lt;NA&gt;     &lt;NA&gt;      &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399431 samples      2022-09-14     &lt;NA&gt;      &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399428 samples      2022-09-14     &lt;NA&gt;      &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399416 samples      2022-09-14     &lt;NA&gt;      &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399423 samples      2022-09-14     &lt;NA&gt;      &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt;             geographic location (longitude) environmental package</span></span>
<span><span class="co">#&gt; ERS14399436                            &lt;NA&gt;                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399429                            &lt;NA&gt;                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399431                            &lt;NA&gt;                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399428                            &lt;NA&gt;                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399416                            &lt;NA&gt;                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399423                            &lt;NA&gt;                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt;             sequencing method geographic location (latitude) analysis-completed</span></span>
<span><span class="co">#&gt; ERS14399436              &lt;NA&gt;                           &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399429              &lt;NA&gt;                           &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399431              &lt;NA&gt;                           &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399428              &lt;NA&gt;                           &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399416              &lt;NA&gt;                           &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399423              &lt;NA&gt;                           &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt;             geo-loc-name instrument model</span></span>
<span><span class="co">#&gt; ERS14399436         &lt;NA&gt;             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399429         &lt;NA&gt;             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399431         &lt;NA&gt;             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399428         &lt;NA&gt;             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399416         &lt;NA&gt;             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; ERS14399423         &lt;NA&gt;             &lt;NA&gt;</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="find-relevent-analyses-accessions">Find relevent <strong>analyses</strong> accessions<a class="anchor" aria-label="anchor" href="#find-relevent-analyses-accessions"></a>
</h3>
<p>Now we want to find analysis accessions. Each sample might have
multiple analyses. Each analysis ID corresponds to a single run of a
particular pipeline on a single sample in a single study.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analyses_accessions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchAnalysis.html">searchAnalysis</a></span><span class="op">(</span><span class="va">mg</span>, <span class="st">"samples"</span>, <span class="va">samples</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span></span></code></pre></div>
<p>By running the <code><a href="../reference/searchAnalysis.html">searchAnalysis()</a></code> function, we get
analysis IDs of samples that we fed as an input.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">analyses_accessions</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "MGYA00652201" "MGYA00652185" "MGYA00643487" "MGYA00643486" "MGYA00643485"</span></span>
<span><span class="co">#&gt; [6] "MGYA00643484"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fetch-metadata">Fetch metadata<a class="anchor" aria-label="anchor" href="#fetch-metadata"></a>
</h3>
<p>We can now check the metadata to get hint of what kind of data we
have. We use <code><a href="../reference/getMetadata.html">getMetadata()</a></code> function to fetch data based on
analysis IDs.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analyses_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMetadata.html">getMetadata</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">analyses_accessions</span><span class="op">)</span></span></code></pre></div>
<p>Metadata includes for example information on how analysis was
conducted and what kind of samples were analyzed.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">analyses_metadata</span><span class="op">)</span></span>
<span><span class="co">#&gt;              analysis_analysis-status analysis_pipeline-version</span></span>
<span><span class="co">#&gt; MGYA00652201                completed                       5.0</span></span>
<span><span class="co">#&gt; MGYA00652185                completed                       5.0</span></span>
<span><span class="co">#&gt; MGYA00643487                completed                       5.0</span></span>
<span><span class="co">#&gt; MGYA00643486                completed                       5.0</span></span>
<span><span class="co">#&gt; MGYA00643485                completed                       5.0</span></span>
<span><span class="co">#&gt; MGYA00643484                completed                       5.0</span></span>
<span><span class="co">#&gt;              analysis_experiment-type analysis_accession analysis_is-private</span></span>
<span><span class="co">#&gt; MGYA00652201                 assembly       MGYA00652201                TRUE</span></span>
<span><span class="co">#&gt; MGYA00652185                 assembly       MGYA00652185                TRUE</span></span>
<span><span class="co">#&gt; MGYA00643487                 assembly       MGYA00643487                TRUE</span></span>
<span><span class="co">#&gt; MGYA00643486                 assembly       MGYA00643486                TRUE</span></span>
<span><span class="co">#&gt; MGYA00643485                 assembly       MGYA00643485                TRUE</span></span>
<span><span class="co">#&gt; MGYA00643484                 assembly       MGYA00643484                TRUE</span></span>
<span><span class="co">#&gt;              analysis_complete-time analysis_instrument-platform</span></span>
<span><span class="co">#&gt; MGYA00652201    2023-08-07T11:15:25                     ILLUMINA</span></span>
<span><span class="co">#&gt; MGYA00652185    2023-08-07T10:52:31                     ILLUMINA</span></span>
<span><span class="co">#&gt; MGYA00643487    2023-04-23T08:53:23                     ILLUMINA</span></span>
<span><span class="co">#&gt; MGYA00643486    2023-04-23T07:57:38                     ILLUMINA</span></span>
<span><span class="co">#&gt; MGYA00643485    2023-04-23T06:14:11                     ILLUMINA</span></span>
<span><span class="co">#&gt; MGYA00643484    2023-04-23T05:00:21                     ILLUMINA</span></span>
<span><span class="co">#&gt;              analysis_instrument-model analysis_Submitted nucleotide sequences</span></span>
<span><span class="co">#&gt; MGYA00652201     Illumina NovaSeq 6000                                  223726</span></span>
<span><span class="co">#&gt; MGYA00652185     Illumina NovaSeq 6000                                  292409</span></span>
<span><span class="co">#&gt; MGYA00643487     Illumina NovaSeq 6000                                  162292</span></span>
<span><span class="co">#&gt; MGYA00643486     Illumina NovaSeq 6000                                  233327</span></span>
<span><span class="co">#&gt; MGYA00643485     Illumina NovaSeq 6000                                  318625</span></span>
<span><span class="co">#&gt; MGYA00643484     Illumina NovaSeq 6000                                  341952</span></span>
<span><span class="co">#&gt;              analysis_Nucleotide sequences after format-specific filtering</span></span>
<span><span class="co">#&gt; MGYA00652201                                                        223726</span></span>
<span><span class="co">#&gt; MGYA00652185                                                        292409</span></span>
<span><span class="co">#&gt; MGYA00643487                                                        162292</span></span>
<span><span class="co">#&gt; MGYA00643486                                                        233327</span></span>
<span><span class="co">#&gt; MGYA00643485                                                        318625</span></span>
<span><span class="co">#&gt; MGYA00643484                                                        341952</span></span>
<span><span class="co">#&gt;              analysis_Nucleotide sequences after length filtering</span></span>
<span><span class="co">#&gt; MGYA00652201                                               223726</span></span>
<span><span class="co">#&gt; MGYA00652185                                               292409</span></span>
<span><span class="co">#&gt; MGYA00643487                                               162292</span></span>
<span><span class="co">#&gt; MGYA00643486                                               233327</span></span>
<span><span class="co">#&gt; MGYA00643485                                               318625</span></span>
<span><span class="co">#&gt; MGYA00643484                                               341952</span></span>
<span><span class="co">#&gt;              analysis_Nucleotide sequences after undetermined bases filtering</span></span>
<span><span class="co">#&gt; MGYA00652201                                                           223726</span></span>
<span><span class="co">#&gt; MGYA00652185                                                           292409</span></span>
<span><span class="co">#&gt; MGYA00643487                                                           162292</span></span>
<span><span class="co">#&gt; MGYA00643486                                                           233327</span></span>
<span><span class="co">#&gt; MGYA00643485                                                           318625</span></span>
<span><span class="co">#&gt; MGYA00643484                                                           341952</span></span>
<span><span class="co">#&gt;              analysis_Reads with predicted CDS</span></span>
<span><span class="co">#&gt; MGYA00652201                            223146</span></span>
<span><span class="co">#&gt; MGYA00652185                            291506</span></span>
<span><span class="co">#&gt; MGYA00643487                            159860</span></span>
<span><span class="co">#&gt; MGYA00643486                            232596</span></span>
<span><span class="co">#&gt; MGYA00643485                            317405</span></span>
<span><span class="co">#&gt; MGYA00643484                            341116</span></span>
<span><span class="co">#&gt;              analysis_Reads with predicted RNA</span></span>
<span><span class="co">#&gt; MGYA00652201                              3008</span></span>
<span><span class="co">#&gt; MGYA00652185                              4472</span></span>
<span><span class="co">#&gt; MGYA00643487                              3518</span></span>
<span><span class="co">#&gt; MGYA00643486                              3513</span></span>
<span><span class="co">#&gt; MGYA00643485                              5420</span></span>
<span><span class="co">#&gt; MGYA00643484                              4849</span></span>
<span><span class="co">#&gt;              analysis_Reads with InterProScan match analysis_Predicted CDS</span></span>
<span><span class="co">#&gt; MGYA00652201                                 182364                 489141</span></span>
<span><span class="co">#&gt; MGYA00652185                                 241005                 674191</span></span>
<span><span class="co">#&gt; MGYA00643487                                 124819                 519411</span></span>
<span><span class="co">#&gt; MGYA00643486                                 192842                 542005</span></span>
<span><span class="co">#&gt; MGYA00643485                                 259359                 893435</span></span>
<span><span class="co">#&gt; MGYA00643484                                 280422                 826459</span></span>
<span><span class="co">#&gt;              analysis_Predicted CDS with InterProScan match</span></span>
<span><span class="co">#&gt; MGYA00652201                                         331022</span></span>
<span><span class="co">#&gt; MGYA00652185                                         472354</span></span>
<span><span class="co">#&gt; MGYA00643487                                         330317</span></span>
<span><span class="co">#&gt; MGYA00643486                                         379304</span></span>
<span><span class="co">#&gt; MGYA00643485                                         585352</span></span>
<span><span class="co">#&gt; MGYA00643484                                         572175</span></span>
<span><span class="co">#&gt;              analysis_Total InterProScan matches</span></span>
<span><span class="co">#&gt; MGYA00652201                             1035846</span></span>
<span><span class="co">#&gt; MGYA00652185                             1520845</span></span>
<span><span class="co">#&gt; MGYA00643487                             1134950</span></span>
<span><span class="co">#&gt; MGYA00643486                             1221863</span></span>
<span><span class="co">#&gt; MGYA00643485                             1928199</span></span>
<span><span class="co">#&gt; MGYA00643484                             1847000</span></span>
<span><span class="co">#&gt;              analysis_Predicted SSU sequences analysis_Predicted LSU sequences</span></span>
<span><span class="co">#&gt; MGYA00652201                              141                              222</span></span>
<span><span class="co">#&gt; MGYA00652185                              268                              394</span></span>
<span><span class="co">#&gt; MGYA00643487                              217                              330</span></span>
<span><span class="co">#&gt; MGYA00643486                              199                              345</span></span>
<span><span class="co">#&gt; MGYA00643485                              273                              431</span></span>
<span><span class="co">#&gt; MGYA00643484                              243                              399</span></span>
<span><span class="co">#&gt;              analysis_acc_type study_attributes.accession</span></span>
<span><span class="co">#&gt; MGYA00652201     analysis-jobs               MGYS00006211</span></span>
<span><span class="co">#&gt; MGYA00652185     analysis-jobs               MGYS00006211</span></span>
<span><span class="co">#&gt; MGYA00643487     analysis-jobs               MGYS00006211</span></span>
<span><span class="co">#&gt; MGYA00643486     analysis-jobs               MGYS00006211</span></span>
<span><span class="co">#&gt; MGYA00643485     analysis-jobs               MGYS00006211</span></span>
<span><span class="co">#&gt; MGYA00643484     analysis-jobs               MGYS00006211</span></span>
<span><span class="co">#&gt;              study_attributes.bioproject study_attributes.samples-count</span></span>
<span><span class="co">#&gt; MGYA00652201                  PRJEB58755                             23</span></span>
<span><span class="co">#&gt; MGYA00652185                  PRJEB58755                             23</span></span>
<span><span class="co">#&gt; MGYA00643487                  PRJEB58755                             23</span></span>
<span><span class="co">#&gt; MGYA00643486                  PRJEB58755                             23</span></span>
<span><span class="co">#&gt; MGYA00643485                  PRJEB58755                             23</span></span>
<span><span class="co">#&gt; MGYA00643484                  PRJEB58755                             23</span></span>
<span><span class="co">#&gt;              study_attributes.is-private study_attributes.secondary-accession</span></span>
<span><span class="co">#&gt; MGYA00652201                       FALSE                            ERP143823</span></span>
<span><span class="co">#&gt; MGYA00652185                       FALSE                            ERP143823</span></span>
<span><span class="co">#&gt; MGYA00643487                       FALSE                            ERP143823</span></span>
<span><span class="co">#&gt; MGYA00643486                       FALSE                            ERP143823</span></span>
<span><span class="co">#&gt; MGYA00643485                       FALSE                            ERP143823</span></span>
<span><span class="co">#&gt; MGYA00643484                       FALSE                            ERP143823</span></span>
<span><span class="co">#&gt;              study_attributes.centre-name</span></span>
<span><span class="co">#&gt; MGYA00652201  Leitat Technological Center</span></span>
<span><span class="co">#&gt; MGYA00652185  Leitat Technological Center</span></span>
<span><span class="co">#&gt; MGYA00643487  Leitat Technological Center</span></span>
<span><span class="co">#&gt; MGYA00643486  Leitat Technological Center</span></span>
<span><span class="co">#&gt; MGYA00643485  Leitat Technological Center</span></span>
<span><span class="co">#&gt; MGYA00643484  Leitat Technological Center</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                           study_attributes.study-abstract</span></span>
<span><span class="co">#&gt; MGYA00652201 Elimination of several antibiotics in water by bioelectrochemical cells. The main objective is study how the concentration of antibiotic resistant genes (ARG) changed depending on the voltage application.</span></span>
<span><span class="co">#&gt; MGYA00652185 Elimination of several antibiotics in water by bioelectrochemical cells. The main objective is study how the concentration of antibiotic resistant genes (ARG) changed depending on the voltage application.</span></span>
<span><span class="co">#&gt; MGYA00643487 Elimination of several antibiotics in water by bioelectrochemical cells. The main objective is study how the concentration of antibiotic resistant genes (ARG) changed depending on the voltage application.</span></span>
<span><span class="co">#&gt; MGYA00643486 Elimination of several antibiotics in water by bioelectrochemical cells. The main objective is study how the concentration of antibiotic resistant genes (ARG) changed depending on the voltage application.</span></span>
<span><span class="co">#&gt; MGYA00643485 Elimination of several antibiotics in water by bioelectrochemical cells. The main objective is study how the concentration of antibiotic resistant genes (ARG) changed depending on the voltage application.</span></span>
<span><span class="co">#&gt; MGYA00643484 Elimination of several antibiotics in water by bioelectrochemical cells. The main objective is study how the concentration of antibiotic resistant genes (ARG) changed depending on the voltage application.</span></span>
<span><span class="co">#&gt;                               study_attributes.study-name</span></span>
<span><span class="co">#&gt; MGYA00652201 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00652185 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00643487 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00643486 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00643485 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00643484 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt;              study_attributes.data-origination study_attributes.last-update</span></span>
<span><span class="co">#&gt; MGYA00652201                         SUBMITTED          2023-08-07T11:15:24</span></span>
<span><span class="co">#&gt; MGYA00652185                         SUBMITTED          2023-08-07T11:15:24</span></span>
<span><span class="co">#&gt; MGYA00643487                         SUBMITTED          2023-08-07T11:15:24</span></span>
<span><span class="co">#&gt; MGYA00643486                         SUBMITTED          2023-08-07T11:15:24</span></span>
<span><span class="co">#&gt; MGYA00643485                         SUBMITTED          2023-08-07T11:15:24</span></span>
<span><span class="co">#&gt; MGYA00643484                         SUBMITTED          2023-08-07T11:15:24</span></span>
<span><span class="co">#&gt;              study_accession study_acc_type sample_biosample sample_accession</span></span>
<span><span class="co">#&gt; MGYA00652201    MGYS00006211        studies   SAMEA112288610      ERS14399436</span></span>
<span><span class="co">#&gt; MGYA00652185    MGYS00006211        studies   SAMEA112288603      ERS14399429</span></span>
<span><span class="co">#&gt; MGYA00643487    MGYS00006211        studies   SAMEA112288605      ERS14399431</span></span>
<span><span class="co">#&gt; MGYA00643486    MGYS00006211        studies   SAMEA112288602      ERS14399428</span></span>
<span><span class="co">#&gt; MGYA00643485    MGYS00006211        studies   SAMEA112288590      ERS14399416</span></span>
<span><span class="co">#&gt; MGYA00643484    MGYS00006211        studies   SAMEA112288597      ERS14399423</span></span>
<span><span class="co">#&gt;              sample_sample-desc               sample_environment-biome</span></span>
<span><span class="co">#&gt; MGYA00652201            biofilm Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; MGYA00652185            biofilm Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; MGYA00643487            biomass                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643486            biofilm                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643485            biofilm                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643484            biofilm                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                          sample_environment-feature</span></span>
<span><span class="co">#&gt; MGYA00652201 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; MGYA00652185 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; MGYA00643487                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643486                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643485                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643484                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                   sample_environment-material sample_sample-name</span></span>
<span><span class="co">#&gt; MGYA00652201 Biofilm material (ENVO:01000156)            GREEN68</span></span>
<span><span class="co">#&gt; MGYA00652185 Biofilm material (ENVO:01000156)            GREEN61</span></span>
<span><span class="co">#&gt; MGYA00643487                             &lt;NA&gt;            GREEN63</span></span>
<span><span class="co">#&gt; MGYA00643486                             &lt;NA&gt;            GREEN60</span></span>
<span><span class="co">#&gt; MGYA00643485                             &lt;NA&gt;            GREEN48</span></span>
<span><span class="co">#&gt; MGYA00643484                             &lt;NA&gt;            GREEN55</span></span>
<span><span class="co">#&gt;              sample_sample-alias  sample_last-update</span></span>
<span><span class="co">#&gt; MGYA00652201             GREEN68 2023-08-07T11:15:24</span></span>
<span><span class="co">#&gt; MGYA00652185             GREEN61 2023-08-07T10:52:31</span></span>
<span><span class="co">#&gt; MGYA00643487             GREEN63 2023-04-23T08:53:19</span></span>
<span><span class="co">#&gt; MGYA00643486             GREEN60 2023-04-23T07:57:35</span></span>
<span><span class="co">#&gt; MGYA00643485             GREEN48 2023-04-23T06:14:08</span></span>
<span><span class="co">#&gt; MGYA00643484             GREEN55 2023-04-23T05:00:17</span></span>
<span><span class="co">#&gt;                                       sample_project name</span></span>
<span><span class="co">#&gt; MGYA00652201 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00652185 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00643487 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00643486 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00643485 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt; MGYA00643484 ARGs study in bioelectrochemical remediation</span></span>
<span><span class="co">#&gt;              sample_geographic location (country and/or sea,region)</span></span>
<span><span class="co">#&gt; MGYA00652201                                                  Spain</span></span>
<span><span class="co">#&gt; MGYA00652185                                                  Spain</span></span>
<span><span class="co">#&gt; MGYA00643487                                                  Spain</span></span>
<span><span class="co">#&gt; MGYA00643486                                                  Spain</span></span>
<span><span class="co">#&gt; MGYA00643485                                                  Spain</span></span>
<span><span class="co">#&gt; MGYA00643484                                                  Spain</span></span>
<span><span class="co">#&gt;              sample_collection date             sample_environment (biome)</span></span>
<span><span class="co">#&gt; MGYA00652201             2022-09-14 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; MGYA00652185             2022-09-14 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; MGYA00643487             2022-09-14                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643486             2022-09-14                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643485             2022-09-14                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643484             2022-09-14                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                        sample_environment (feature)</span></span>
<span><span class="co">#&gt; MGYA00652201 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; MGYA00652185 Laboratory environment (ENVO_01001405)</span></span>
<span><span class="co">#&gt; MGYA00643487                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643486                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643485                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643484                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                 sample_environment (material) sample_ENA checklist</span></span>
<span><span class="co">#&gt; MGYA00652201 Biofilm material (ENVO:01000156)            ERC000023</span></span>
<span><span class="co">#&gt; MGYA00652185 Biofilm material (ENVO:01000156)            ERC000023</span></span>
<span><span class="co">#&gt; MGYA00643487                             &lt;NA&gt;            ERC000023</span></span>
<span><span class="co">#&gt; MGYA00643486                             &lt;NA&gt;            ERC000023</span></span>
<span><span class="co">#&gt; MGYA00643485                             &lt;NA&gt;            ERC000023</span></span>
<span><span class="co">#&gt; MGYA00643484                             &lt;NA&gt;            ERC000023</span></span>
<span><span class="co">#&gt;              sample_acc_type assembly_accession</span></span>
<span><span class="co">#&gt; MGYA00652201         samples        ERZ20300939</span></span>
<span><span class="co">#&gt; MGYA00652185         samples        ERZ20300942</span></span>
<span><span class="co">#&gt; MGYA00643487         samples        ERZ16299686</span></span>
<span><span class="co">#&gt; MGYA00643486         samples        ERZ16299690</span></span>
<span><span class="co">#&gt; MGYA00643485         samples        ERZ16299649</span></span>
<span><span class="co">#&gt; MGYA00643484         samples        ERZ16299683</span></span>
<span><span class="co">#&gt;                                                      biome_string</span></span>
<span><span class="co">#&gt; MGYA00652201 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; MGYA00652185 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; MGYA00643487 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; MGYA00643486 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; MGYA00643485 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt; MGYA00643484 root:Environmental:Aquatic:Freshwater:Drinking water</span></span>
<span><span class="co">#&gt;              sample_collection-date sample_latitude sample_longitude</span></span>
<span><span class="co">#&gt; MGYA00652201                   &lt;NA&gt;            &lt;NA&gt;             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00652185                   &lt;NA&gt;            &lt;NA&gt;             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643487             2022-09-14            &lt;NA&gt;             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643486             2022-09-14            &lt;NA&gt;             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643485             2022-09-14            &lt;NA&gt;             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643484             2022-09-14            &lt;NA&gt;             &lt;NA&gt;</span></span>
<span><span class="co">#&gt;              sample_investigation type sample_geographic location (longitude)</span></span>
<span><span class="co">#&gt; MGYA00652201                      &lt;NA&gt;                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00652185                      &lt;NA&gt;                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643487                      &lt;NA&gt;                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643486                      &lt;NA&gt;                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643485                      &lt;NA&gt;                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643484                      &lt;NA&gt;                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;              sample_environmental package sample_sequencing method</span></span>
<span><span class="co">#&gt; MGYA00652201                         &lt;NA&gt;                     &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00652185                         &lt;NA&gt;                     &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643487                         &lt;NA&gt;                     &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643486                         &lt;NA&gt;                     &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643485                         &lt;NA&gt;                     &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643484                         &lt;NA&gt;                     &lt;NA&gt;</span></span>
<span><span class="co">#&gt;              sample_geographic location (latitude) run_accession</span></span>
<span><span class="co">#&gt; MGYA00652201                                  &lt;NA&gt;          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00652185                                  &lt;NA&gt;          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643487                                  &lt;NA&gt;          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643486                                  &lt;NA&gt;          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643485                                  &lt;NA&gt;          &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643484                                  &lt;NA&gt;          &lt;NA&gt;</span></span>
<span><span class="co">#&gt;              sample_analysis-completed sample_geo-loc-name</span></span>
<span><span class="co">#&gt; MGYA00652201                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00652185                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643487                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643486                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643485                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643484                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">#&gt;              sample_instrument model</span></span>
<span><span class="co">#&gt; MGYA00652201                    &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00652185                    &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643487                    &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643486                    &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643485                    &lt;NA&gt;</span></span>
<span><span class="co">#&gt; MGYA00643484                    &lt;NA&gt;</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fetch-microbiome-data">Fetch microbiome data<a class="anchor" aria-label="anchor" href="#fetch-microbiome-data"></a>
</h3>
<p>After we have selected the data to fetch, we can use
<code><a href="../reference/getResult.html">getResult()</a></code></p>
<p>The output is <em><a href="https://bioconductor.org/packages/3.19/TreeSummarizedExperiment" class="external-link">TreeSummarizedExperiment</a></em>
(<code>TreeSE</code>) or <em><a href="https://bioconductor.org/packages/3.19/MultiAssayExperiment" class="external-link">MultiAssayExperiment</a></em>
(<code>MAE</code>) depending on the dataset. If the dataset includes
only taxonomic profiling data, the output is a single
<code>TreeSE</code>. If dataset includes also functional data, the
output is multiple <code>TreeSE</code> objects that are linked together
by utilizing <code>MAE</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getResult.html">getResult</a></span><span class="op">(</span><span class="va">mg</span>, accession <span class="op">=</span> <span class="va">analyses_accessions</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span></span>
<span><span class="co">#&gt; A MultiAssayExperiment object of 6 listed</span></span>
<span><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">#&gt;  Containing an ExperimentList class object of length 6:</span></span>
<span><span class="co">#&gt;  [1] microbiota: TreeSummarizedExperiment with 3506 rows and 50 columns</span></span>
<span><span class="co">#&gt;  [2] go-slim: TreeSummarizedExperiment with 116 rows and 38 columns</span></span>
<span><span class="co">#&gt;  [3] go-terms: TreeSummarizedExperiment with 3133 rows and 38 columns</span></span>
<span><span class="co">#&gt;  [4] interpro-identifiers: TreeSummarizedExperiment with 18223 rows and 38 columns</span></span>
<span><span class="co">#&gt;  [5] taxonomy: TreeSummarizedExperiment with 3617 rows and 50 columns</span></span>
<span><span class="co">#&gt;  [6] taxonomy-lsu: TreeSummarizedExperiment with 3378 rows and 42 columns</span></span>
<span><span class="co">#&gt; Functionality:</span></span>
<span><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code></pre></div>
<p>You can get access to individual <code>TreeSE</code> object in
<code>MAE</code> by specifying index or name.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; class: TreeSummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 3506 50 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(3506): 82608 62797 ... 5820 6794</span></span>
<span><span class="co">#&gt; rowData names(9): Kingdom Phylum ... taxonomy1 taxonomy</span></span>
<span><span class="co">#&gt; colnames(50): MGYA00144458 MGYA00144419 ... MGYA00652185 MGYA00652201</span></span>
<span><span class="co">#&gt; colData names(64): analysis_analysis.status analysis_pipeline.version</span></span>
<span><span class="co">#&gt;   ... sample_geo.loc.name sample_instrument.model</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; rowLinks: NULL</span></span>
<span><span class="co">#&gt; rowTree: NULL</span></span>
<span><span class="co">#&gt; colLinks: NULL</span></span>
<span><span class="co">#&gt; colTree: NULL</span></span></code></pre></div>
<p><code>TreeSE</code> object is uniquely positioned to support
<code>SummarizedExperiment</code>-based microbiome data manipulation and
visualization. Moreover, it enables access to <code>miaverse</code>
tools. For example, we can estimate diversity of samples…</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/estimateDiversity.html" class="external-link">estimateDiversity</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, index <span class="op">=</span> <span class="st">"shannon"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: scuttle</span></span>
<span><span class="co">#&gt; Loading required package: ggplot2</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"shannon"</span>, x <span class="op">=</span> <span class="st">"sample_environment..biome."</span><span class="op">)</span></span></code></pre></div>
<p><img src="MGnifyR_files/figure-html/calculate_diversity-1.png" width="864"></p>
<p>… and plot abundances of most abundant phyla.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Agglomerate data</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExps</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/splitByRanks.html" class="external-link">splitByRanks</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">miaViz</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: ggraph</span></span>
<span></span>
<span><span class="co"># Plot top taxa</span></span>
<span><span class="va">top_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/summaries.html" class="external-link">getTopFeatures</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"Phylum"</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/miaViz/man/plotAbundance.html" class="external-link">plotAbundance</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"Phylum"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>, <span class="op">]</span>, rank <span class="op">=</span> <span class="st">"Phylum"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MGnifyR_files/figure-html/plot_abundance-1.png" width="700"></p>
<p>We can also perform other analyses such as principal component
analysis to microbial profiling data by utilizing miaverse tools.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Apply relative transformation</span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span><span class="co"># Perform PCoA</span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html" class="external-link">runMDS</a></span><span class="op">(</span></span>
<span>    <span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>    FUN <span class="op">=</span> <span class="fu">vegan</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></span>, method <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span></span>
<span><span class="co"># Plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span></span>
<span>    <span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"MDS"</span>, colour_by <span class="op">=</span> <span class="st">"sample_environment..biome."</span><span class="op">)</span></span></code></pre></div>
<p><img src="MGnifyR_files/figure-html/pcoa-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="fetch-raw-files">Fetch raw files<a class="anchor" aria-label="anchor" href="#fetch-raw-files"></a>
</h3>
<p>While <code><a href="../reference/getResult.html">getResult()</a></code> can be utilized to retrieve microbial
profiling data, <code><a href="../reference/getData.html">getData()</a></code> can be used more flexibly to
retrieve any kind of data from the database. It returns data as simple
data.frame or list format.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">publications</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getData.html">getData</a></span><span class="op">(</span><span class="va">mg</span>, type <span class="op">=</span> <span class="st">"publications"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">publications</span><span class="op">)</span></span>
<span><span class="co">#&gt;   document.id         type       id attributes.pubmed-id</span></span>
<span><span class="co">#&gt; 1           1 publications 36363763             36363763</span></span>
<span><span class="co">#&gt; 2           2 publications 35614211             35614211</span></span>
<span><span class="co">#&gt; 3           3 publications 35614182             35614182</span></span>
<span><span class="co">#&gt; 4           4 publications 35208912             35208912</span></span>
<span><span class="co">#&gt; 5           5 publications 35178461             35178461</span></span>
<span><span class="co">#&gt; 6           6 publications 35154023             35154023</span></span>
<span><span class="co">#&gt;   attributes.pubmed-central-id</span></span>
<span><span class="co">#&gt; 1                           NA</span></span>
<span><span class="co">#&gt; 2                           NA</span></span>
<span><span class="co">#&gt; 3                           NA</span></span>
<span><span class="co">#&gt; 4                           NA</span></span>
<span><span class="co">#&gt; 5                           NA</span></span>
<span><span class="co">#&gt; 6                           NA</span></span>
<span><span class="co">#&gt;                                                                                                                                                                attributes.pub-title</span></span>
<span><span class="co">#&gt; 1                                          Agricultural Crops Grown in Laboratory Conditions on Chernevaya Taiga Soil Demonstrate Unique Composition of the Rhizosphere Microbiota.</span></span>
<span><span class="co">#&gt; 2                                                                                                     Discovery of bioactive microbial gene products in inflammatory bowel disease.</span></span>
<span><span class="co">#&gt; 3                                                                                                              Lake microbiome and trophy fluctuations of the ancient hemp rettery.</span></span>
<span><span class="co">#&gt; 4 Mycobiome-Host Coevolution? The Mycobiome of Ancestral Human Populations Seems to Be Different and Less Diverse Than Those of Extant Native and Urban-Industrialized Populations.</span></span>
<span><span class="co">#&gt; 5                                                             Effects of triclosan on bacterial community composition and &lt;i&gt;Vibrio&lt;/i&gt; populations in natural seawater microcosms.</span></span>
<span><span class="co">#&gt; 6                                                       Field Site-Specific Effects of an &lt;i&gt;Azospirillum&lt;/i&gt; Seed Inoculant on Key Microbial Functional Groups in the Rhizosphere.</span></span>
<span><span class="co">#&gt;   attributes.pub-abstract</span></span>
<span><span class="co">#&gt; 1                      NA</span></span>
<span><span class="co">#&gt; 2                      NA</span></span>
<span><span class="co">#&gt; 3                      NA</span></span>
<span><span class="co">#&gt; 4                      NA</span></span>
<span><span class="co">#&gt; 5                      NA</span></span>
<span><span class="co">#&gt; 6                      NA</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                               attributes.authors</span></span>
<span><span class="co">#&gt; 1                                                                                                                                                                         Kravchenko I, Rayko M, Tikhonova E, Konopkin A, Abakumov E, Lapidus A.</span></span>
<span><span class="co">#&gt; 2 Zhang Y, Bhosle A, Bae S, McIver LJ, Pishchany G, Accorsi EK, Thompson KN, Arze C, Wang Y, Subramanian A, Kearney SM, Pawluk A, Plichta DR, Rahnavard A, Shafquat A, Xavier RJ, Vlamakis H, Garrett WS, Krueger A, Huttenhower C, Franzosa EA.</span></span>
<span><span class="co">#&gt; 3                                                                                                                      Iwańska O, Latoch P, Suchora M, Pidek IA, Huber M, Bubak I, Kopik N, Kovalenko M, Gąsiorowski M, Armache JP, Starosta AL.</span></span>
<span><span class="co">#&gt; 4                                                                                                                                                                       Reynoso-García J, Narganes-Storde Y, Santiago-Rodriguez TM, Toranzos GA.</span></span>
<span><span class="co">#&gt; 5                                                                                                                                                                                      Lydon KA, Glinski DA, Westrich JR, Henderson WM, Lipp EK.</span></span>
<span><span class="co">#&gt; 6                                                                                                                                                     Renoud S, Vacheron J, Abrouk D, Prigent-Combaret C, Legendre L, Muller D, Moënne-Loccoz Y.</span></span>
<span><span class="co">#&gt;                   attributes.doi        attributes.isbn</span></span>
<span><span class="co">#&gt; 1 10.3390/microorganisms10112171              2076-2607</span></span>
<span><span class="co">#&gt; 2     10.1038/s41586-022-04648-7 0028-0836; 1476-4687; </span></span>
<span><span class="co">#&gt; 3     10.1038/s41598-022-12761-w              2045-2322</span></span>
<span><span class="co">#&gt; 4 10.3390/microorganisms10020459              2076-2607</span></span>
<span><span class="co">#&gt; 5           10.1525/elementa.141              2325-1026</span></span>
<span><span class="co">#&gt; 6      10.3389/fmicb.2021.760512              1664-302x</span></span>
<span><span class="co">#&gt;   attributes.published-year               attributes.pub-type attributes.issue</span></span>
<span><span class="co">#&gt; 1                      2022                   journal article               NA</span></span>
<span><span class="co">#&gt; 2                      2022 research-article; journal article               NA</span></span>
<span><span class="co">#&gt; 3                      2022                   journal article               NA</span></span>
<span><span class="co">#&gt; 4                      2022                   journal article               NA</span></span>
<span><span class="co">#&gt; 5                      2017                   journal article               NA</span></span>
<span><span class="co">#&gt; 6                      2021                   journal article               NA</span></span>
<span><span class="co">#&gt;   attributes.volume attributes.raw-pages attributes.iso-journal</span></span>
<span><span class="co">#&gt; 1                10                 2171         Microorganisms</span></span>
<span><span class="co">#&gt; 2               606              754-760                 Nature</span></span>
<span><span class="co">#&gt; 3                12                 8846                Sci Rep</span></span>
<span><span class="co">#&gt; 4                10                  n/a         Microorganisms</span></span>
<span><span class="co">#&gt; 5                 5                 1-16    Elementa (Wash D C)</span></span>
<span><span class="co">#&gt; 6                12               760512        Front Microbiol</span></span>
<span><span class="co">#&gt;   attributes.medline-journal attributes.pub-url attributes.studies-count</span></span>
<span><span class="co">#&gt; 1                         NA                 NA                        1</span></span>
<span><span class="co">#&gt; 2                         NA                 NA                        1</span></span>
<span><span class="co">#&gt; 3                         NA                 NA                        2</span></span>
<span><span class="co">#&gt; 4                         NA                 NA                        1</span></span>
<span><span class="co">#&gt; 5                         NA                 NA                        1</span></span>
<span><span class="co">#&gt; 6                         NA                 NA                        3</span></span>
<span><span class="co">#&gt;   attributes.samples-count</span></span>
<span><span class="co">#&gt; 1                        5</span></span>
<span><span class="co">#&gt; 2                      176</span></span>
<span><span class="co">#&gt; 3                       18</span></span>
<span><span class="co">#&gt; 4                       57</span></span>
<span><span class="co">#&gt; 5                       24</span></span>
<span><span class="co">#&gt; 6                        3</span></span>
<span><span class="co">#&gt;                                                                    links.self</span></span>
<span><span class="co">#&gt; 1 https://www.ebi.ac.uk/metagenomics/api/v1/publications/36363763?format=json</span></span>
<span><span class="co">#&gt; 2 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35614211?format=json</span></span>
<span><span class="co">#&gt; 3 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35614182?format=json</span></span>
<span><span class="co">#&gt; 4 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35208912?format=json</span></span>
<span><span class="co">#&gt; 5 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35178461?format=json</span></span>
<span><span class="co">#&gt; 6 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35154023?format=json</span></span>
<span><span class="co">#&gt;                                                   relationships.studies.links.related</span></span>
<span><span class="co">#&gt; 1 https://www.ebi.ac.uk/metagenomics/api/v1/publications/36363763/studies?format=json</span></span>
<span><span class="co">#&gt; 2 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35614211/studies?format=json</span></span>
<span><span class="co">#&gt; 3 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35614182/studies?format=json</span></span>
<span><span class="co">#&gt; 4 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35208912/studies?format=json</span></span>
<span><span class="co">#&gt; 5 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35178461/studies?format=json</span></span>
<span><span class="co">#&gt; 6 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35154023/studies?format=json</span></span>
<span><span class="co">#&gt;                                                   relationships.samples.links.related</span></span>
<span><span class="co">#&gt; 1 https://www.ebi.ac.uk/metagenomics/api/v1/publications/36363763/samples?format=json</span></span>
<span><span class="co">#&gt; 2 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35614211/samples?format=json</span></span>
<span><span class="co">#&gt; 3 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35614182/samples?format=json</span></span>
<span><span class="co">#&gt; 4 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35208912/samples?format=json</span></span>
<span><span class="co">#&gt; 5 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35178461/samples?format=json</span></span>
<span><span class="co">#&gt; 6 https://www.ebi.ac.uk/metagenomics/api/v1/publications/35154023/samples?format=json</span></span>
<span><span class="co">#&gt;         ..JSON</span></span>
<span><span class="co">#&gt; 1 publicat....</span></span>
<span><span class="co">#&gt; 2 publicat....</span></span>
<span><span class="co">#&gt; 3 publicat....</span></span>
<span><span class="co">#&gt; 4 publicat....</span></span>
<span><span class="co">#&gt; 5 publicat....</span></span>
<span><span class="co">#&gt; 6 publicat....</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fetch-sequence-files">Fetch sequence files<a class="anchor" aria-label="anchor" href="#fetch-sequence-files"></a>
</h3>
<p>Finally, we can use <code><a href="../reference/getFile.html">searchFile()</a></code> and
<code><a href="../reference/getFile.html">getFile()</a></code> to retrieve other MGnify pipeline outputs such as
merged sequence reads, assembled contigs, and details of the functional
analyses.</p>
<p>With <code><a href="../reference/getFile.html">searchFile()</a></code>, we can search files from the
database.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dl_urls</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">searchFile</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">analyses_accessions</span>, type <span class="op">=</span> <span class="st">"analyses"</span><span class="op">)</span></span></code></pre></div>
<p>The returned table contains search results related to analyses that
we fed as an input. The table contains information on file and also URL
address from where the file can be loaded.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">target_urls</span> <span class="op">&lt;-</span> <span class="va">dl_urls</span><span class="op">[</span></span>
<span>    <span class="va">dl_urls</span><span class="op">$</span><span class="va">attributes.description.label</span> <span class="op">==</span> <span class="st">"Predicted alpha tmRNA"</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">target_urls</span><span class="op">)</span></span>
<span><span class="co">#&gt;         type                                       id</span></span>
<span><span class="co">#&gt; 24  analyses ERZ20300939_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 75  analyses ERZ20300942_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 126 analyses ERZ16299686_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 177 analyses ERZ16299690_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 228 analyses ERZ16299649_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 279 analyses ERZ16299683_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt;                             attributes.alias attributes.file.format.name</span></span>
<span><span class="co">#&gt; 24  ERZ20300939_alpha_tmRNA.RF01849.fasta.gz                       FASTA</span></span>
<span><span class="co">#&gt; 75  ERZ20300942_alpha_tmRNA.RF01849.fasta.gz                       FASTA</span></span>
<span><span class="co">#&gt; 126 ERZ16299686_alpha_tmRNA.RF01849.fasta.gz                       FASTA</span></span>
<span><span class="co">#&gt; 177 ERZ16299690_alpha_tmRNA.RF01849.fasta.gz                       FASTA</span></span>
<span><span class="co">#&gt; 228 ERZ16299649_alpha_tmRNA.RF01849.fasta.gz                       FASTA</span></span>
<span><span class="co">#&gt; 279 ERZ16299683_alpha_tmRNA.RF01849.fasta.gz                       FASTA</span></span>
<span><span class="co">#&gt;     attributes.file.format.extension attributes.file.format.compression</span></span>
<span><span class="co">#&gt; 24                             fasta                               TRUE</span></span>
<span><span class="co">#&gt; 75                             fasta                               TRUE</span></span>
<span><span class="co">#&gt; 126                            fasta                               TRUE</span></span>
<span><span class="co">#&gt; 177                            fasta                               TRUE</span></span>
<span><span class="co">#&gt; 228                            fasta                               TRUE</span></span>
<span><span class="co">#&gt; 279                            fasta                               TRUE</span></span>
<span><span class="co">#&gt;     attributes.description.label</span></span>
<span><span class="co">#&gt; 24         Predicted alpha tmRNA</span></span>
<span><span class="co">#&gt; 75         Predicted alpha tmRNA</span></span>
<span><span class="co">#&gt; 126        Predicted alpha tmRNA</span></span>
<span><span class="co">#&gt; 177        Predicted alpha tmRNA</span></span>
<span><span class="co">#&gt; 228        Predicted alpha tmRNA</span></span>
<span><span class="co">#&gt; 279        Predicted alpha tmRNA</span></span>
<span><span class="co">#&gt;                                 attributes.description.description</span></span>
<span><span class="co">#&gt; 24  Predicted Alphaproteobacteria transfer-messenger RNA (RF01849)</span></span>
<span><span class="co">#&gt; 75  Predicted Alphaproteobacteria transfer-messenger RNA (RF01849)</span></span>
<span><span class="co">#&gt; 126 Predicted Alphaproteobacteria transfer-messenger RNA (RF01849)</span></span>
<span><span class="co">#&gt; 177 Predicted Alphaproteobacteria transfer-messenger RNA (RF01849)</span></span>
<span><span class="co">#&gt; 228 Predicted Alphaproteobacteria transfer-messenger RNA (RF01849)</span></span>
<span><span class="co">#&gt; 279 Predicted Alphaproteobacteria transfer-messenger RNA (RF01849)</span></span>
<span><span class="co">#&gt;     attributes.group.type attributes.file.checksum.checksum</span></span>
<span><span class="co">#&gt; 24        non-coding RNAs                                  </span></span>
<span><span class="co">#&gt; 75        non-coding RNAs                                  </span></span>
<span><span class="co">#&gt; 126       non-coding RNAs                                  </span></span>
<span><span class="co">#&gt; 177       non-coding RNAs                                  </span></span>
<span><span class="co">#&gt; 228       non-coding RNAs                                  </span></span>
<span><span class="co">#&gt; 279       non-coding RNAs                                  </span></span>
<span><span class="co">#&gt;     attributes.file.checksum.checksum.algorithm</span></span>
<span><span class="co">#&gt; 24                                             </span></span>
<span><span class="co">#&gt; 75                                             </span></span>
<span><span class="co">#&gt; 126                                            </span></span>
<span><span class="co">#&gt; 177                                            </span></span>
<span><span class="co">#&gt; 228                                            </span></span>
<span><span class="co">#&gt; 279                                            </span></span>
<span><span class="co">#&gt;     relationships.pipeline.data.type relationships.pipeline.data.id</span></span>
<span><span class="co">#&gt; 24                         pipelines                            5.0</span></span>
<span><span class="co">#&gt; 75                         pipelines                            5.0</span></span>
<span><span class="co">#&gt; 126                        pipelines                            5.0</span></span>
<span><span class="co">#&gt; 177                        pipelines                            5.0</span></span>
<span><span class="co">#&gt; 228                        pipelines                            5.0</span></span>
<span><span class="co">#&gt; 279                        pipelines                            5.0</span></span>
<span><span class="co">#&gt;                                          relationships.pipeline.related</span></span>
<span><span class="co">#&gt; 24  https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/5.0?format=json</span></span>
<span><span class="co">#&gt; 75  https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/5.0?format=json</span></span>
<span><span class="co">#&gt; 126 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/5.0?format=json</span></span>
<span><span class="co">#&gt; 177 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/5.0?format=json</span></span>
<span><span class="co">#&gt; 228 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/5.0?format=json</span></span>
<span><span class="co">#&gt; 279 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/5.0?format=json</span></span>
<span><span class="co">#&gt;                                                                                                      download_url</span></span>
<span><span class="co">#&gt; 24  https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00652201/file/ERZ20300939_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 75  https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00652185/file/ERZ20300942_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 126 https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00643487/file/ERZ16299686_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 177 https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00643486/file/ERZ16299690_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 228 https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00643485/file/ERZ16299649_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt; 279 https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00643484/file/ERZ16299683_alpha_tmRNA.RF01849.fasta.gz</span></span>
<span><span class="co">#&gt;        accession</span></span>
<span><span class="co">#&gt; 24  MGYA00652201</span></span>
<span><span class="co">#&gt; 75  MGYA00652185</span></span>
<span><span class="co">#&gt; 126 MGYA00643487</span></span>
<span><span class="co">#&gt; 177 MGYA00643486</span></span>
<span><span class="co">#&gt; 228 MGYA00643485</span></span>
<span><span class="co">#&gt; 279 MGYA00643484</span></span></code></pre></div>
<p>Finally, we can download the files with <code><a href="../reference/getFile.html">getFile()</a></code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Just select a single file from the target_urls list for demonstration.</span></span>
<span><span class="va">file_url</span> <span class="op">&lt;-</span> <span class="va">target_urls</span><span class="op">$</span><span class="va">download_url</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">cached_location</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">getFile</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">file_url</span><span class="op">)</span></span></code></pre></div>
<p>The function returns a path where the file is stored.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Where are the files?</span></span>
<span><span class="va">cached_location</span></span>
<span><span class="co">#&gt; [1] "/.MGnifyR_cache/analyses/MGYA00652201/file/ERZ20300939_alpha_tmRNA.RF01849.fasta.gz"</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R Under development (unstable) (2024-02-21 r85967)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] miaViz_1.11.0                   ggraph_2.2.0                   </span></span>
<span><span class="co">#&gt;  [3] scater_1.31.2                   ggplot2_3.5.0                  </span></span>
<span><span class="co">#&gt;  [5] scuttle_1.13.0                  MGnifyR_0.99.22                </span></span>
<span><span class="co">#&gt;  [7] biomformat_1.31.0               mia_1.11.1                     </span></span>
<span><span class="co">#&gt;  [9] MultiAssayExperiment_1.29.1     TreeSummarizedExperiment_2.11.0</span></span>
<span><span class="co">#&gt; [11] Biostrings_2.71.2               XVector_0.43.1                 </span></span>
<span><span class="co">#&gt; [13] SingleCellExperiment_1.25.0     SummarizedExperiment_1.33.3    </span></span>
<span><span class="co">#&gt; [15] Biobase_2.63.0                  GenomicRanges_1.55.3           </span></span>
<span><span class="co">#&gt; [17] GenomeInfoDb_1.39.6             IRanges_2.37.1                 </span></span>
<span><span class="co">#&gt; [19] S4Vectors_0.41.3                BiocGenerics_0.49.1            </span></span>
<span><span class="co">#&gt; [21] MatrixGenerics_1.15.0           matrixStats_1.2.0              </span></span>
<span><span class="co">#&gt; [23] knitr_1.45                      BiocStyle_2.31.0               </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] jsonlite_1.8.8              tidyjson_0.3.2             </span></span>
<span><span class="co">#&gt;   [3] magrittr_2.0.3              ggbeeswarm_0.7.2           </span></span>
<span><span class="co">#&gt;   [5] farver_2.1.1                rmarkdown_2.25             </span></span>
<span><span class="co">#&gt;   [7] fs_1.6.3                    zlibbioc_1.49.0            </span></span>
<span><span class="co">#&gt;   [9] ragg_1.2.7                  vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [11] memoise_2.0.1               DelayedMatrixStats_1.25.1  </span></span>
<span><span class="co">#&gt;  [13] RCurl_1.98-1.14             ggtree_3.11.0              </span></span>
<span><span class="co">#&gt;  [15] htmltools_0.5.7             S4Arrays_1.3.3             </span></span>
<span><span class="co">#&gt;  [17] BiocBaseUtils_1.5.0         BiocNeighbors_1.21.2       </span></span>
<span><span class="co">#&gt;  [19] Rhdf5lib_1.25.1             gridGraphics_0.5-1         </span></span>
<span><span class="co">#&gt;  [21] SparseArray_1.3.4           rhdf5_2.47.4               </span></span>
<span><span class="co">#&gt;  [23] sass_0.4.8                  bslib_0.6.1                </span></span>
<span><span class="co">#&gt;  [25] desc_1.4.3                  plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;  [27] DECIPHER_2.31.2             cachem_1.0.8               </span></span>
<span><span class="co">#&gt;  [29] igraph_2.0.2                lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt;  [31] pkgconfig_2.0.3             rsvd_1.0.5                 </span></span>
<span><span class="co">#&gt;  [33] Matrix_1.6-5                R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [35] fastmap_1.1.1               GenomeInfoDbData_1.2.11    </span></span>
<span><span class="co">#&gt;  [37] aplot_0.2.2                 digest_0.6.34              </span></span>
<span><span class="co">#&gt;  [39] ggnewscale_0.4.10           colorspace_2.1-0           </span></span>
<span><span class="co">#&gt;  [41] patchwork_1.2.0             irlba_2.3.5.1              </span></span>
<span><span class="co">#&gt;  [43] textshaping_0.3.7           vegan_2.6-4                </span></span>
<span><span class="co">#&gt;  [45] beachmat_2.19.1             labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [47] fansi_1.0.6                 urltools_1.7.3             </span></span>
<span><span class="co">#&gt;  [49] polyclip_1.10-6             httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [51] abind_1.4-5                 mgcv_1.9-1                 </span></span>
<span><span class="co">#&gt;  [53] compiler_4.4.0              withr_3.0.0                </span></span>
<span><span class="co">#&gt;  [55] BiocParallel_1.37.0         viridis_0.6.5              </span></span>
<span><span class="co">#&gt;  [57] DBI_1.2.2                   highr_0.10                 </span></span>
<span><span class="co">#&gt;  [59] ggforce_0.4.2               MASS_7.3-60.2              </span></span>
<span><span class="co">#&gt;  [61] DelayedArray_0.29.5         bluster_1.13.0             </span></span>
<span><span class="co">#&gt;  [63] permute_0.9-7               tools_4.4.0                </span></span>
<span><span class="co">#&gt;  [65] vipor_0.4.7                 beeswarm_0.4.0             </span></span>
<span><span class="co">#&gt;  [67] ape_5.7-1                   glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [69] nlme_3.1-164                rhdf5filters_1.15.2        </span></span>
<span><span class="co">#&gt;  [71] grid_4.4.0                  cluster_2.1.6              </span></span>
<span><span class="co">#&gt;  [73] reshape2_1.4.4              generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [75] gtable_0.3.4                tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [77] tidygraph_1.3.1             BiocSingular_1.19.0        </span></span>
<span><span class="co">#&gt;  [79] ScaledMatrix_1.11.0         utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [81] ggrepel_0.9.5               pillar_1.9.0               </span></span>
<span><span class="co">#&gt;  [83] stringr_1.5.1               yulab.utils_0.1.4          </span></span>
<span><span class="co">#&gt;  [85] splines_4.4.0               tweenr_2.0.3               </span></span>
<span><span class="co">#&gt;  [87] dplyr_1.1.4                 treeio_1.27.0              </span></span>
<span><span class="co">#&gt;  [89] lattice_0.22-5              tidyselect_1.2.0           </span></span>
<span><span class="co">#&gt;  [91] DirichletMultinomial_1.45.0 gridExtra_2.3              </span></span>
<span><span class="co">#&gt;  [93] bookdown_0.37               xfun_0.42                  </span></span>
<span><span class="co">#&gt;  [95] graphlayouts_1.1.0          stringi_1.8.3              </span></span>
<span><span class="co">#&gt;  [97] ggfun_0.1.4                 lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [99] yaml_2.3.8                  evaluate_0.23              </span></span>
<span><span class="co">#&gt; [101] codetools_0.2-19            tibble_3.2.1               </span></span>
<span><span class="co">#&gt; [103] BiocManager_1.30.22         ggplotify_0.1.2            </span></span>
<span><span class="co">#&gt; [105] cli_3.6.2                   systemfonts_1.0.5          </span></span>
<span><span class="co">#&gt; [107] munsell_0.5.0               jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [109] Rcpp_1.0.12                 triebeard_0.4.1            </span></span>
<span><span class="co">#&gt; [111] parallel_4.4.0              pkgdown_2.0.7              </span></span>
<span><span class="co">#&gt; [113] assertthat_0.2.1            sparseMatrixStats_1.15.0   </span></span>
<span><span class="co">#&gt; [115] bitops_1.0-7                decontam_1.23.0            </span></span>
<span><span class="co">#&gt; [117] viridisLite_0.4.2           tidytree_0.4.6             </span></span>
<span><span class="co">#&gt; [119] scales_1.3.0                purrr_1.0.2                </span></span>
<span><span class="co">#&gt; [121] crayon_1.5.2                rlang_1.1.3</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Tuomas Borman, Ben Allen, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
