<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Metagenomics bioinformatics at MGnify • MGnifyR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Metagenomics bioinformatics at MGnify">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MGnifyR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.30</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/MGnifyR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/MGnify_course.html">Metagenomics bioinformatics at MGnify</a></li>
    <li><a class="dropdown-item" href="../articles/MGnifyR_long.html">MGnifyR, extended vignette</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/EBI-Metagenomics/MGnifyR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Metagenomics bioinformatics at MGnify</h1>
            
            <h4 data-toc-skip class="date">2024-09-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/minor/vignettes/MGnify_course.Rmd" class="external-link"><code>vignettes/MGnify_course.Rmd</code></a></small>
      <div class="d-none name"><code>MGnify_course.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this notebook we aim to demonstrate how the <code>MGnifyR</code>
tool can be used to fetch data of a MGnify microbiome data resource. We
then showcase how to analyze the datausing advanced microbiome data
science tools, including estimating alpha and beta diversity, as well as
performing differential abundance analysis.</p>
<p><a href="https://www.bioconductor.org/packages/release/bioc/html/MGnifyR.html" class="external-link"><code>MGnifyR</code></a>
is an R/Bioconductor package that provides a set of tools for easily
accessing and processing MGnify data in R, making queries to MGnify
databases through the <a href="https://www.ebi.ac.uk/metagenomics/api/v1/" class="external-link">MGnify API</a>.</p>
<p>The benefit of <code>MGnifyR</code> is that it streamlines data
access, allowing users to fetch data either in its “raw” format or
directly as a <a href="https://microbiome.github.io/OMA/docs/devel/pages/containers.html" class="external-link"><code>TreeSummarizedExperiment</code>
(<code>TreeSE</code>)</a> object. This enables seamless integration with
custom workflows for analysis.</p>
<p>Utilizing <code>TreeSE</code> provides access to a wide range of
tools within Bioconductor’s <code>SummarizedExperiment</code>
(<code>SE</code>) ecosystem. It also integrates with the <a href="https://microbiome.github.io/mia/" class="external-link"><code>mia</code> package</a>,
which offers microbiome-specific methods within the <code>SE</code>
framework.</p>
<p>For more information on microbiome data science in Bioconductor,
refer to <a href="https://microbiome.github.io/OMA/docs/devel/" class="external-link">Orchestrating
Microbiome Analysis (OMA) online book</a>.</p>
</div>
<div class="section level2">
<h2 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># List of packages that we need</span></span>
<span><span class="va">packages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ANCOMBC"</span>, <span class="st">"MGnifyR"</span>, <span class="st">"mia"</span>,  <span class="st">"miaViz"</span>, <span class="st">"scater"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get packages that are already installed</span></span>
<span><span class="va">packages_already_installed</span> <span class="op">&lt;-</span> <span class="va">packages</span><span class="op">[</span> <span class="va">packages</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span> <span class="op">]</span></span>
<span></span>
<span><span class="co"># Get packages that need to be installed</span></span>
<span><span class="va">packages_need_to_install</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">setdiff</a></span><span class="op">(</span> <span class="va">packages</span>, <span class="va">packages_already_installed</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># Loads BiocManager into the session. Install it if it is not already installed.</span></span>
<span><span class="kw">if</span><span class="op">(</span> <span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span><span class="op">)</span> <span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># If there are packages that need to be installed, installs them with BiocManager</span></span>
<span><span class="co"># Updates old packages.</span></span>
<span><span class="kw">if</span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">packages_need_to_install</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="op">)</span> <span class="op">{</span></span>
<span>   <span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">packages_need_to_install</span>, ask <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Load all packages into session. Stop if there are packages that were not</span></span>
<span><span class="co"># successfully loaded</span></span>
<span><span class="va">pkgs_not_loaded</span> <span class="op">&lt;-</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">packages</span>, <span class="va">require</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">pkgs_not_loaded</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pkgs_not_loaded</span><span class="op">)</span><span class="op">[</span> <span class="va">pkgs_not_loaded</span> <span class="op">]</span></span>
<span><span class="kw">if</span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">pkgs_not_loaded</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="op">)</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span></span>
<span>        <span class="st">"Error in loading the following packages into the session: '"</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">pkgs_not_loaded</span>, collapse <span class="op">=</span> <span class="st">"', '"</span><span class="op">)</span>, <span class="st">"'"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-import">Data import<a class="anchor" aria-label="anchor" href="#data-import"></a>
</h2>
<p>To interact with the MGnify database, we need to create a
<code>MgnifyClient</code> object. This object allows us to store options
for data fetching. For instance, we can configure it to use a cache for
improved efficiency.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create the MgnifyClient object with caching enabled</span></span>
<span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MgnifyClient.html">MgnifyClient</a></span><span class="op">(</span></span>
<span>  useCache <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cacheDir <span class="op">=</span> <span class="st">"/home/training"</span> <span class="co"># Set this to your desired cache directory</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>In this workflow, we will fetch taxonomy annotations and metadata
from the study <a href="https://www.ebi.ac.uk/metagenomics/studies/MGYS00005154" class="external-link">“MGYS00005154”</a>.
The dataset focuses on the human gut microbiome, analyzed across
different geographic regions.</p>
<p>We can now search for all analyses associated with the certain study.
The analysis refers to metagenomic runs performed to samples. Each
sample can have multiple runs made, which is why we work with analyses
and not with samples; analysis identifier points to a single entity.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">study_id</span> <span class="op">&lt;-</span> <span class="st">"MGYS00005154"</span></span>
<span><span class="va">analysis_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchAnalysis.html">searchAnalysis</a></span><span class="op">(</span><span class="va">mg</span>, <span class="st">"studies"</span>, <span class="va">study_id</span><span class="op">)</span></span></code></pre></div>
<p>Now we are ready to load the metadata on the analyses to get idea on
what kind of data we are dealing with.</p>
<p>There are currently (17 Sep 2024) almost 1,000 analyses available.
Downloading whole dataset will take some time, which is why we use
memory cache.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMetadata.html">getMetadata</a></span><span class="op">(</span><span class="va">mg</span>, accession <span class="op">=</span> <span class="va">analysis_id</span><span class="op">)</span></span></code></pre></div>
<p>We can see that there are analyses that are performed with different
pipelines. Let’s take only those analyses that are generated with the
pipeline version 5.0.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="va">metadata</span><span class="op">[</span><span class="va">metadata</span><span class="op">[[</span><span class="st">"analysis_pipeline-version"</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="st">"5.0"</span>, <span class="op">]</span></span></code></pre></div>
<p>We have now analyses that each point to unique sample. The final step
is to fetch abundance tables in <code>TreeSummarizedExperiment</code>
(<code>TreeSE</code>) format.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getResult.html">getResult</a></span><span class="op">(</span></span>
<span>    <span class="va">mg</span>,</span>
<span>    accession <span class="op">=</span> <span class="va">metadata</span><span class="op">[[</span><span class="st">"analysis_accession"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>    get.func <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>    <span class="op">)</span></span>
<span><span class="va">tse</span></span></code></pre></div>
<p>The fetched data is <code>TreeSE</code> object, including taxonomy
annotations. See <a href="https://microbiome.github.io/OMA/docs/devel/pages/containers.html" class="external-link">OMA
online book</a> on how to handle the data in this format.</p>
</div>
<div class="section level2">
<h2 id="preprocessing">Preprocessing<a class="anchor" aria-label="anchor" href="#preprocessing"></a>
</h2>
<p>Below, we agglomerate the data to the Order level, meaning we
summarize the abundances at this specific taxonomic rank. The OMA
provides a detailed <a href="https://microbiome.github.io/OMA/docs/devel/pages/agglomeration.html" class="external-link">chapter</a>
explaining agglomeration in more depth.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tse_order</span> <span class="op">&lt;-</span> <span class="fu">agglomerateByRank</span><span class="op">(</span><span class="va">tse</span>, rank <span class="op">=</span> <span class="st">"Order"</span><span class="op">)</span></span></code></pre></div>
<p>Because of the unique properties of microbiome data, we have to apply
transformations. Here, we perform relative transformation. You can find
more information on transformations from <a href="https://microbiome.github.io/OMA/docs/devel/pages/transformation.html" class="external-link">OMA</a>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Transform the main TreeSE</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">transformAssay</span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span><span class="co"># Transform the agglomerated TreeSE</span></span>
<span><span class="va">tse_order</span> <span class="op">&lt;-</span> <span class="fu">transformAssay</span><span class="op">(</span><span class="va">tse_order</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="alpha-diversity">Alpha diversity<a class="anchor" aria-label="anchor" href="#alpha-diversity"></a>
</h2>
<p>Alpha diversity measures community diversity within a sample. Learn
more on community diversity from <a href="https://microbiome.github.io/OMA/docs/devel/pages/alpha_diversity.html" class="external-link">here</a>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate alpha diversity</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">addAlpha</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a plot</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">plotColData</span><span class="op">(</span></span>
<span>  <span class="va">tse</span>,</span>
<span>  y <span class="op">=</span> <span class="st">"shannon_diversity"</span>,</span>
<span>  x <span class="op">=</span> <span class="st">"sample_geographic.location..country.and.or.sea.region."</span>,</span>
<span>  show_boxplot <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
<p>We can test whether the diversity differences are statistically
significant. We utilize Mann Whithney U test (or Wilcoxon test).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/pairwise.wilcox.test.html" class="external-link">pairwise.wilcox.test</a></span><span class="op">(</span></span>
<span>    <span class="va">tse</span><span class="op">[[</span><span class="st">"shannon_diversity"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>    <span class="va">tse</span><span class="op">[[</span><span class="st">"sample_geographic.location..country.and.or.sea.region."</span><span class="op">]</span><span class="op">]</span>,</span>
<span>    p.adjust.method <span class="op">=</span> <span class="st">"fdr"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>To add p-values to the plot, see <a href="https://microbiome.github.io/OMA/docs/devel/pages/alpha_diversity.html#visualizing-significance-in-group-wise-comparisons" class="external-link">OMA</a>.</p>
</div>
<div class="section level2">
<h2 id="beta-diversity">Beta diversity<a class="anchor" aria-label="anchor" href="#beta-diversity"></a>
</h2>
<p>We can assess the differences in microbial compositions between
samples, aiming to identify patterns in the data that are associated
with covariates.</p>
<p>To achieve this, we perform Principal Coordinate Analysis (PCoA)
using Bray-Curtis dissimilarity.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform PCoA</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">runMDS</span><span class="op">(</span></span>
<span>  <span class="va">tse</span>,</span>
<span>  FUN <span class="op">=</span> <span class="va">getDissimilarity</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"bray"</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"relabundance"</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># Visualize PCoA</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">plotReducedDim</span><span class="op">(</span></span>
<span>  <span class="va">tse</span>,</span>
<span>  dimred <span class="op">=</span> <span class="st">"MDS"</span>,</span>
<span>  colour_by <span class="op">=</span> <span class="st">"sample_geographic.location..country.and.or.sea.region."</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
<p>See <a href="https://microbiome.github.io/OMA/docs/devel/pages/beta_diversity.html" class="external-link">community
similarity chapter</a> from OMA for more information.</p>
</div>
<div class="section level2">
<h2 id="differential-abundance-analysis-daa">Differential abundance analysis (DAA)<a class="anchor" aria-label="anchor" href="#differential-abundance-analysis-daa"></a>
</h2>
<p>In DAA, we analyze whether abundances of certain features vary
between study groups. Again, OMA has a dedicated chapter also on this <a href="https://microbiome.github.io/OMA/docs/devel/pages/differential_abundance.html" class="external-link">topic</a>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform DAA</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">ancombc2</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">tse_order</span>,</span>
<span>    assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    fix_formula <span class="op">=</span> <span class="st">"sample_geographic.location..country.and.or.sea.region."</span>,</span>
<span>    p_adj_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>Next we visualize features that have the lowest p-values.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the most significant features</span></span>
<span><span class="va">n_top</span> <span class="op">&lt;-</span> <span class="fl">4</span></span>
<span><span class="va">res_tab</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">[[</span><span class="st">"res"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">res_tab</span> <span class="op">&lt;-</span> <span class="va">res_tab</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">res_tab</span><span class="op">[[</span><span class="st">"q_(Intercept)"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">top_feat</span> <span class="op">&lt;-</span> <span class="va">res_tab</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">n_top</span><span class="op">)</span>, <span class="st">"taxon"</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Create a plot</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">plotExpression</span><span class="op">(</span></span>
<span>  <span class="va">tse_order</span>,</span>
<span>  features <span class="op">=</span> <span class="va">top_feat</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>  x <span class="op">=</span> <span class="st">"sample_geographic.location..country.and.or.sea.region."</span>,</span>
<span>  show_boxplot <span class="op">=</span> <span class="cn">TRUE</span>, show_violin <span class="op">=</span> <span class="cn">FALSE</span>, point_shape <span class="op">=</span> <span class="cn">NA</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tuomas Borman, Ben Allen, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
