<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MGnifyR: interface to MGnify database • MGnifyR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MGnifyR: interface to MGnify database">
<meta property="og:description" content="MGnifyR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MGnifyR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MGnifyR_long.html">MGnifyR: interface to MGnify database</a>
    </li>
    <li>
      <a href="../articles/MgnifyR.html">MGnifyR: interface to MGnify database</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/EBI-Metagenomics/MGnifyR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MGnifyR: interface to MGnify database</h1>
            
            <h4 data-toc-skip class="date">2023-10-30</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/HEAD/vignettes/MgnifyR.Rmd" class="external-link"><code>vignettes/MgnifyR.Rmd</code></a></small>
      <div class="hidden name"><code>MgnifyR.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>MGnifyR</code> is a package designed to ease access to the
EBI’s <a href="https://www.ebi.ac.uk/metagenomics" class="external-link">MGnify</a> resource,
allowing searching and retrieval of multiple datasets for downstream
analysis.</p>
<p>The latest version of MGnifyR seamlessly integrates with the <a href="https://microbiome.github.io/" class="external-link">miaverse framework</a> providing
access to cutting-edge tools in microbiome down-stream analytics.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><code>MGnifyR</code> is currently hosted on GitHub, and can be
installed using via <code>devtools</code>. <code>MGnifyR</code> should
be built using the following snippet.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">MGnifyR</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-mgnifyr-package">Load <code>MGnifyR</code> package<a class="anchor" aria-label="anchor" href="#load-mgnifyr-package"></a>
</h2>
<p>Once installed, <code>MGnifyR</code> is made available in the usual
way.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/EBI-Metagenomics/MGnifyR" class="external-link">MGnifyR</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: mia</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: SummarizedExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: MatrixGenerics</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: matrixStats</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'MatrixGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">##     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">##     rowWeightedSds, rowWeightedVars</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomicRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">##     table, tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomeInfoDb</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biobase'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyMissing, rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: SingleCellExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: TreeSummarizedExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biostrings</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: XVector</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biostrings'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     strsplit</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: MultiAssayExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: biomformat</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'MGnifyR'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     url, warnings</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="create-a-client">Create a client<a class="anchor" aria-label="anchor" href="#create-a-client"></a>
</h2>
<p>All functions in <code>MGnifyR</code> make use of a
<code>MgnifyClient</code> object to keep track of the JSONAPI url, disk
cache location and user access tokens. Thus the first thing to do when
starting any analysis is to instantiate this object. The following
snippet creates this.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MgnifyClient.html">MgnifyClient</a></span><span class="op">(</span>useache <span class="op">=</span> <span class="cn">TRUE</span>, cacheDir <span class="op">=</span> <span class="st">"~/.MGnify_cache"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="functions-for-fetching-the-data">Functions for fetching the data<a class="anchor" aria-label="anchor" href="#functions-for-fetching-the-data"></a>
</h2>
<div class="section level3">
<h3 id="search-data">Search data<a class="anchor" aria-label="anchor" href="#search-data"></a>
</h3>
<p>Below, we fetch information on samples of drinking water.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fetch studies</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doQuery.html">doQuery</a></span><span class="op">(</span></span>
<span>    <span class="va">mg</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"samples"</span>,</span>
<span>    biome_name <span class="op">=</span> <span class="st">"root:Environmental:Aquatic:Freshwater:Drinking water"</span>,</span>
<span>    max.hits <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="find-relevent-analyses-accessions">Find relevent <strong>analyses</strong> accessions<a class="anchor" aria-label="anchor" href="#find-relevent-analyses-accessions"></a>
</h3>
<p>Now we want to find analysis accessions. Each sample might have
multiple analyses. Each analysis ID corresponds to a single run of a
particular pipeline on a single sample in a single study.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analyses_accessions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchAnalysis.html">searchAnalysis</a></span><span class="op">(</span><span class="va">mg</span>, <span class="st">"samples"</span>, <span class="va">samples</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Fetching analyses...</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |======                                                                |   8%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |=============================                                         |  42%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |==================================                                    |  48%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================                                  |  52%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |================================================                      |  68%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |=====================================================                 |  76%
  |                                                                            
  |=======================================================               |  78%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |================================================================      |  92%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |===================================================================   |  96%
  |                                                                            
  |===================================================================== |  98%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">analyses_accessions</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "MGYA00575724" "MGYA00575725" "MGYA00575727" "MGYA00575715" "MGYA00575722"</span></span>
<span><span class="co">## [6] "MGYA00575720"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="fetch-metadata">Fetch metadata<a class="anchor" aria-label="anchor" href="#fetch-metadata"></a>
</h3>
<p>We can now check the metadata to get hint of what kind of data we
have.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analyses_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMetadata.html">getMetadata</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">analyses_accessions</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Fetching metadata...</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |======                                                                |   8%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |=============================                                         |  42%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |==================================                                    |  48%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================                                  |  52%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |================================================                      |  68%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |=====================================================                 |  76%
  |                                                                            
  |=======================================================               |  78%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |================================================================      |  92%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |===================================================================   |  96%
  |                                                                            
  |===================================================================== |  98%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">analyses_metadata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##              analysis_analysis-status analysis_pipeline-version</span></span>
<span><span class="co">## MGYA00575724                completed                       5.0</span></span>
<span><span class="co">## MGYA00575725                completed                       5.0</span></span>
<span><span class="co">## MGYA00575727                completed                       5.0</span></span>
<span><span class="co">## MGYA00575715                completed                       5.0</span></span>
<span><span class="co">## MGYA00575722                completed                       5.0</span></span>
<span><span class="co">## MGYA00575720                completed                       5.0</span></span>
<span><span class="co">##              analysis_accession analysis_experiment-type analysis_is-private</span></span>
<span><span class="co">## MGYA00575724       MGYA00575724              metagenomic               FALSE</span></span>
<span><span class="co">## MGYA00575725       MGYA00575725              metagenomic               FALSE</span></span>
<span><span class="co">## MGYA00575727       MGYA00575727              metagenomic               FALSE</span></span>
<span><span class="co">## MGYA00575715       MGYA00575715              metagenomic               FALSE</span></span>
<span><span class="co">## MGYA00575722       MGYA00575722       metatranscriptomic               FALSE</span></span>
<span><span class="co">## MGYA00575720       MGYA00575720              metagenomic               FALSE</span></span>
<span><span class="co">##              analysis_complete-time analysis_instrument-platform</span></span>
<span><span class="co">## MGYA00575724    2020-11-10T18:30:29                     ILLUMINA</span></span>
<span><span class="co">## MGYA00575725    2020-11-10T18:32:51                     ILLUMINA</span></span>
<span><span class="co">## MGYA00575727    2020-11-10T18:38:27                     ILLUMINA</span></span>
<span><span class="co">## MGYA00575715    2020-11-10T17:57:30                     ILLUMINA</span></span>
<span><span class="co">## MGYA00575722    2020-11-10T18:19:20                     ILLUMINA</span></span>
<span><span class="co">## MGYA00575720    2020-11-10T18:13:57                     ILLUMINA</span></span>
<span><span class="co">##              analysis_instrument-model analysis_Submitted nucleotide sequences</span></span>
<span><span class="co">## MGYA00575724       Illumina HiSeq 4000                                19391040</span></span>
<span><span class="co">## MGYA00575725       Illumina HiSeq 4000                                13081790</span></span>
<span><span class="co">## MGYA00575727       Illumina HiSeq 4000                                15271382</span></span>
<span><span class="co">## MGYA00575715       Illumina HiSeq 4000                                34688870</span></span>
<span><span class="co">## MGYA00575722       Illumina HiSeq 4000                                 3987829</span></span>
<span><span class="co">## MGYA00575720       Illumina HiSeq 4000                                15536089</span></span>
<span><span class="co">##              analysis_Nucleotide sequences after format-specific filtering</span></span>
<span><span class="co">## MGYA00575724                                                       3004272</span></span>
<span><span class="co">## MGYA00575725                                                       1430476</span></span>
<span><span class="co">## MGYA00575727                                                       1887308</span></span>
<span><span class="co">## MGYA00575715                                                       2395935</span></span>
<span><span class="co">## MGYA00575722                                                        571525</span></span>
<span><span class="co">## MGYA00575720                                                       2069502</span></span>
<span><span class="co">##              analysis_Nucleotide sequences after length filtering</span></span>
<span><span class="co">## MGYA00575724                                              3004272</span></span>
<span><span class="co">## MGYA00575725                                              1430476</span></span>
<span><span class="co">## MGYA00575727                                              1887308</span></span>
<span><span class="co">## MGYA00575715                                              2395935</span></span>
<span><span class="co">## MGYA00575722                                               571525</span></span>
<span><span class="co">## MGYA00575720                                              2069502</span></span>
<span><span class="co">##              analysis_Nucleotide sequences after undetermined bases filtering</span></span>
<span><span class="co">## MGYA00575724                                                          3004272</span></span>
<span><span class="co">## MGYA00575725                                                          1430476</span></span>
<span><span class="co">## MGYA00575727                                                          1887308</span></span>
<span><span class="co">## MGYA00575715                                                          2395935</span></span>
<span><span class="co">## MGYA00575722                                                           571525</span></span>
<span><span class="co">## MGYA00575720                                                          2069502</span></span>
<span><span class="co">##              analysis_Reads with predicted CDS</span></span>
<span><span class="co">## MGYA00575724                           2937209</span></span>
<span><span class="co">## MGYA00575725                           1315125</span></span>
<span><span class="co">## MGYA00575727                           1844775</span></span>
<span><span class="co">## MGYA00575715                           2342337</span></span>
<span><span class="co">## MGYA00575722                             70416</span></span>
<span><span class="co">## MGYA00575720                           1925881</span></span>
<span><span class="co">##              analysis_Reads with predicted RNA</span></span>
<span><span class="co">## MGYA00575724                              6051</span></span>
<span><span class="co">## MGYA00575725                              4104</span></span>
<span><span class="co">## MGYA00575727                              5466</span></span>
<span><span class="co">## MGYA00575715                              5788</span></span>
<span><span class="co">## MGYA00575722                            492865</span></span>
<span><span class="co">## MGYA00575720                              5010</span></span>
<span><span class="co">##              analysis_Reads with InterProScan match analysis_Predicted CDS</span></span>
<span><span class="co">## MGYA00575724                                1245195                3041953</span></span>
<span><span class="co">## MGYA00575725                                 523429                1364476</span></span>
<span><span class="co">## MGYA00575727                                 736241                1915057</span></span>
<span><span class="co">## MGYA00575715                                1217927                2428753</span></span>
<span><span class="co">## MGYA00575722                                  21828                  71591</span></span>
<span><span class="co">## MGYA00575720                                 837311                2002834</span></span>
<span><span class="co">##              analysis_Predicted CDS with InterProScan match</span></span>
<span><span class="co">## MGYA00575724                                        1248546</span></span>
<span><span class="co">## MGYA00575725                                         524726</span></span>
<span><span class="co">## MGYA00575727                                         738641</span></span>
<span><span class="co">## MGYA00575715                                        1222496</span></span>
<span><span class="co">## MGYA00575722                                          21870</span></span>
<span><span class="co">## MGYA00575720                                         839747</span></span>
<span><span class="co">##              analysis_Total InterProScan matches</span></span>
<span><span class="co">## MGYA00575724                             2178818</span></span>
<span><span class="co">## MGYA00575725                              922118</span></span>
<span><span class="co">## MGYA00575727                             1303006</span></span>
<span><span class="co">## MGYA00575715                             2161541</span></span>
<span><span class="co">## MGYA00575722                               30925</span></span>
<span><span class="co">## MGYA00575720                             1480643</span></span>
<span><span class="co">##              analysis_Predicted SSU sequences analysis_Predicted LSU sequences</span></span>
<span><span class="co">## MGYA00575724                             1587                             2780</span></span>
<span><span class="co">## MGYA00575725                             1413                             1765</span></span>
<span><span class="co">## MGYA00575727                             1675                             2561</span></span>
<span><span class="co">## MGYA00575715                             1206                             2215</span></span>
<span><span class="co">## MGYA00575722                           223845                           292981</span></span>
<span><span class="co">## MGYA00575720                             1386                             2207</span></span>
<span><span class="co">##              analysis_acc_type study_attributes.bioproject</span></span>
<span><span class="co">## MGYA00575724     analysis-jobs                  PRJEB40814</span></span>
<span><span class="co">## MGYA00575725     analysis-jobs                  PRJEB40814</span></span>
<span><span class="co">## MGYA00575727     analysis-jobs                  PRJEB40814</span></span>
<span><span class="co">## MGYA00575715     analysis-jobs                  PRJEB40814</span></span>
<span><span class="co">## MGYA00575722     analysis-jobs                  PRJEB40814</span></span>
<span><span class="co">## MGYA00575720     analysis-jobs                  PRJEB40814</span></span>
<span><span class="co">##              study_attributes.samples-count study_attributes.accession</span></span>
<span><span class="co">## MGYA00575724                             15               MGYS00005650</span></span>
<span><span class="co">## MGYA00575725                             15               MGYS00005650</span></span>
<span><span class="co">## MGYA00575727                             15               MGYS00005650</span></span>
<span><span class="co">## MGYA00575715                             15               MGYS00005650</span></span>
<span><span class="co">## MGYA00575722                             15               MGYS00005650</span></span>
<span><span class="co">## MGYA00575720                             15               MGYS00005650</span></span>
<span><span class="co">##              study_attributes.is-private study_attributes.secondary-accession</span></span>
<span><span class="co">## MGYA00575724                       FALSE                            ERP124491</span></span>
<span><span class="co">## MGYA00575725                       FALSE                            ERP124491</span></span>
<span><span class="co">## MGYA00575727                       FALSE                            ERP124491</span></span>
<span><span class="co">## MGYA00575715                       FALSE                            ERP124491</span></span>
<span><span class="co">## MGYA00575722                       FALSE                            ERP124491</span></span>
<span><span class="co">## MGYA00575720                       FALSE                            ERP124491</span></span>
<span><span class="co">##              study_attributes.centre-name</span></span>
<span><span class="co">## MGYA00575724                       US EPA</span></span>
<span><span class="co">## MGYA00575725                       US EPA</span></span>
<span><span class="co">## MGYA00575727                       US EPA</span></span>
<span><span class="co">## MGYA00575715                       US EPA</span></span>
<span><span class="co">## MGYA00575722                       US EPA</span></span>
<span><span class="co">## MGYA00575720                       US EPA</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          study_attributes.study-abstract</span></span>
<span><span class="co">## MGYA00575724 The water microbiome in the drinking water distribution systems (DWDSs) of five waterworks in Finland with different raw water sources and treatment processes was explored. The sampled DWDSs were from two waterworks AB with non-disinfected, recharged groundwater as source water and from three waterworks utilizing chlorinated water (two DWDSs of surface waterworks CD and one of ground waterworks E). The water microbiome was characterized by Illumina high-throughput sequencing technology.</span></span>
<span><span class="co">## MGYA00575725 The water microbiome in the drinking water distribution systems (DWDSs) of five waterworks in Finland with different raw water sources and treatment processes was explored. The sampled DWDSs were from two waterworks AB with non-disinfected, recharged groundwater as source water and from three waterworks utilizing chlorinated water (two DWDSs of surface waterworks CD and one of ground waterworks E). The water microbiome was characterized by Illumina high-throughput sequencing technology.</span></span>
<span><span class="co">## MGYA00575727 The water microbiome in the drinking water distribution systems (DWDSs) of five waterworks in Finland with different raw water sources and treatment processes was explored. The sampled DWDSs were from two waterworks AB with non-disinfected, recharged groundwater as source water and from three waterworks utilizing chlorinated water (two DWDSs of surface waterworks CD and one of ground waterworks E). The water microbiome was characterized by Illumina high-throughput sequencing technology.</span></span>
<span><span class="co">## MGYA00575715 The water microbiome in the drinking water distribution systems (DWDSs) of five waterworks in Finland with different raw water sources and treatment processes was explored. The sampled DWDSs were from two waterworks AB with non-disinfected, recharged groundwater as source water and from three waterworks utilizing chlorinated water (two DWDSs of surface waterworks CD and one of ground waterworks E). The water microbiome was characterized by Illumina high-throughput sequencing technology.</span></span>
<span><span class="co">## MGYA00575722 The water microbiome in the drinking water distribution systems (DWDSs) of five waterworks in Finland with different raw water sources and treatment processes was explored. The sampled DWDSs were from two waterworks AB with non-disinfected, recharged groundwater as source water and from three waterworks utilizing chlorinated water (two DWDSs of surface waterworks CD and one of ground waterworks E). The water microbiome was characterized by Illumina high-throughput sequencing technology.</span></span>
<span><span class="co">## MGYA00575720 The water microbiome in the drinking water distribution systems (DWDSs) of five waterworks in Finland with different raw water sources and treatment processes was explored. The sampled DWDSs were from two waterworks AB with non-disinfected, recharged groundwater as source water and from three waterworks utilizing chlorinated water (two DWDSs of surface waterworks CD and one of ground waterworks E). The water microbiome was characterized by Illumina high-throughput sequencing technology.</span></span>
<span><span class="co">##                                                                                                                                            study_attributes.study-name</span></span>
<span><span class="co">## MGYA00575724 Metagenomic and metatranscriptomic analysis of the microbial community in drinking water distribution systems of ground and surface waterworks in Finland</span></span>
<span><span class="co">## MGYA00575725 Metagenomic and metatranscriptomic analysis of the microbial community in drinking water distribution systems of ground and surface waterworks in Finland</span></span>
<span><span class="co">## MGYA00575727 Metagenomic and metatranscriptomic analysis of the microbial community in drinking water distribution systems of ground and surface waterworks in Finland</span></span>
<span><span class="co">## MGYA00575715 Metagenomic and metatranscriptomic analysis of the microbial community in drinking water distribution systems of ground and surface waterworks in Finland</span></span>
<span><span class="co">## MGYA00575722 Metagenomic and metatranscriptomic analysis of the microbial community in drinking water distribution systems of ground and surface waterworks in Finland</span></span>
<span><span class="co">## MGYA00575720 Metagenomic and metatranscriptomic analysis of the microbial community in drinking water distribution systems of ground and surface waterworks in Finland</span></span>
<span><span class="co">##              study_attributes.data-origination study_attributes.last-update</span></span>
<span><span class="co">## MGYA00575724                         SUBMITTED          2021-06-02T16:00:57</span></span>
<span><span class="co">## MGYA00575725                         SUBMITTED          2021-06-02T16:00:57</span></span>
<span><span class="co">## MGYA00575727                         SUBMITTED          2021-06-02T16:00:57</span></span>
<span><span class="co">## MGYA00575715                         SUBMITTED          2021-06-02T16:00:57</span></span>
<span><span class="co">## MGYA00575722                         SUBMITTED          2021-06-02T16:00:57</span></span>
<span><span class="co">## MGYA00575720                         SUBMITTED          2021-06-02T16:00:57</span></span>
<span><span class="co">##              study_accession study_acc_type sample_latitude sample_longitude</span></span>
<span><span class="co">## MGYA00575724    MGYS00005650        studies           61.92            25.75</span></span>
<span><span class="co">## MGYA00575725    MGYS00005650        studies           61.92            25.75</span></span>
<span><span class="co">## MGYA00575727    MGYS00005650        studies           61.92            25.75</span></span>
<span><span class="co">## MGYA00575715    MGYS00005650        studies           61.92            25.75</span></span>
<span><span class="co">## MGYA00575722    MGYS00005650        studies           61.92            25.75</span></span>
<span><span class="co">## MGYA00575720    MGYS00005650        studies           61.92            25.75</span></span>
<span><span class="co">##              sample_biosample sample_accession sample_collection-date</span></span>
<span><span class="co">## MGYA00575724     SAMEA7465223       ERS5222927             2015-09-01</span></span>
<span><span class="co">## MGYA00575725     SAMEA7465220       ERS5222924             2015-08-25</span></span>
<span><span class="co">## MGYA00575727     SAMEA7465213       ERS5222917             2015-08-03</span></span>
<span><span class="co">## MGYA00575715     SAMEA7465227       ERS5222931             2015-09-21</span></span>
<span><span class="co">## MGYA00575722     SAMEA7465224       ERS5222928             2015-08-24</span></span>
<span><span class="co">## MGYA00575720     SAMEA7465221       ERS5222925             2015-08-31</span></span>
<span><span class="co">##                                                                                                                                                                                         sample_sample-desc</span></span>
<span><span class="co">## MGYA00575724            Raw water source [Surface water], Treatment [Ferric sulfate coagulation, clarification, sand filtration, ozonisation, activated carbon filtration], Disinfection [UV-light, NH2Cl]</span></span>
<span><span class="co">## MGYA00575725                          Raw water source [Surface water], Treatment [Ferric sulfate coagulation, flotation, sand filtration, activated carbon filtration], Disinfection [UV-light, ClO2, Cl]</span></span>
<span><span class="co">## MGYA00575727 Raw water source [Artificial groundwater], Treatment [Aeration, lime stabilization, flocculation, clarification, addition of sulphuric acid, sand filtration], Disinfection [No disinfection]</span></span>
<span><span class="co">## MGYA00575715                                                                                    Raw water source [Groundwater], Treatment [Aeration, limestone filtration], Disinfection [UV-light, NaOCl]</span></span>
<span><span class="co">## MGYA00575722            Raw water source [Surface water], Treatment [Ferric sulfate coagulation, clarification, sand filtration, ozonisation, activated carbon filtration], Disinfection [UV-light, NH2Cl]</span></span>
<span><span class="co">## MGYA00575720                          Raw water source [Surface water], Treatment [Ferric sulfate coagulation, flotation, sand filtration, activated carbon filtration], Disinfection [UV-light, ClO2, Cl]</span></span>
<span><span class="co">##              sample_environment-biome         sample_environment-feature</span></span>
<span><span class="co">## MGYA00575724               Freshwater Drinking water distribution system</span></span>
<span><span class="co">## MGYA00575725               Freshwater Drinking water distribution system</span></span>
<span><span class="co">## MGYA00575727               Freshwater Drinking water distribution system</span></span>
<span><span class="co">## MGYA00575715               Freshwater Drinking water distribution system</span></span>
<span><span class="co">## MGYA00575722               Freshwater Drinking water distribution system</span></span>
<span><span class="co">## MGYA00575720               Freshwater Drinking water distribution system</span></span>
<span><span class="co">##              sample_environment-material sample_sample-name sample_sample-alias</span></span>
<span><span class="co">## MGYA00575724                       water    THS_D2_SW_CHM_D     THS_D2_SW_CHM_D</span></span>
<span><span class="co">## MGYA00575725                       water THS_C1_SWW_U_CHL_D  THS_C1_SWW_U_CHL_D</span></span>
<span><span class="co">## MGYA00575727                       water     THS_A1_GW_ND_D      THS_A1_GW_ND_D</span></span>
<span><span class="co">## MGYA00575715                       water    THS_E2_GW_CHL_D     THS_E2_GW_CHL_D</span></span>
<span><span class="co">## MGYA00575722                       water    THS_D1_SW_CHM_R     THS_D1_SW_CHM_R</span></span>
<span><span class="co">## MGYA00575720                       water THS_C2_SWW_U_CHL_D  THS_C2_SWW_U_CHL_D</span></span>
<span><span class="co">##               sample_last-update sample_investigation type</span></span>
<span><span class="co">## MGYA00575724 2021-06-02T16:00:58                metagenome</span></span>
<span><span class="co">## MGYA00575725 2021-06-02T15:55:47                metagenome</span></span>
<span><span class="co">## MGYA00575727 2021-06-02T15:42:10                metagenome</span></span>
<span><span class="co">## MGYA00575715 2021-06-02T15:29:33                metagenome</span></span>
<span><span class="co">## MGYA00575722 2021-06-02T15:26:44         metatranscriptome</span></span>
<span><span class="co">## MGYA00575720 2021-06-02T15:20:26                metagenome</span></span>
<span><span class="co">##                                                              sample_project name</span></span>
<span><span class="co">## MGYA00575724 DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">## MGYA00575725 DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">## MGYA00575727 DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">## MGYA00575715 DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">## MGYA00575722 DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">## MGYA00575720 DWDSOME (Microbiome Dynamics in Drinking Water Distribution System)</span></span>
<span><span class="co">##              sample_geographic location (longitude)</span></span>
<span><span class="co">## MGYA00575724                                  25.75</span></span>
<span><span class="co">## MGYA00575725                                  25.75</span></span>
<span><span class="co">## MGYA00575727                                  25.75</span></span>
<span><span class="co">## MGYA00575715                                  25.75</span></span>
<span><span class="co">## MGYA00575722                                  25.75</span></span>
<span><span class="co">## MGYA00575720                                  25.75</span></span>
<span><span class="co">##              sample_geographic location (country and/or sea,region)</span></span>
<span><span class="co">## MGYA00575724                                                Finland</span></span>
<span><span class="co">## MGYA00575725                                                Finland</span></span>
<span><span class="co">## MGYA00575727                                                Finland</span></span>
<span><span class="co">## MGYA00575715                                                Finland</span></span>
<span><span class="co">## MGYA00575722                                                Finland</span></span>
<span><span class="co">## MGYA00575720                                                Finland</span></span>
<span><span class="co">##              sample_collection date sample_environment (biome)</span></span>
<span><span class="co">## MGYA00575724             2015-09-01                 Freshwater</span></span>
<span><span class="co">## MGYA00575725             2015-08-25                 Freshwater</span></span>
<span><span class="co">## MGYA00575727             2015-08-03                 Freshwater</span></span>
<span><span class="co">## MGYA00575715             2015-09-21                 Freshwater</span></span>
<span><span class="co">## MGYA00575722             2015-08-24                 Freshwater</span></span>
<span><span class="co">## MGYA00575720             2015-08-31                 Freshwater</span></span>
<span><span class="co">##                    sample_environment (feature) sample_environment (material)</span></span>
<span><span class="co">## MGYA00575724 Drinking water distribution system                         water</span></span>
<span><span class="co">## MGYA00575725 Drinking water distribution system                         water</span></span>
<span><span class="co">## MGYA00575727 Drinking water distribution system                         water</span></span>
<span><span class="co">## MGYA00575715 Drinking water distribution system                         water</span></span>
<span><span class="co">## MGYA00575722 Drinking water distribution system                         water</span></span>
<span><span class="co">## MGYA00575720 Drinking water distribution system                         water</span></span>
<span><span class="co">##                                 sample_environmental package</span></span>
<span><span class="co">## MGYA00575724 miscellaneous natural or artificial environment</span></span>
<span><span class="co">## MGYA00575725 miscellaneous natural or artificial environment</span></span>
<span><span class="co">## MGYA00575727 miscellaneous natural or artificial environment</span></span>
<span><span class="co">## MGYA00575715 miscellaneous natural or artificial environment</span></span>
<span><span class="co">## MGYA00575722 miscellaneous natural or artificial environment</span></span>
<span><span class="co">## MGYA00575720 miscellaneous natural or artificial environment</span></span>
<span><span class="co">##              sample_sequencing method sample_geographic location (latitude)</span></span>
<span><span class="co">## MGYA00575724                 Illumina                                 61.92</span></span>
<span><span class="co">## MGYA00575725                 Illumina                                 61.92</span></span>
<span><span class="co">## MGYA00575727                 Illumina                                 61.92</span></span>
<span><span class="co">## MGYA00575715                 Illumina                                 61.92</span></span>
<span><span class="co">## MGYA00575722                 Illumina                                 61.92</span></span>
<span><span class="co">## MGYA00575720                 Illumina                                 61.92</span></span>
<span><span class="co">##              sample_ENA checklist sample_acc_type run_accession</span></span>
<span><span class="co">## MGYA00575724            ERC000025         samples    ERR4702560</span></span>
<span><span class="co">## MGYA00575725            ERC000025         samples    ERR4702557</span></span>
<span><span class="co">## MGYA00575727            ERC000025         samples    ERR4702550</span></span>
<span><span class="co">## MGYA00575715            ERC000025         samples    ERR4702564</span></span>
<span><span class="co">## MGYA00575722            ERC000025         samples    ERR4702561</span></span>
<span><span class="co">## MGYA00575720            ERC000025         samples    ERR4702558</span></span>
<span><span class="co">##                                                                        biome_string</span></span>
<span><span class="co">## MGYA00575724 root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">## MGYA00575725 root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">## MGYA00575727 root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">## MGYA00575715 root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">## MGYA00575722 root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">## MGYA00575720 root:Environmental:Aquatic:Freshwater:Drinking water:Delivery networks</span></span>
<span><span class="co">##              sample_analysis-completed sample_geo-loc-name</span></span>
<span><span class="co">## MGYA00575724                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">## MGYA00575725                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">## MGYA00575727                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">## MGYA00575715                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">## MGYA00575722                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">## MGYA00575720                      &lt;NA&gt;                &lt;NA&gt;</span></span>
<span><span class="co">##              sample_instrument model</span></span>
<span><span class="co">## MGYA00575724                    &lt;NA&gt;</span></span>
<span><span class="co">## MGYA00575725                    &lt;NA&gt;</span></span>
<span><span class="co">## MGYA00575727                    &lt;NA&gt;</span></span>
<span><span class="co">## MGYA00575715                    &lt;NA&gt;</span></span>
<span><span class="co">## MGYA00575722                    &lt;NA&gt;</span></span>
<span><span class="co">## MGYA00575720                    &lt;NA&gt;</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="fetch-microbiome-data">Fetch microbiome data<a class="anchor" aria-label="anchor" href="#fetch-microbiome-data"></a>
</h3>
<p>After we have selected the data to fetch, we can use
<code><a href="../reference/getResult.html">getResult()</a></code></p>
<p>The output is <code>TreeSummarizedExperiment</code>
(<code>TreeSE</code>) or <code>MultiAssayExperiment</code>
(<code>MAE</code>) depending on the dataset. If the dataset includes
only taxonomic profiling data, the output is a single
<code>TreeSE</code>. If dataset includes also functional data, the
output is multiple <code>TreeSE</code> objects that are linked together
by utilizing <code>MAE</code>.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getResult.html">getResult</a></span><span class="op">(</span><span class="va">mg</span>, accession <span class="op">=</span> <span class="va">analyses_accessions</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Fetching functional data...</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |======                                                                |   8%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |=====================                                                 |  30%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR4076725</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |========================                                              |  34%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR4076727</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |===========================                                           |  38%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR4076729</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |=============================                                         |  42%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR4076731</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |===============================                                       |  44%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR4076732</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |==================================                                    |  48%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR4076734</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |===================================                                   |  50%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270676</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |====================================                                  |  52%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270677</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |======================================                                |  54%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270678</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |=======================================                               |  56%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270679</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |=========================================                             |  58%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270680</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |==========================================                            |  60%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270681</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |===========================================                           |  62%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270682</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |=============================================                         |  64%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270683</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |==============================================                        |  66%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270684</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |================================================                      |  68%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270685</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |=================================================                     |  70%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270686</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |==================================================                    |  72%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270687</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |====================================================                  |  74%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270688</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |=====================================================                 |  76%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270689</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |=======================================================               |  78%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270690</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |========================================================              |  80%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270691</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |=========================================================             |  82%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270692</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |===========================================================           |  84%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270693</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |============================================================          |  86%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270694</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |==============================================================        |  88%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270695</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |===============================================================       |  90%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270696</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |================================================================      |  92%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270697</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |==================================================================    |  94%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270698</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |===================================================================   |  96%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270699</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |===================================================================== |  98%</code></pre>
<pre><code><span><span class="co">## Warning: Failed to find column ERR2270700</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |======================================================================| 100%</code></pre>
<pre><code><span><span class="co">## Fetching taxonomy data...</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |======                                                                |   8%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |=============================                                         |  42%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |==================================                                    |  48%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================                                  |  52%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |================================================                      |  68%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |=====================================================                 |  76%
  |                                                                            
  |=======================================================               |  78%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |================================================================      |  92%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |===================================================================   |  96%
  |                                                                            
  |===================================================================== |  98%
  |                                                                            
  |======================================================================| 100%</code></pre>
<pre><code><span><span class="co">## Merging with full join...</span></span></code></pre>
<pre><code><span><span class="co">## 1/50</span></span>
<span><span class="fl">2</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">3</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">4</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">5</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">6</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">7</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">8</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">9</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">10</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">11</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">12</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">13</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">14</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">15</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">16</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">17</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">18</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">19</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">20</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">21</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">22</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">23</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">24</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">25</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">26</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">27</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">28</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">29</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">30</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">31</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">32</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">33</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">34</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">35</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">36</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">37</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">38</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">39</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">40</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">41</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">42</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">43</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">44</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">45</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">46</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">47</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">48</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">49</span><span class="op">/</span><span class="fl">50</span></span>
<span><span class="fl">50</span><span class="op">/</span><span class="fl">50</span></span></code></pre>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 6 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 6:</span></span>
<span><span class="co">##  [1] microbiota: TreeSummarizedExperiment with 3937 rows and 50 columns</span></span>
<span><span class="co">##  [2] go-slim: TreeSummarizedExperiment with 116 rows and 15 columns</span></span>
<span><span class="co">##  [3] go-terms: TreeSummarizedExperiment with 2873 rows and 15 columns</span></span>
<span><span class="co">##  [4] interpro-identifiers: TreeSummarizedExperiment with 15729 rows and 15 columns</span></span>
<span><span class="co">##  [5] taxonomy: TreeSummarizedExperiment with 3518 rows and 50 columns</span></span>
<span><span class="co">##  [6] taxonomy-lsu: TreeSummarizedExperiment with 3246 rows and 19 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>You can get access to individual <code>TreeSE</code> object in
<code>MAE</code> by specifying index or name.</p>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 3937 50 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(3937): 82608 62797 ... 5820 6794</span></span>
<span><span class="co">## rowData names(8): Kingdom Phylum ... Species taxonomy1</span></span>
<span><span class="co">## colnames(50): MGYA00144458 MGYA00144419 ... MGYA00575726 MGYA00575727</span></span>
<span><span class="co">## colData names(63): analysis_analysis.status analysis_pipeline.version</span></span>
<span><span class="co">##   ... sample_geo.loc.name sample_instrument.model</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## rowLinks: NULL</span></span>
<span><span class="co">## rowTree: NULL</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
<p><code>TreeSE</code> object is uniquely positioned to support
<code>SummarizedExperiment</code>-based microbiome data manipulation and
visualization. Moreover, it enables access to <code>miaverse</code>
tools. For example, we can estimate diversity of samples.</p>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/estimateDiversity.html" class="external-link">estimateDiversity</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, index <span class="op">=</span> <span class="st">"shannon"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: scuttle</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: ggplot2</span></span></code></pre>
<div class="sourceCode" id="cb124"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"shannon"</span>, x <span class="op">=</span> <span class="st">"sample_environment.feature"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MgnifyR_files/figure-html/calculate_diversity-1.png" width="700"></p>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Agglomerate data</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExps</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/splitByRanks.html" class="external-link">splitByRanks</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">miaViz</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: ggraph</span></span></code></pre>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot top taxa</span></span>
<span><span class="va">top_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/summaries.html" class="external-link">getTopFeatures</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"Phylum"</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/miaViz/man/plotAbundance.html" class="external-link">plotAbundance</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"Phylum"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>, <span class="op">]</span>, rank <span class="op">=</span> <span class="st">"Phylum"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MgnifyR_files/figure-html/plot_abundance-1.png" width="700"></p>
<p>We can perform principal component analysis to microbial profiling
data by utilizing miaverse tools.</p>
<div class="sourceCode" id="cb128"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Apply relative transformation</span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformCounts</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: The 'transformCounts' function is deprecated. Use 'transformAssay'</span></span>
<span><span class="co">## instead.</span></span></code></pre>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform PCoA</span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html" class="external-link">runMDS</a></span><span class="op">(</span></span>
<span>    <span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>    FUN <span class="op">=</span> <span class="fu">vegan</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></span>, method <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span></span>
<span><span class="co"># Plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"MDS"</span>, colour_by <span class="op">=</span> <span class="st">"sample_environment.feature"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MgnifyR_files/figure-html/pcoa-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="fetch-sequence-files">Fetch sequence files<a class="anchor" aria-label="anchor" href="#fetch-sequence-files"></a>
</h3>
<p>Finally, we can use <code><a href="../reference/getFile.html">searchFile()</a></code> and
<code><a href="../reference/getFile.html">getFile()</a></code> to retrieve other MGnify pipeline outputs such as
merged sequence reads, assembled contigs, and details of the functional
analyses.</p>
<p>With <code><a href="../reference/getFile.html">searchFile()</a></code>, we can search files from the
database.</p>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find list of available downloads, and filter for </span></span>
<span><span class="va">dl_urls</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">searchFile</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">analyses_accessions</span>, type <span class="op">=</span> <span class="st">"analyses"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Searching files...</span></span></code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |======                                                                |   8%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |=============================                                         |  42%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |==================================                                    |  48%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================                                  |  52%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |================================================                      |  68%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |=====================================================                 |  76%
  |                                                                            
  |=======================================================               |  78%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |================================================================      |  92%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |===================================================================   |  96%
  |                                                                            
  |===================================================================== |  98%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb134"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">target_urls</span> <span class="op">&lt;-</span> <span class="va">dl_urls</span><span class="op">[</span></span>
<span>    <span class="va">dl_urls</span><span class="op">$</span><span class="va">attributes.description.label</span> <span class="op">==</span> <span class="st">"Predicted CDS with annotation"</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">target_urls</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         type                                           id</span></span>
<span><span class="co">## 489 analyses ERR2270676_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">## 502 analyses ERR2270677_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">## 515 analyses ERR2270678_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">## 529 analyses ERR2270679_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">## 543 analyses ERR2270680_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">## 557 analyses ERR2270681_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">##                                 attributes.alias attributes.file.format.name</span></span>
<span><span class="co">## 489 ERR2270676_MERGED_FASTQ_CDS_annotated.faa.gz                       FASTA</span></span>
<span><span class="co">## 502 ERR2270677_MERGED_FASTQ_CDS_annotated.faa.gz                       FASTA</span></span>
<span><span class="co">## 515 ERR2270678_MERGED_FASTQ_CDS_annotated.faa.gz                       FASTA</span></span>
<span><span class="co">## 529 ERR2270679_MERGED_FASTQ_CDS_annotated.faa.gz                       FASTA</span></span>
<span><span class="co">## 543 ERR2270680_MERGED_FASTQ_CDS_annotated.faa.gz                       FASTA</span></span>
<span><span class="co">## 557 ERR2270681_MERGED_FASTQ_CDS_annotated.faa.gz                       FASTA</span></span>
<span><span class="co">##     attributes.file.format.extension attributes.file.format.compression</span></span>
<span><span class="co">## 489                            fasta                               TRUE</span></span>
<span><span class="co">## 502                            fasta                               TRUE</span></span>
<span><span class="co">## 515                            fasta                               TRUE</span></span>
<span><span class="co">## 529                            fasta                               TRUE</span></span>
<span><span class="co">## 543                            fasta                               TRUE</span></span>
<span><span class="co">## 557                            fasta                               TRUE</span></span>
<span><span class="co">##      attributes.description.label</span></span>
<span><span class="co">## 489 Predicted CDS with annotation</span></span>
<span><span class="co">## 502 Predicted CDS with annotation</span></span>
<span><span class="co">## 515 Predicted CDS with annotation</span></span>
<span><span class="co">## 529 Predicted CDS with annotation</span></span>
<span><span class="co">## 543 Predicted CDS with annotation</span></span>
<span><span class="co">## 557 Predicted CDS with annotation</span></span>
<span><span class="co">##                         attributes.description.description</span></span>
<span><span class="co">## 489 Predicted coding sequences with InterPro match (FASTA)</span></span>
<span><span class="co">## 502 Predicted coding sequences with InterPro match (FASTA)</span></span>
<span><span class="co">## 515 Predicted coding sequences with InterPro match (FASTA)</span></span>
<span><span class="co">## 529 Predicted coding sequences with InterPro match (FASTA)</span></span>
<span><span class="co">## 543 Predicted coding sequences with InterPro match (FASTA)</span></span>
<span><span class="co">## 557 Predicted coding sequences with InterPro match (FASTA)</span></span>
<span><span class="co">##     attributes.group.type attributes.file.checksum.checksum</span></span>
<span><span class="co">## 489         Sequence data                                  </span></span>
<span><span class="co">## 502         Sequence data                                  </span></span>
<span><span class="co">## 515         Sequence data                                  </span></span>
<span><span class="co">## 529         Sequence data                                  </span></span>
<span><span class="co">## 543         Sequence data                                  </span></span>
<span><span class="co">## 557         Sequence data                                  </span></span>
<span><span class="co">##     attributes.file.checksum.checksum.algorithm</span></span>
<span><span class="co">## 489                                            </span></span>
<span><span class="co">## 502                                            </span></span>
<span><span class="co">## 515                                            </span></span>
<span><span class="co">## 529                                            </span></span>
<span><span class="co">## 543                                            </span></span>
<span><span class="co">## 557                                            </span></span>
<span><span class="co">##     relationships.pipeline.data.type relationships.pipeline.data.id</span></span>
<span><span class="co">## 489                        pipelines                            4.1</span></span>
<span><span class="co">## 502                        pipelines                            4.1</span></span>
<span><span class="co">## 515                        pipelines                            4.1</span></span>
<span><span class="co">## 529                        pipelines                            4.1</span></span>
<span><span class="co">## 543                        pipelines                            4.1</span></span>
<span><span class="co">## 557                        pipelines                            4.1</span></span>
<span><span class="co">##                                          relationships.pipeline.related</span></span>
<span><span class="co">## 489 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/4.1?format=json</span></span>
<span><span class="co">## 502 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/4.1?format=json</span></span>
<span><span class="co">## 515 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/4.1?format=json</span></span>
<span><span class="co">## 529 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/4.1?format=json</span></span>
<span><span class="co">## 543 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/4.1?format=json</span></span>
<span><span class="co">## 557 https://www.ebi.ac.uk/metagenomics/api/v1/pipelines/4.1?format=json</span></span>
<span><span class="co">##                                                                                                          download_url</span></span>
<span><span class="co">## 489 https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00144458/file/ERR2270676_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">## 502 https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00144419/file/ERR2270677_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">## 515 https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00144465/file/ERR2270678_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">## 529 https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00144453/file/ERR2270679_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">## 543 https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00144480/file/ERR2270680_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">## 557 https://www.ebi.ac.uk/metagenomics/api/v1/analyses/MGYA00144495/file/ERR2270681_MERGED_FASTQ_CDS_annotated.faa.gz</span></span>
<span><span class="co">##        accession</span></span>
<span><span class="co">## 489 MGYA00144458</span></span>
<span><span class="co">## 502 MGYA00144419</span></span>
<span><span class="co">## 515 MGYA00144465</span></span>
<span><span class="co">## 529 MGYA00144453</span></span>
<span><span class="co">## 543 MGYA00144480</span></span>
<span><span class="co">## 557 MGYA00144495</span></span></code></pre>
<p>Finally, we can download the files with <code><a href="../reference/getFile.html">getFile()</a></code>.</p>
<div class="sourceCode" id="cb136"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Just select a single file from the target_urls list for demonstration.</span></span>
<span></span>
<span><span class="co"># Default behavior - use local cache.</span></span>
<span><span class="va">cached_location</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">getFile</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">target_urls</span><span class="op">$</span><span class="va">download_url</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Where are the files?</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cached_location</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "NA/analyses/MGYA00144458/file/ERR2270676_MERGED_FASTQ_CDS_annotated.faa.gz"</span></span></code></pre>
<div class="sourceCode" id="cb138"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] miaViz_1.10.0                   ggraph_2.1.0                   </span></span>
<span><span class="co">##  [3] scater_1.30.0                   ggplot2_3.4.4                  </span></span>
<span><span class="co">##  [5] scuttle_1.12.0                  MGnifyR_0.99.1                 </span></span>
<span><span class="co">##  [7] biomformat_1.30.0               mia_1.11.1                     </span></span>
<span><span class="co">##  [9] MultiAssayExperiment_1.28.0     TreeSummarizedExperiment_2.10.0</span></span>
<span><span class="co">## [11] Biostrings_2.70.1               XVector_0.42.0                 </span></span>
<span><span class="co">## [13] SingleCellExperiment_1.24.0     SummarizedExperiment_1.32.0    </span></span>
<span><span class="co">## [15] Biobase_2.62.0                  GenomicRanges_1.54.0           </span></span>
<span><span class="co">## [17] GenomeInfoDb_1.38.0             IRanges_2.36.0                 </span></span>
<span><span class="co">## [19] S4Vectors_0.40.1                BiocGenerics_0.48.0            </span></span>
<span><span class="co">## [21] MatrixGenerics_1.14.0           matrixStats_1.0.0              </span></span>
<span><span class="co">## [23] BiocStyle_2.30.0               </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] jsonlite_1.8.7              magrittr_2.0.3             </span></span>
<span><span class="co">##   [3] ggbeeswarm_0.7.2            farver_2.1.1               </span></span>
<span><span class="co">##   [5] rmarkdown_2.25              fs_1.6.3                   </span></span>
<span><span class="co">##   [7] zlibbioc_1.48.0             ragg_1.2.6                 </span></span>
<span><span class="co">##   [9] vctrs_0.6.4                 memoise_2.0.1              </span></span>
<span><span class="co">##  [11] DelayedMatrixStats_1.24.0   RCurl_1.98-1.12            </span></span>
<span><span class="co">##  [13] ggtree_3.10.0               BiocBaseUtils_1.4.0        </span></span>
<span><span class="co">##  [15] htmltools_0.5.6.1           S4Arrays_1.2.0             </span></span>
<span><span class="co">##  [17] curl_5.1.0                  BiocNeighbors_1.20.0       </span></span>
<span><span class="co">##  [19] Rhdf5lib_1.24.0             gridGraphics_0.5-1         </span></span>
<span><span class="co">##  [21] SparseArray_1.2.0           rhdf5_2.46.0               </span></span>
<span><span class="co">##  [23] sass_0.4.7                  bslib_0.5.1                </span></span>
<span><span class="co">##  [25] desc_1.4.2                  plyr_1.8.9                 </span></span>
<span><span class="co">##  [27] DECIPHER_2.30.0             cachem_1.0.8               </span></span>
<span><span class="co">##  [29] igraph_1.5.1                lifecycle_1.0.3            </span></span>
<span><span class="co">##  [31] pkgconfig_2.0.3             rsvd_1.0.5                 </span></span>
<span><span class="co">##  [33] Matrix_1.6-1.1              R6_2.5.1                   </span></span>
<span><span class="co">##  [35] fastmap_1.1.1               GenomeInfoDbData_1.2.11    </span></span>
<span><span class="co">##  [37] aplot_0.2.2                 digest_0.6.33              </span></span>
<span><span class="co">##  [39] ggnewscale_0.4.9            colorspace_2.1-0           </span></span>
<span><span class="co">##  [41] patchwork_1.1.3             rprojroot_2.0.3            </span></span>
<span><span class="co">##  [43] irlba_2.3.5.1               textshaping_0.3.7          </span></span>
<span><span class="co">##  [45] RSQLite_2.3.2               vegan_2.6-5                </span></span>
<span><span class="co">##  [47] beachmat_2.18.0             labeling_0.4.3             </span></span>
<span><span class="co">##  [49] fansi_1.0.5                 urltools_1.7.3             </span></span>
<span><span class="co">##  [51] polyclip_1.10-6             httr_1.4.7                 </span></span>
<span><span class="co">##  [53] abind_1.4-5                 mgcv_1.9-0                 </span></span>
<span><span class="co">##  [55] compiler_4.3.1              withr_2.5.2                </span></span>
<span><span class="co">##  [57] bit64_4.0.5                 BiocParallel_1.36.0        </span></span>
<span><span class="co">##  [59] viridis_0.6.4               DBI_1.1.3                  </span></span>
<span><span class="co">##  [61] highr_0.10                  ggforce_0.4.1              </span></span>
<span><span class="co">##  [63] MASS_7.3-60                 DelayedArray_0.28.0        </span></span>
<span><span class="co">##  [65] bluster_1.12.0              permute_0.9-7              </span></span>
<span><span class="co">##  [67] tools_4.3.1                 vipor_0.4.5                </span></span>
<span><span class="co">##  [69] beeswarm_0.4.0              ape_5.7-1                  </span></span>
<span><span class="co">##  [71] glue_1.6.2                  nlme_3.1-163               </span></span>
<span><span class="co">##  [73] rhdf5filters_1.14.0         grid_4.3.1                 </span></span>
<span><span class="co">##  [75] cluster_2.1.4               reshape2_1.4.4             </span></span>
<span><span class="co">##  [77] generics_0.1.3              gtable_0.3.4               </span></span>
<span><span class="co">##  [79] tidyr_1.3.0                 tidygraph_1.2.3            </span></span>
<span><span class="co">##  [81] BiocSingular_1.18.0         ScaledMatrix_1.10.0        </span></span>
<span><span class="co">##  [83] utf8_1.2.4                  ggrepel_0.9.4              </span></span>
<span><span class="co">##  [85] pillar_1.9.0                stringr_1.5.0              </span></span>
<span><span class="co">##  [87] yulab.utils_0.1.0           splines_4.3.1              </span></span>
<span><span class="co">##  [89] tweenr_2.0.2                dplyr_1.1.3                </span></span>
<span><span class="co">##  [91] treeio_1.26.0               lattice_0.21-9             </span></span>
<span><span class="co">##  [93] bit_4.0.5                   tidyselect_1.2.0           </span></span>
<span><span class="co">##  [95] DirichletMultinomial_1.44.0 knitr_1.45                 </span></span>
<span><span class="co">##  [97] gridExtra_2.3               bookdown_0.36              </span></span>
<span><span class="co">##  [99] xfun_0.40                   graphlayouts_1.0.1         </span></span>
<span><span class="co">## [101] stringi_1.7.12              ggfun_0.1.3                </span></span>
<span><span class="co">## [103] lazyeval_0.2.2              yaml_2.3.7                 </span></span>
<span><span class="co">## [105] evaluate_0.22               codetools_0.2-19           </span></span>
<span><span class="co">## [107] tibble_3.2.1                BiocManager_1.30.22        </span></span>
<span><span class="co">## [109] ggplotify_0.1.2             cli_3.6.1                  </span></span>
<span><span class="co">## [111] systemfonts_1.0.5           munsell_0.5.0              </span></span>
<span><span class="co">## [113] jquerylib_0.1.4             Rcpp_1.0.11                </span></span>
<span><span class="co">## [115] triebeard_0.4.1             parallel_4.3.1             </span></span>
<span><span class="co">## [117] pkgdown_2.0.7               blob_1.2.4                 </span></span>
<span><span class="co">## [119] sparseMatrixStats_1.14.0    bitops_1.0-7               </span></span>
<span><span class="co">## [121] decontam_1.22.0             viridisLite_0.4.2          </span></span>
<span><span class="co">## [123] tidytree_0.4.5              scales_1.2.1               </span></span>
<span><span class="co">## [125] purrr_1.0.2                 crayon_1.5.2               </span></span>
<span><span class="co">## [127] rlang_1.1.1</span></span></code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Tuomas Borman, Ben Allen, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
