<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MGnifyR">
<title>MGnifyR, extended vignette • MGnifyR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="MGnifyR, extended vignette">
<meta property="og:description" content="MGnifyR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MGnifyR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.29</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/MGnifyR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/MGnifyR_long.html">MGnifyR, extended vignette</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/EBI-Metagenomics/MGnifyR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>MGnifyR, extended vignette</h1>
            
            <h4 data-toc-skip class="date">2024-03-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/HEAD/vignettes/MGnifyR_long.Rmd" class="external-link"><code>vignettes/MGnifyR_long.Rmd</code></a></small>
      <div class="d-none name"><code>MGnifyR_long.Rmd</code></div>
    </div>

    
    
<p><a href="http://github.com/EBI-Metagenomics/MGnifyR" class="external-link">MGnifyR
homepage</a></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>MGnifyR</code> is a package designed to ease access to the
EBI’s <a href="https://www.ebi.ac.uk/metagenomics" class="external-link">MGnify</a> resource,
allowing searching and retrieval of multiple datasets for downstream
analysis. While MGnify pipelines are undoubtedly useful, as currently
implemented they produce results on a strictly per-sample basis. While
some whole study results are available, comparisons across studies are
difficult. The <code>MGnifyR</code> package is designed to facilitate
cross-study analyses by handling all the per-sample data retrieval and
merging details internally, leaving the user free to perform the
analysis as they see fit.</p>
<p>The latest version of MGnifyR seamlessly integrates with the <a href="https://microbiome.github.io/" class="external-link">miaverse framework</a> providing
access to tools in microbiome down-stream analytics. This integration
enables users to leverage optimized and standardized methods for
analyzing the microbiome. Additionally, users can benefit from a
comprehensive tutorial book that offers valuable guidance and
support.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><code>MGnifyR</code> is currently hosted on GitHub, and can be
installed using via <code>devtools</code>. <code>MGnifyR</code> should
be built using the following snippet.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MGnifyR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-mgnifyr-package">Load <code>MGnifyR</code> package<a class="anchor" aria-label="anchor" href="#load-mgnifyr-package"></a>
</h2>
<p>Once installed, <code>MGnifyR</code> is made available in the usual
way.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/EBI-Metagenomics/MGnifyR" class="external-link">MGnifyR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-a-client">Create a client<a class="anchor" aria-label="anchor" href="#create-a-client"></a>
</h2>
<p>All functions in <code>MGnifyR</code> make use of a
<code>MgnifyClient</code> object to keep track of the JSONAPI url, disk
cache location and user access tokens. Thus the first thing to do when
starting any analysis is to instantiate this object. The following
snippet creates this.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MgnifyClient.html">MgnifyClient</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">mg</span></span></code></pre></div>
<p>It’s recommended that local caching is enabled with
<code>useCache = TRUE</code>. Queries to the MGnify API can be quite
slow, particularly when retrieving multipage results for many analyses
(such as many <code>Interpro</code> results). Using a local disk cache
can significantly speed up subsequent work, bypassing the need to
re-query the API. Use of the cache should be entirely transparent, as
the caching occurs at the raw data level. The cache can persist across
<code>MGnifyR</code> sessions, and can even be used for multiple
sessions simultaneously - provided that different sets of accessions are
queried at once.</p>
<p>Optionally, a username and password may be specified during client
creation, causing <code>MGnifyR</code> to attempt retrieval of an
authentication token from the API. Doing so gives access to non-public
results, such as those currently under an author imposed embargo
period.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MgnifyClient.html">MgnifyClient</a></span><span class="op">(</span></span>
<span>    username <span class="op">=</span> <span class="st">"Webin-username"</span>, password <span class="op">=</span> <span class="st">"your-password"</span>, useCache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="functions-for-fetching-the-data">Functions for fetching the data<a class="anchor" aria-label="anchor" href="#functions-for-fetching-the-data"></a>
</h2>
<div class="section level3">
<h3 id="search-data">Search data<a class="anchor" aria-label="anchor" href="#search-data"></a>
</h3>
<p><code>MGnifyR</code> gives users access to the complete range of
search functionality implemented in the MGnify JSON API. A single
function <code><a href="../reference/doQuery.html">doQuery()</a></code> is used to do perform this searching,
allowing Studies, Samples, Runs and Accession to be interrogated from a
common interface. As with all MGnifyR functions the first argument
<code>client</code> must be a valid <code>MgnifyClient</code> instance.
The only remaining <strong>required</strong> parameter is
<code>qtype</code>, specifying the type of data to be queried, and may
be one of <code>studies</code>, <code>samples</code>, <code>runs</code>,
<code>analyses</code> or <code>assemblies</code>. Other general
parameter include <code>max.hits</code>.</p>
<p>Unlike most other <code>MGnifyR</code> high level functions, caching
is turned off by default for <code><a href="../reference/doQuery.html">doQuery()</a></code>. New data and
analyses are being added to MGnify all the time, so enabling caching by
default may lead to out-of-date search results for long-lived sessions.
However, it’s easy to switch back on, and may be useful in many cases.
Also, given the huge and ever increasing number of datasets available in
MGnify, a limit to the number of results returned may be set using
<code>max.hits</code>. By default this is set to 200, which for most
exploratory queries should be sufficient. It may be increased or
decreased by directly specifying <code>max.hits</code>, and disabled
completely (no limit) by setting <code>max.hits=NULL</code>.</p>
<p>In most cases we will want to be more specific about the search, and
will also use either an <code>accession</code> parameter, or the many
filter options available through the API, and discussed below.
Specifying an <code>accession</code> id, which in the case of
<code>samples</code>, <code>runs</code> and <code>assemblies</code> may
be a vector of ids, returns a data.frame of metadata with one row per
matching accession.</p>
<p>If <code>accession</code> is <code>NULL</code> (the default) then
remaining parameters define the filters applied by the API to the search
result. Details of these parameters are given in
<code><a href="../reference/doQuery.html">help(doQuery)</a></code>. By way of example though, supposing we are
interested in amplicon Illumina samples from the arctic, we might try
the following query:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">northpolar</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doQuery.html">doQuery</a></span><span class="op">(</span></span>
<span>    <span class="va">mg</span>, <span class="st">"samples"</span>, latitude_gte<span class="op">=</span><span class="fl">60.0</span>, experiment_type<span class="op">=</span><span class="st">"amplicon"</span>,</span>
<span>    biome_name<span class="op">=</span><span class="st">"Soil"</span>, instrument_platform <span class="op">=</span> <span class="st">"Illumina"</span>, max.hits <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">northpolar</span><span class="op">)</span></span></code></pre></div>
<p>Specifying an <code>accession</code> parameter will restrict results
to just those matching that particular entry, be it a study, sample or
run. For example, to retrieve information for study “MGYS00002891”:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">study_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doQuery.html">doQuery</a></span><span class="op">(</span><span class="va">mg</span>, <span class="st">"studies"</span>, accession<span class="op">=</span><span class="st">"MGYS00002891"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">study_samples</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="find-relevent-analyses-accessions">Find relevent <strong>analyses</strong> accessions<a class="anchor" aria-label="anchor" href="#find-relevent-analyses-accessions"></a>
</h3>
<p>Having obtained a particular set of search hits, it’s now time to
retrieve the associated results. General automated analysis is
complicated by the MGnify database design, wherein for example samples
may be shared between multiple studies, or studies analysed multiple
times using different versions of the pipeline. Navigating these
“many-to-one” relationships can be tricky, so <code>MGnifyR</code>
resorts to using <code>analyses</code> accessions as it’s canonical
identifier. Each analysis corresponds to a single run of a particular
pipeline on a single sample in a single study. The downside of this
approach is that queries returning <code>studies</code>,
<code>samples</code> (or anything other than <code>analyses</code>)
accessions need converting to the corresponding
<code>analyses</code>.</p>
<p><code>MGnifyR</code> therefore provides a helper function to handle
this conversion - <code><a href="../reference/searchAnalysis.html">searchAnalysis()</a></code>. Following on from our
previous search, we have a list of <code>study</code> accessions, so to
convert to corresponding <code>analyses</code> we use:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analyses_accessions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchAnalysis.html">searchAnalysis</a></span><span class="op">(</span></span>
<span>    <span class="va">mg</span>, type<span class="op">=</span><span class="st">"studies"</span>, accession <span class="op">=</span> <span class="va">study_samples</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">analyses_accessions</span><span class="op">)</span></span></code></pre></div>
<p>A useful side effect of the above call is that some attribute
metadata for each sample has now been retrieved and stored in the local
cache. Thus subsequent API calls for these samples (which will occur
multiple times in later steps) will be significantly faster.</p>
<p>It’s important to be aware that the results of a
<code><a href="../reference/searchAnalysis.html">searchAnalysis()</a></code> command will not necessarily be a
one-to-one match with the input accessions. <code>MGnify</code> analysis
runs are sometimes performed multiple times, perhaps using different
versions of the pipeline. Thus further filtering of the result list may
be required, but is easily performed and is illustrated in the next
section.</p>
</div>
<div class="section level3">
<h3 id="fetch-metadata">Fetch metadata<a class="anchor" aria-label="anchor" href="#fetch-metadata"></a>
</h3>
<p>At this point we have a long list of analysis instances (with
potential duplicates) corresponding to the samples previously found. We
use the <code>getMetadata</code> function to download and combine all
associated <code>sample</code>, <code>run</code> and <code>study</code>
metadata, which we then filter as required to include only the rows we
want.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analyses_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMetadata.html">getMetadata</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">analyses_accessions</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">analyses_metadata</span><span class="op">)</span></span></code></pre></div>
<p>The resulting data.frame has columns with names prefixed with their
source type. For example, “sample_xxx” columns correspond to metadata
gleaned from querying an accession’s <code>sample</code> entry. MGnify
allows quite flexible specification of arbitray metadata at submission
time, in many cases leading to quite sparse <code>data.frame</code>
results if accession queries are sourced from more than one study. For
instance, if only one sample contains an entry for “sample_soil_PH”,
entries for other rows will be filled with <code>NA</code>.
<code>MGnifyR</code> does not automatically clean these missing values -
instead opting to allow the the user to choose the a correct action. The
particular study we’re looking at is from the marine biome, suppose we
were interested in only those samples or analyses for which the sampling
depth was known. The following snippet filters the full
<code>data.frame</code> selecting only entries which contain a valid
<code>sample_depth</code>. It’s worth noting the <code>as.numeric</code>
call to ensure the column is converted to <code>numeric</code> type
before it is checked. <em>All</em> sample data from MGnifyR is initially
retrieved as type <code>character</code>, and it’s up to the user to
make sure ostensibly numeric entries are converted properly.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">known_depths</span> <span class="op">&lt;-</span> <span class="va">analyses_metadata</span><span class="op">[</span></span>
<span>    <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">analyses_metadata</span><span class="op">$</span><span class="va">sample_depth</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="co"># How many are left?</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">known_depths</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fetch-microbiome-data">Fetch microbiome data<a class="anchor" aria-label="anchor" href="#fetch-microbiome-data"></a>
</h3>
<p>Having selected the analyses we wish to examine further,
<code><a href="../reference/getResult.html">getResult()</a></code> is used to both download associated OTU tables
and taxonomy, and join all results into a single <em><a href="https://bioconductor.org/packages/3.19/TreeSummarizedExperiment" class="external-link">TreeSummarizedExperiment</a></em>
(<code>TreeSE</code>) object. TreeSE is becoming a defacto standard for
taxonomic abundance <em>munging</em> in R. <code>TreeSE</code> objects
integrate abundance, taxonomic, phylogenetic, sample and sequence data
into a single object, with powerful facilities for filtering, processing
and plotting the results. Compared to <em><a href="https://bioconductor.org/packages/3.19/phyloseq" class="external-link">phyloseq</a></em>
object, <code>TreeSE</code> is more scalable and capable for efficient
data analysis.</p>
<p><code>miaverse</code> framework is developed around
<code>TreeSE</code> data container. It provides tools for analysis and
visualization. Moreover, it includes a comprehensive tutorial book
called <a href="https://microbiome.github.io/OMA/" class="external-link">OMA</a>.</p>
<div class="section level4">
<h4 id="amplicon-sequencing">Amplicon sequencing<a class="anchor" aria-label="anchor" href="#amplicon-sequencing"></a>
</h4>
<p>When the dataset includes amplicon sequencing data, i.e., the dataset
does not include function predictions, <code><a href="../reference/getResult.html">getResult()</a></code> method
returns the dataset as a <code>TreeSE</code> by default. See other
output types from the function documentation.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getResult.html">getResult</a></span><span class="op">(</span><span class="va">mg</span>, accession <span class="op">=</span> <span class="va">analyses_accessions</span>, get.func <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tse</span></span></code></pre></div>
<p><code>TreeSE</code> object is uniquely positioned to support <em><a href="https://bioconductor.org/packages/3.19/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>-based
microbiome data manipulation and visualization. Moreover, it enables
access to <code>miaverse</code> tools. For example, we can estimate
diversity of samples.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia" class="external-link">mia</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/estimateDiversity.html" class="external-link">estimateDiversity</a></span><span class="op">(</span><span class="va">tse</span>, index <span class="op">=</span> <span class="st">"shannon"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"shannon"</span>, x <span class="op">=</span> <span class="st">"sample_geo.loc.name"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">miaViz</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/miaViz/man/plotAbundance.html" class="external-link">plotAbundance</a></span><span class="op">(</span><span class="va">tse</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[[</span><span class="st">"Kingdom"</span><span class="op">]</span><span class="op">]</span> <span class="op">)</span>, <span class="op">]</span>, rank <span class="op">=</span> <span class="st">"Kingdom"</span><span class="op">)</span></span></code></pre></div>
<p>If needed, <code>TreeSE</code> can be converted to
<code>phyloseq</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/makePhyloseqFromTreeSE.html" class="external-link">makePhyloseqFromTreeSE</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="va">pseq</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="metagenomics">Metagenomics<a class="anchor" aria-label="anchor" href="#metagenomics"></a>
</h4>
<p>Although the previous queries have been based on the results from
<code><a href="../reference/doQuery.html">doQuery()</a></code>, from now on we will concentrate on combining and
comparing results from specific studies. Since newly performed analyses
are retrieved first in the <code><a href="../reference/doQuery.html">doQuery()</a></code> call, it’s likely that
by the time this vignette is read, the query results will be different.
This is principally due to the rapid increase in MGnify submissions,
leading to a potential lack of consistency between even closely spaced
queries. As mentioned previously, it may be best to use
<code>useCache=FALSE</code> from <code>MgnifyCLient</code> object for
<code><a href="../reference/doQuery.html">doQuery()</a></code> calls, to ensure queries are actually returning
the latest data.</p>
<p>For the remainder of this vignette however, we’ll be comparing 3
ostensibly different studies. A study of saltmarsh soils from York
University, human faecal samples from a survey of healthy Sardinians,
and a set of samples from hydrothermal vents in the Mid-Cayman rise in
the Carribbean Sea. To simplify things, only the first 20 samples from
each study will be used. Furthermore, the intention is only to
demonstrate the functionality of the MGnifyR package, rather than
produce scientifically rigorous results.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">soil</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchAnalysis.html">searchAnalysis</a></span><span class="op">(</span><span class="va">mg</span>, <span class="st">"studies"</span>, <span class="st">"MGYS00001447"</span><span class="op">)</span></span>
<span><span class="va">human</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchAnalysis.html">searchAnalysis</a></span><span class="op">(</span><span class="va">mg</span>, <span class="st">"studies"</span>, <span class="st">"MGYS00001442"</span><span class="op">)</span></span>
<span><span class="va">marine</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchAnalysis.html">searchAnalysis</a></span><span class="op">(</span><span class="va">mg</span>, <span class="st">"studies"</span>, <span class="st">"MGYS00001282"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine analyses</span></span>
<span><span class="va">all_accessions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">soil</span>, <span class="va">human</span>, <span class="va">marine</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">all_accessions</span><span class="op">)</span></span></code></pre></div>
<p>The first step with this new accession list is, as previously, to
retrieve the associated metadata using <code><a href="../reference/getMetadata.html">getMetadata()</a></code>, and
as seen with the <code><a href="../reference/doQuery.html">doQuery()</a></code> results, the returned
<code>data.frame</code> contains a large number of columns. Being
autogenerated and flexible, the column names can be a little difficult
to predict, but examining <code>colnames(full_metadata)</code> should
make things clearer.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">full_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMetadata.html">getMetadata</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">all_accessions</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">full_metadata</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">full_metadata</span><span class="op">)</span></span></code></pre></div>
<p>From <code>full_metadata</code> we get an idea of the type of data
we’re dealing with, and can extract useul information such as sequencing
platform, source biome, etc. The next code snippet tallies a few of
these columns to give an idea about what’s available. The boxplot also
indicates that while within study read counts are similar, we probably
need to use some sort of normalization procedure when comparing across
samples. We might also want to drop particularly low read coverage
samples from further analysis.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load ggplot2 </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Distribution of sample source material:</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">full_metadata</span><span class="op">$</span><span class="va">`sample_environment-material`</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#What sequencing machine(s) were used?</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">full_metadata</span><span class="op">$</span><span class="va">`sample_instrument model`</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Boxplot of raw read counts:</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>    <span class="va">full_metadata</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">study_accession</span>, y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">`analysis_Submitted nucleotide sequences`</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>group<span class="op">=</span><span class="va">study_accession</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"log(submitted reads)"</span><span class="op">)</span></span></code></pre></div>
<p>Again, we can fetch the data by calling <code><a href="../reference/getResult.html">getResult()</a></code>.
<code>bulk.dl=TRUE</code> has the potential to significantly speed up
data retrieval. MGnify makes its functional results available in two
separate ways, either on a per-analysis basis through the web api, or at
the whole study level as large files, tab separated (TSV), and with
columns representing the results for each analysis. When
<code>bulk.dl</code> is <code>FALSE</code>, <code>MGnifyR</code> queries
the web api to get results which (given some functional analyses results
may consist of thousands of entries) may take significant time. Setting
<code>bulk.dl</code> to <code>TRUE</code> causes <code>MGnifyR</code> to
determine the source study associated with a particular
<code>analysis</code> and to instead download and parse its
corresponding results file. Since this result file contains entries for
all analyses associated with the study, by taking advantage of
<code>MGnifyR</code>’s local caching this single download provides
results for many future analyses. In some cases this affords several
orders of magnitude speedup over the api query case.</p>
<p>Unfortunately, column entries in the per-study results files do not
always directly correspond to those from a particular analysis run,
causing the retrieval to fail. The principal cause of this is believed
to be the running of multiple analyses jobs on the same sample. Thus for
reliability, <code>bulk.dl</code> is <code>FALSE</code> by default. As a
general recommendation though, you should try setting it
<code>TRUE</code> the first time <code><a href="../reference/getResult.html">getResult()</a></code> is used on a
set of accessions. If this fails, setting <code>bulk.dl</code> to
<code>FALSE</code> will enable the more robust approach allowing the
analysis to continue. It might take a while though. Hopefully in the
future the sample/analysis correspondence mismatches will be fixed and
the default <code>bulk.dl</code> will be switch to
<code>TRUE</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getResult.html">getResult</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">all_accessions</span>, bulk.dl <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mae</span></span></code></pre></div>
<p>For metagenomic samples, the result is <em><a href="https://bioconductor.org/packages/3.19/MultiAssayExperiment" class="external-link">MultiAssayExperiment</a></em>
(<code>MAE</code>) which links multiple <code>TreeSE</code> objects into
one dataset. These <code>TreeSE</code> objects include taxonomic
profiling data along with functional data in unique objects. Each
objects is linked with each other by their sample names. You can get
access to individual object or experiment by specifying index or
name.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>We can perform principal component analysis to microbial profiling
data by utilizing miaverse tools.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Apply relative transformation</span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span><span class="co"># Perform PCoA</span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html" class="external-link">runMDS</a></span><span class="op">(</span></span>
<span>    <span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>    FUN <span class="op">=</span> <span class="fu">vegan</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></span>, method <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span></span>
<span><span class="co"># Plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"MDS"</span>, colour_by <span class="op">=</span> <span class="st">"sample_environment.feature"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="fetch-raw-files">Fetch raw files<a class="anchor" aria-label="anchor" href="#fetch-raw-files"></a>
</h3>
<p>While <code><a href="../reference/getResult.html">getResult()</a></code> can be utilized to retrieve microbial
profiling data, <code><a href="../reference/getData.html">getData()</a></code> can be used more flexibly to
retrieve any kind of data from the database. It returns data as simple
data.frame or list format.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kegg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getData.html">getData</a></span><span class="op">(</span></span>
<span>    <span class="va">mg</span>, type <span class="op">=</span> <span class="st">"kegg-modules"</span>, accession <span class="op">=</span> <span class="st">"MGYA00642773"</span>,</span>
<span>    accession.type <span class="op">=</span> <span class="st">"analyses"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">kegg</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fetch-sequence-files">Fetch sequence files<a class="anchor" aria-label="anchor" href="#fetch-sequence-files"></a>
</h3>
<p>Finally, we can use <code><a href="../reference/getFile.html">searchFile()</a></code> and
<code><a href="../reference/getFile.html">getFile()</a></code> to retrieve other MGnify pipeline outputs such as
merged sequence reads, assembled contigs, and details of the functional
analyses. <code><a href="../reference/getFile.html">searchFile()</a></code> is a simple wrapper function which,
when supplied a list of accessions, finds the urls of the files we’re
after. In most cases we’ll want to filter the returned list down to only
the files of interest, which is easily done on the resulting data.frame
object. In addition to the actual download location (the
<code>download_url</code> column), extra columns include file type,
contents and compression. It’s recommended that the
<code>colnames</code> of the <code>data.frame</code> be examined to get
a grasp on the available metadata. To demonstrate the process, the code
below retrieves a data.frame containing all available downloads for each
accession we’ve been examining previously. It then filters this to
retain only those files corresponding retain the annotated amino acid
sequence files.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find list of available downloads</span></span>
<span><span class="va">dl_urls</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">searchFile</a></span><span class="op">(</span></span>
<span>    <span class="va">mg</span>, <span class="va">full_metadata</span><span class="op">$</span><span class="va">analysis_accession</span>, type <span class="op">=</span> <span class="st">"analyses"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter table</span></span>
<span><span class="va">target_urls</span> <span class="op">&lt;-</span> <span class="va">dl_urls</span><span class="op">[</span></span>
<span>    <span class="va">dl_urls</span><span class="op">$</span><span class="va">attributes.description.label</span> <span class="op">==</span> <span class="st">"Predicted CDS with annotation"</span>, <span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">target_urls</span><span class="op">)</span></span></code></pre></div>
<p>To list the types of available files, and guide the filtering,
something like the following might be useful.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">dl_urls</span><span class="op">$</span><span class="va">attributes.description.label</span><span class="op">)</span></span></code></pre></div>
<p>Unlike other <code>MGnifyR</code> functions,
<code><a href="../reference/getFile.html">searchFile()</a></code> is not limited to <code>analyses</code>, and
by specifying <code>accession_type</code> other results types may be
found. For instance, while general <code>genome</code> functionality is
not yet integrated into <code>MGnifyR</code>, we can retrieve associated
files for a particular <code>genome</code> accession with the
following:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genome_urls</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">searchFile</a></span><span class="op">(</span><span class="va">mg</span>, <span class="st">"MGYG000433953"</span>, type <span class="op">=</span> <span class="st">"genomes"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">genome_urls</span><span class="op">[</span> , <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"id"</span>, <span class="st">"attributes.file.format.name"</span>, <span class="st">"download_url"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p>Having found the a set of target urls, the final step is to use
<code><a href="../reference/getFile.html">getFile()</a></code> to actually retrieve the file. Unlike other
functions, this only works with a single url location at once, so each
entry in <code>target_urls</code> from above must be downloaded
individually - easily done by either looping or <code>apply</code>ing
over the list.</p>
<p>If the files are intended to be used with external programs, it might
be easiest to provide a <code>file</code> parameter to the function
call, which specifies a local filename for writing the file. By default
<code>MGnifyR</code> will use the local cache, which can make getting to
the file afterwards more awkward. Regardless, the default behaviour of
<code><a href="../reference/getFile.html">getFile()</a></code> is to retrieve the file specified in the
parameter <code>url</code>, save it to disk, and return the filepath it
was saved to.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Just select a single file from the target_urls list for demonstration.</span></span>
<span></span>
<span><span class="co"># Default behavior - use local cache.</span></span>
<span><span class="va">cached_location1</span> <span class="op">=</span> <span class="fu"><a href="../reference/getFile.html">getFile</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">target_urls</span><span class="op">$</span><span class="va">download_url</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Specifying a file</span></span>
<span><span class="va">cached_location2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">getFile</a></span><span class="op">(</span></span>
<span>    <span class="va">mg</span>, <span class="va">target_urls</span><span class="op">$</span><span class="va">download_url</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cached_location</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">cached_location1</span>, <span class="va">cached_location2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Where are the files?</span></span>
<span><span class="va">cached_location</span></span></code></pre></div>
<p>A second download option is available, which allows built-in parsing
of the file. If we know ahead of time what processing will be performed,
it may be possible to integrate it into a function, pass this function
to <code><a href="../reference/getFile.html">getFile()</a></code> as the <code>read.func</code> argument. The
function in question should take a single argument (the complete path
name of the locally downloaded file) and the result of the call will be
returned in place of the usual output file name.</p>
<p>Alternatively the files could first be downloaded in the standard
way, and then processed using this same function in a loop. Therefore in
many cases the <code>read.func</code> parameter is redundant. However,
many of the outputs from MGnify can be quite large, meaning local
storage of many files may become an issue. By providing a
<code>read_func</code> parameter (and necessarily setting from
<code>MgnifyClient</code> object: <code>useCache=FALSE</code>) analysis
of a large number of datasets may be possible with minimal storage
requirements.</p>
<p>To illustrate, suppose we were interested in retrieving all detected
sequences matching a particular PFAM motif in a set of analyses. The
simple function below uses the <code>Biostrings</code> package to read
an amino acid fasta file, searches for a matching PFAM tag in the
sequence name, and then tallies up the unique sequences into a single
data.frame row. In this case the PFAM motif identifies sequences coding
for the amoC gene, found in both ammonia and methane oxidizing
organisms, but any other filtering method could be used.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Simple function to a count of unique sequences matching PFAM amoC/mmoC motif</span></span>
<span><span class="va">getAmoCseqs</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">fname</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">fname</span><span class="op">)</span></span>
<span>    <span class="va">tgtvec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"PF04896"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sequences</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">sequences</span><span class="op">[</span><span class="va">tgtvec</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Having defined the function, it just remains to include it in the
call to <code><a href="../reference/getFile.html">getFile()</a></code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Just download a single accession for demonstration, specifying a read_function</span></span>
<span><span class="va">amoC_seq_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">getFile</a></span><span class="op">(</span></span>
<span>    <span class="va">mg</span>, <span class="va">target_urls</span><span class="op">$</span><span class="va">download_url</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, read_func <span class="op">=</span> <span class="va">getAmoCseqs</span><span class="op">)</span></span>
<span></span>
<span><span class="va">amoC_seq_counts</span></span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Tuomas Borman, Ben Allen, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
