<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>MGnifyR • MGnifyR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="MGnifyR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MGnifyR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.30</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/MGnifyR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/MGnify_course.html">Metagenomics bioinformatics at MGnify</a></li>
    <li><a class="dropdown-item" href="../articles/MGnifyR_long.html">MGnifyR, extended vignette</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/EBI-Metagenomics/MGnifyR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>MGnifyR</h1>
            
            <h4 data-toc-skip class="date">2024-09-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/EBI-Metagenomics/MGnifyR/blob/minor/vignettes/MGnifyR.Rmd" class="external-link"><code>vignettes/MGnifyR.Rmd</code></a></small>
      <div class="d-none name"><code>MGnifyR.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>MGnifyR</code> is a package designed to ease access to the
EBI’s <a href="https://www.ebi.ac.uk/metagenomics" class="external-link">MGnify</a> resource,
allowing searching and retrieval of multiple datasets for downstream
analysis.</p>
<p>The latest version of MGnifyR seamlessly integrates with the <a href="https://microbiome.github.io/" class="external-link">miaverse framework</a> providing
access to cutting-edge tools in microbiome down-stream analytics.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><code>MGnifyR</code> is hosted on Bioconductor, and can be installed
using via <code>BiocManager</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MGnifyR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-mgnifyr-package">Load <code>MGnifyR</code> package<a class="anchor" aria-label="anchor" href="#load-mgnifyr-package"></a>
</h2>
<p>Once installed, <code>MGnifyR</code> is made available in the usual
way.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/EBI-Metagenomics/MGnifyR" class="external-link">MGnifyR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: MultiAssayExperiment</span></span>
<span><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span><span class="co">#&gt;     tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span><span class="co">#&gt; Loading required package: TreeSummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: SingleCellExperiment</span></span>
<span><span class="co">#&gt; Loading required package: Biostrings</span></span>
<span><span class="co">#&gt; Loading required package: XVector</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biostrings'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     strsplit</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-a-client">Create a client<a class="anchor" aria-label="anchor" href="#create-a-client"></a>
</h2>
<p>All functions in <code>MGnifyR</code> make use of a
<code>MgnifyClient</code> object to keep track of the JSONAPI url, disk
cache location and user access tokens. Thus the first thing to do when
starting any analysis is to instantiate this object. The following
snippet creates this.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MgnifyClient.html">MgnifyClient</a></span><span class="op">(</span>useCache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">mg</span></span>
<span><span class="co">#&gt; An object of class "MgnifyClient"</span></span>
<span><span class="co">#&gt; Slot "databaseUrl":</span></span>
<span><span class="co">#&gt; [1] "https://www.ebi.ac.uk/metagenomics/api/v1"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "authTok":</span></span>
<span><span class="co">#&gt; [1] NA</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "useCache":</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "cacheDir":</span></span>
<span><span class="co">#&gt; [1] "/tmp/RtmpmzZIqo/.MGnifyR_cache"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "showWarnings":</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "clearCache":</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "verbose":</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>The <code>MgnifyClient</code> object contains slots for each of the
previously mentioned settings.</p>
</div>
<div class="section level2">
<h2 id="functions-for-fetching-the-data">Functions for fetching the data<a class="anchor" aria-label="anchor" href="#functions-for-fetching-the-data"></a>
</h2>
<div class="section level3">
<h3 id="search-data">Search data<a class="anchor" aria-label="anchor" href="#search-data"></a>
</h3>
<p><code><a href="../reference/doQuery.html">doQuery()</a></code> function can be utilized to search results
such as samples and studies from MGnify database. Below, we fetch
information drinking water samples.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fetch studies</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doQuery.html">doQuery</a></span><span class="op">(</span></span>
<span>    <span class="va">mg</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"samples"</span>,</span>
<span>    biome_name <span class="op">=</span> <span class="st">"root:Environmental:Aquatic:Freshwater:Drinking water"</span>,</span>
<span>    max.hits <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p>The result is a table containing accession IDs and description – in
this case – on samples.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "biosample"            "accession"            "sample-desc"         </span></span>
<span><span class="co">#&gt; [4] "environment-biome"    "environment-feature"  "environment-material"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="find-relevent-analyses-accessions">Find relevent <strong>analyses</strong> accessions<a class="anchor" aria-label="anchor" href="#find-relevent-analyses-accessions"></a>
</h3>
<p>Now we want to find analysis accessions. Each sample might have
multiple analyses. Each analysis ID corresponds to a single run of a
particular pipeline on a single sample in a single study.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analyses_accessions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/searchAnalysis.html">searchAnalysis</a></span><span class="op">(</span><span class="va">mg</span>, <span class="st">"samples"</span>, <span class="va">samples</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span></span></code></pre></div>
<p>By running the <code><a href="../reference/searchAnalysis.html">searchAnalysis()</a></code> function, we get a
vector of analysis IDs of samples that we fed as an input.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analyses_accessions</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "MGYA00652201" "MGYA00652185" "MGYA00643487" "MGYA00643486" "MGYA00643485"</span></span>
<span><span class="co">#&gt; [6] "MGYA00643484"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fetch-metadata">Fetch metadata<a class="anchor" aria-label="anchor" href="#fetch-metadata"></a>
</h3>
<p>We can now check the metadata to get hint of what kind of data we
have. We use <code><a href="../reference/getMetadata.html">getMetadata()</a></code> function to fetch data based on
analysis IDs.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">analyses_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMetadata.html">getMetadata</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">analyses_accessions</span><span class="op">)</span></span></code></pre></div>
<p>The returned value is a <code>data.frame</code> that includes
metadata for example on how analysis was conducted and what kind of
samples were analyzed.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">analyses_metadata</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "analysis_analysis-status"  "analysis_pipeline-version"</span></span>
<span><span class="co">#&gt; [3] "analysis_experiment-type"  "analysis_accession"       </span></span>
<span><span class="co">#&gt; [5] "analysis_is-private"       "analysis_complete-time"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fetch-microbiome-data">Fetch microbiome data<a class="anchor" aria-label="anchor" href="#fetch-microbiome-data"></a>
</h3>
<p>After we have selected the data to fetch, we can use
<code><a href="../reference/getResult.html">getResult()</a></code></p>
<p>The output is <em><a href="https://bioconductor.org/packages/3.20/TreeSummarizedExperiment" class="external-link">TreeSummarizedExperiment</a></em>
(<code>TreeSE</code>) or <em><a href="https://bioconductor.org/packages/3.20/MultiAssayExperiment" class="external-link">MultiAssayExperiment</a></em>
(<code>MAE</code>) depending on the dataset. If the dataset includes
only taxonomic profiling data, the output is a single
<code>TreeSE</code>. If dataset includes also functional data, the
output is multiple <code>TreeSE</code> objects that are linked together
by utilizing <code>MAE</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getResult.html">getResult</a></span><span class="op">(</span><span class="va">mg</span>, accession <span class="op">=</span> <span class="va">analyses_accessions</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span></span>
<span><span class="co">#&gt; A MultiAssayExperiment object of 6 listed</span></span>
<span><span class="co">#&gt;  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">#&gt;  Containing an ExperimentList class object of length 6:</span></span>
<span><span class="co">#&gt;  [1] microbiota: TreeSummarizedExperiment with 3506 rows and 50 columns</span></span>
<span><span class="co">#&gt;  [2] go-slim: TreeSummarizedExperiment with 116 rows and 38 columns</span></span>
<span><span class="co">#&gt;  [3] go-terms: TreeSummarizedExperiment with 3133 rows and 38 columns</span></span>
<span><span class="co">#&gt;  [4] interpro-identifiers: TreeSummarizedExperiment with 18223 rows and 38 columns</span></span>
<span><span class="co">#&gt;  [5] taxonomy: TreeSummarizedExperiment with 3617 rows and 50 columns</span></span>
<span><span class="co">#&gt;  [6] taxonomy-lsu: TreeSummarizedExperiment with 3378 rows and 42 columns</span></span>
<span><span class="co">#&gt; Functionality:</span></span>
<span><span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">#&gt;  exportClass() - save data to flat files</span></span></code></pre></div>
<p>You can get access to individual <code>TreeSE</code> object in
<code>MAE</code> by specifying index or name.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; class: TreeSummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 3506 50 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(3506): 82608 62797 ... 5820 6794</span></span>
<span><span class="co">#&gt; rowData names(9): Kingdom Phylum ... taxonomy1 taxonomy</span></span>
<span><span class="co">#&gt; colnames(50): MGYA00144458 MGYA00144419 ... MGYA00652185 MGYA00652201</span></span>
<span><span class="co">#&gt; colData names(64): analysis_analysis.status analysis_pipeline.version</span></span>
<span><span class="co">#&gt;   ... sample_geo.loc.name sample_instrument.model</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; rowLinks: NULL</span></span>
<span><span class="co">#&gt; rowTree: NULL</span></span>
<span><span class="co">#&gt; colLinks: NULL</span></span>
<span><span class="co">#&gt; colTree: NULL</span></span></code></pre></div>
<p><code>TreeSE</code> object is uniquely positioned to support
<code>SummarizedExperiment</code>-based microbiome data manipulation and
visualization. Moreover, it enables access to <code>miaverse</code>
tools. For example, we can estimate diversity of samples…</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia" class="external-link">mia</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; This is mia version 1.13.36</span></span>
<span><span class="co">#&gt; - Online documentation and vignettes: https://microbiome.github.io/mia/</span></span>
<span><span class="co">#&gt; - Online book 'Orchestrating Microbiome Analysis (OMA)': https://microbiome.github.io/OMA/docs/devel/</span></span>
<span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/deprecate.html" class="external-link">estimateDiversity</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, index <span class="op">=</span> <span class="st">"shannon"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in estimateDiversity(mae[[1]], index = "shannon"): 'estimateDiversity'</span></span>
<span><span class="co">#&gt; is deprecated. Use 'addAlpha' instead.</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: scuttle</span></span>
<span><span class="co">#&gt; Loading required package: ggplot2</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"shannon"</span>, x <span class="op">=</span> <span class="st">"sample_environment..biome."</span><span class="op">)</span></span></code></pre></div>
<p><img src="MGnifyR_files/figure-html/calculate_diversity-1.png" width="864"></p>
<p>… and plot abundances of most abundant phyla.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Agglomerate data</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExps</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/agglomerate-methods.html" class="external-link">splitByRanks</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">miaViz</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: ggraph</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'miaViz'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:mia':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     plotNMDS</span></span>
<span></span>
<span><span class="co"># Plot top taxa</span></span>
<span><span class="va">top_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/deprecate.html" class="external-link">getTopFeatures</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"Phylum"</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in getTopFeatures(altExp(mae[[1]], "Phylum"), 10): 'getTopFeatures' is</span></span>
<span><span class="co">#&gt; deprecated. Use 'getTop' instead.</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/miaViz/man/plotAbundance.html" class="external-link">plotAbundance</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"Phylum"</span><span class="op">)</span><span class="op">[</span><span class="va">top_taxa</span>, <span class="op">]</span>,</span>
<span>    rank <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>    as.relative <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>    <span class="op">)</span></span>
<span><span class="co">#&gt; Warning: The following values are already present in `metadata` and will be</span></span>
<span><span class="co">#&gt; overwritten: 'agglomerated_by_rank'. Consider using the 'name' argument to</span></span>
<span><span class="co">#&gt; specify alternative names.</span></span></code></pre></div>
<p><img src="MGnifyR_files/figure-html/plot_abundance-1.png" width="700"></p>
<p>We can also perform other analyses such as principal component
analysis to microbial profiling data by utilizing miaverse tools.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Apply relative transformation</span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span><span class="co"># Perform PCoA</span></span>
<span><span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html" class="external-link">runMDS</a></span><span class="op">(</span></span>
<span>    <span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>    FUN <span class="op">=</span> <span class="fu">vegan</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></span>, method <span class="op">=</span> <span class="st">"bray"</span><span class="op">)</span></span>
<span><span class="co"># Plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span></span>
<span>    <span class="va">mae</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"MDS"</span>, colour_by <span class="op">=</span> <span class="st">"sample_environment..biome."</span><span class="op">)</span></span></code></pre></div>
<p><img src="MGnifyR_files/figure-html/pcoa-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="fetch-raw-files">Fetch raw files<a class="anchor" aria-label="anchor" href="#fetch-raw-files"></a>
</h3>
<p>While <code><a href="../reference/getResult.html">getResult()</a></code> can be utilized to retrieve microbial
profiling data, <code><a href="../reference/getData.html">getData()</a></code> can be used more flexibly to
retrieve any kind of data from the database. It returns data as simple
data.frame or list format.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">publications</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getData.html">getData</a></span><span class="op">(</span><span class="va">mg</span>, type <span class="op">=</span> <span class="st">"publications"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">publications</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "document.id"                  "type"                        </span></span>
<span><span class="co">#&gt; [3] "id"                           "attributes.pubmed-id"        </span></span>
<span><span class="co">#&gt; [5] "attributes.pubmed-central-id" "attributes.pub-title"</span></span></code></pre></div>
<p>The result is a <code>data.frame</code> by default. In this case, it
includes information on publications fetched from the data portal.</p>
</div>
<div class="section level3">
<h3 id="fetch-sequence-files">Fetch sequence files<a class="anchor" aria-label="anchor" href="#fetch-sequence-files"></a>
</h3>
<p>Finally, we can use <code><a href="../reference/getFile.html">searchFile()</a></code> and
<code><a href="../reference/getFile.html">getFile()</a></code> to retrieve other MGnify pipeline outputs such as
merged sequence reads, assembled contigs, and details of the functional
analyses.</p>
<p>With <code><a href="../reference/getFile.html">searchFile()</a></code>, we can search files from the
database.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dl_urls</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">searchFile</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">analyses_accessions</span>, type <span class="op">=</span> <span class="st">"analyses"</span><span class="op">)</span></span></code></pre></div>
<p>The returned table contains search results related to analyses that
we fed as an input. The table contains information on file and also URL
address from where the file can be loaded.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">target_urls</span> <span class="op">&lt;-</span> <span class="va">dl_urls</span><span class="op">[</span></span>
<span>    <span class="va">dl_urls</span><span class="op">$</span><span class="va">attributes.description.label</span> <span class="op">==</span> <span class="st">"Predicted alpha tmRNA"</span>, <span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">target_urls</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "type"                               "id"                                </span></span>
<span><span class="co">#&gt; [3] "attributes.alias"                   "attributes.file.format.name"       </span></span>
<span><span class="co">#&gt; [5] "attributes.file.format.extension"   "attributes.file.format.compression"</span></span></code></pre></div>
<p>Finally, we can download the files with <code><a href="../reference/getFile.html">getFile()</a></code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Just select a single file from the target_urls list for demonstration.</span></span>
<span><span class="va">file_url</span> <span class="op">&lt;-</span> <span class="va">target_urls</span><span class="op">$</span><span class="va">download_url</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">cached_location</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getFile.html">getFile</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">file_url</span><span class="op">)</span></span></code></pre></div>
<p>The function returns a path where the file is stored.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Where are the files?</span></span>
<span><span class="va">cached_location</span></span>
<span><span class="co">#&gt; [1] "/.MGnifyR_cache/analyses/MGYA00652201/file/ERZ20300939_alpha_tmRNA.RF01849.fasta.gz"</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] miaViz_1.13.10                  ggraph_2.2.1                   </span></span>
<span><span class="co">#&gt;  [3] scater_1.33.4                   ggplot2_3.5.1                  </span></span>
<span><span class="co">#&gt;  [5] scuttle_1.15.4                  mia_1.13.36                    </span></span>
<span><span class="co">#&gt;  [7] MGnifyR_0.99.30                 TreeSummarizedExperiment_2.13.0</span></span>
<span><span class="co">#&gt;  [9] Biostrings_2.73.1               XVector_0.45.0                 </span></span>
<span><span class="co">#&gt; [11] SingleCellExperiment_1.27.2     MultiAssayExperiment_1.31.5    </span></span>
<span><span class="co">#&gt; [13] SummarizedExperiment_1.35.1     Biobase_2.65.1                 </span></span>
<span><span class="co">#&gt; [15] GenomicRanges_1.57.1            GenomeInfoDb_1.41.1            </span></span>
<span><span class="co">#&gt; [17] IRanges_2.39.2                  S4Vectors_0.43.2               </span></span>
<span><span class="co">#&gt; [19] BiocGenerics_0.51.1             MatrixGenerics_1.17.0          </span></span>
<span><span class="co">#&gt; [21] matrixStats_1.4.1               knitr_1.48                     </span></span>
<span><span class="co">#&gt; [23] BiocStyle_2.33.1               </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] splines_4.4.1               ggplotify_0.1.2            </span></span>
<span><span class="co">#&gt;   [3] urltools_1.7.3              tibble_3.2.1               </span></span>
<span><span class="co">#&gt;   [5] triebeard_0.4.1             polyclip_1.10-7            </span></span>
<span><span class="co">#&gt;   [7] rpart_4.1.23                DirichletMultinomial_1.47.0</span></span>
<span><span class="co">#&gt;   [9] lifecycle_1.0.4             lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [11] MASS_7.3-61                 SnowballC_0.7.1            </span></span>
<span><span class="co">#&gt;  [13] backports_1.5.0             magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;  [15] Hmisc_5.1-3                 sass_0.4.9                 </span></span>
<span><span class="co">#&gt;  [17] rmarkdown_2.28              jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [19] yaml_2.3.10                 DBI_1.2.3                  </span></span>
<span><span class="co">#&gt;  [21] minqa_1.2.8                 abind_1.4-8                </span></span>
<span><span class="co">#&gt;  [23] zlibbioc_1.51.1             purrr_1.0.2                </span></span>
<span><span class="co">#&gt;  [25] yulab.utils_0.1.7           nnet_7.3-19                </span></span>
<span><span class="co">#&gt;  [27] tweenr_2.0.3                sandwich_3.1-0             </span></span>
<span><span class="co">#&gt;  [29] GenomeInfoDbData_1.2.12     ggrepel_0.9.6              </span></span>
<span><span class="co">#&gt;  [31] tokenizers_0.3.0            irlba_2.3.5.1              </span></span>
<span><span class="co">#&gt;  [33] tidytree_0.4.6              vegan_2.6-8                </span></span>
<span><span class="co">#&gt;  [35] rbiom_1.0.3                 tidyjson_0.3.2             </span></span>
<span><span class="co">#&gt;  [37] pkgdown_2.1.1               permute_0.9-7              </span></span>
<span><span class="co">#&gt;  [39] DelayedMatrixStats_1.27.3   codetools_0.2-20           </span></span>
<span><span class="co">#&gt;  [41] DelayedArray_0.31.11        ggforce_0.4.2              </span></span>
<span><span class="co">#&gt;  [43] tidyselect_1.2.1            aplot_0.2.3                </span></span>
<span><span class="co">#&gt;  [45] UCSC.utils_1.1.0            farver_2.1.2               </span></span>
<span><span class="co">#&gt;  [47] lme4_1.1-35.5               ScaledMatrix_1.13.0        </span></span>
<span><span class="co">#&gt;  [49] viridis_0.6.5               base64enc_0.1-3            </span></span>
<span><span class="co">#&gt;  [51] jsonlite_1.8.8              BiocNeighbors_1.99.0       </span></span>
<span><span class="co">#&gt;  [53] decontam_1.25.0             tidygraph_1.3.1            </span></span>
<span><span class="co">#&gt;  [55] Formula_1.2-5               systemfonts_1.1.0          </span></span>
<span><span class="co">#&gt;  [57] ggnewscale_0.5.0            tools_4.4.1                </span></span>
<span><span class="co">#&gt;  [59] treeio_1.29.1               ragg_1.3.3                 </span></span>
<span><span class="co">#&gt;  [61] Rcpp_1.0.13                 glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [63] gridExtra_2.3               SparseArray_1.5.34         </span></span>
<span><span class="co">#&gt;  [65] BiocBaseUtils_1.7.3         xfun_0.47                  </span></span>
<span><span class="co">#&gt;  [67] mgcv_1.9-1                  dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [69] withr_3.0.1                 BiocManager_1.30.25        </span></span>
<span><span class="co">#&gt;  [71] fastmap_1.2.0               boot_1.3-31                </span></span>
<span><span class="co">#&gt;  [73] bluster_1.15.1              fansi_1.0.6                </span></span>
<span><span class="co">#&gt;  [75] digest_0.6.37               rsvd_1.0.5                 </span></span>
<span><span class="co">#&gt;  [77] gridGraphics_0.5-1          R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [79] textshaping_0.4.0           colorspace_2.1-1           </span></span>
<span><span class="co">#&gt;  [81] lpSolve_5.6.21              utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [83] tidyr_1.3.1                 generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [85] data.table_1.16.0           DECIPHER_3.1.4             </span></span>
<span><span class="co">#&gt;  [87] graphlayouts_1.1.1          httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [89] htmlwidgets_1.6.4           S4Arrays_1.5.7             </span></span>
<span><span class="co">#&gt;  [91] pkgconfig_2.0.3             gtable_0.3.5               </span></span>
<span><span class="co">#&gt;  [93] janeaustenr_1.0.0           htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt;  [95] bookdown_0.40               scales_1.3.0               </span></span>
<span><span class="co">#&gt;  [97] ggfun_0.1.6                 rstudioapi_0.16.0          </span></span>
<span><span class="co">#&gt;  [99] reshape2_1.4.4              checkmate_2.3.2            </span></span>
<span><span class="co">#&gt; [101] nlme_3.1-166                nloptr_2.1.1               </span></span>
<span><span class="co">#&gt; [103] cachem_1.1.0                zoo_1.8-12                 </span></span>
<span><span class="co">#&gt; [105] stringr_1.5.1               parallel_4.4.1             </span></span>
<span><span class="co">#&gt; [107] vipor_0.4.7                 foreign_0.8-87             </span></span>
<span><span class="co">#&gt; [109] desc_1.4.3                  pillar_1.9.0               </span></span>
<span><span class="co">#&gt; [111] grid_4.4.1                  vctrs_0.6.5                </span></span>
<span><span class="co">#&gt; [113] slam_0.1-53                 BiocSingular_1.21.3        </span></span>
<span><span class="co">#&gt; [115] beachmat_2.21.6             cluster_2.1.6              </span></span>
<span><span class="co">#&gt; [117] beeswarm_0.4.0              htmlTable_2.4.3            </span></span>
<span><span class="co">#&gt; [119] evaluate_0.24.0             mvtnorm_1.3-1              </span></span>
<span><span class="co">#&gt; [121] cli_3.6.3                   compiler_4.4.1             </span></span>
<span><span class="co">#&gt; [123] rlang_1.1.4                 crayon_1.5.3               </span></span>
<span><span class="co">#&gt; [125] tidytext_0.4.2              labeling_0.4.3             </span></span>
<span><span class="co">#&gt; [127] mediation_4.5.0             plyr_1.8.9                 </span></span>
<span><span class="co">#&gt; [129] fs_1.6.4                    ggbeeswarm_0.7.2           </span></span>
<span><span class="co">#&gt; [131] stringi_1.8.4               viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [133] BiocParallel_1.39.0         assertthat_0.2.1           </span></span>
<span><span class="co">#&gt; [135] munsell_0.5.1               lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt; [137] Matrix_1.7-0                patchwork_1.2.0            </span></span>
<span><span class="co">#&gt; [139] sparseMatrixStats_1.17.2    highr_0.11                 </span></span>
<span><span class="co">#&gt; [141] igraph_2.0.3                memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [143] RcppParallel_5.1.9          bslib_0.8.0                </span></span>
<span><span class="co">#&gt; [145] ggtree_3.13.1               ape_5.8</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tuomas Borman, Ben Allen, Leo Lahti.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
